---
title: "Nebulosa plots for leading edge genes test"
output: html_document
---


```{R}
library(reshape2)
library(dplyr)
library(tidyr)
library(vroom)
library(pheatmap)
library(dplyr)
library(tidyr)
library(reshape2)

#Cell_Type <- gsub("_TREATED_vs_CTRL_GSEA_Sig_Genesets.tsv","",list.files("../Files/Subclusters_GSEA/"))
files <- list.files("../Files/Subclusters_GSEA", full.names = TRUE, pattern = "*_PD_vs_HC_GSEA_SigBHCorrection_Genesets.tsv")
GSEA_Aggr_Table <- vroom(files, id = "Cell_Type")
GSEA_Aggr_Table$Cell_Type <- gsub("../Files/Subclusters_GSEA/|_PD_vs_HC_GSEA_SigBHCorrection_Genesets.tsv","",GSEA_Aggr_Table$Cell_Type)
#GSEA_Aggr_Table$Cell_Type <- #gsub("GLU_Cajal_Retzius","Cajal-Retzius",GSEA_Aggr_Table$Cell_Type)
```


```{R}

Pathway_Table <- read.delim("../Figures/GSEA_and_DE_Gene_Heatmaps/MostCommonDEPathwaysOrder_Subset.tsv")

Pathway_Table$Pathways <- paste("REACTOME_",Pathway_Table$Pathways, sep = "")

Select_Pathways <- c("REACTOME_TOLL_LIKE_RECEPTOR_CASCADES","REACTOME_VESICLE_MEDIATED_TRANSPORT","REACTOME_HSF1_ACTIVATION")

GSEA_Aggr_Table_Select_Pathways <- GSEA_Aggr_Table[GSEA_Aggr_Table$pathway %in% Select_Pathways,]

```


```{R}

test <- unlist(strsplit(GSEA_Aggr_Table_Select_Pathways$leadingEdge[1], split = ", "))

for (path in unique(GSEA_Aggr_Table_Select_Pathways$pathway)){
  temptable <- GSEA_Aggr_Table_Select_Pathways[GSEA_Aggr_Table_Select_Pathways$pathway %in% path,]
  LEvector <- unique(unlist(strsplit(toString(temptable$leadingEdge), split = ", ")))
  assign(paste(path,"_LeadingEdgeGenes", sep = ""),LEvector)
}

HSF1_Frame <- as.data.frame(`REACTOME_HSF1_ACTIVATION_LeadingEdgeGenes`)

write.table(HSF1_Frame, file = "../Files/PD_vs_HC_HSF1_Activation_pathway_table", quote = FALSE, row.names = FALSE)

Toll_Frame <- as.data.frame(`REACTOME_TOLL_LIKE_RECEPTOR_CASCADES_LeadingEdgeGenes`)

write.table(Toll_Frame, file = "../Files/PD_vs_HC_Toll_like_receptor_cascades_pathway_table", quote = FALSE, row.names = FALSE)

Vesicle_Frame <- as.data.frame(`REACTOME_VESICLE_MEDIATED_TRANSPORT_LeadingEdgeGenes`)

write.table(Vesicle_Frame, file = "../Files/PD_vs_HC_vesicle_mediated_transport_pathway_table", quote = FALSE, row.names = FALSE)

```




```{R}

LEvectorFull <- unique(unlist(strsplit(toString(GSEA_Aggr_Table_Select_Pathways$leadingEdge), split = ", ")))

```




###################################################################################################


```{R}

library(reshape2)
library(dplyr)
library(tidyr)
library(vroom)
library(pheatmap)
library(tidyverse)
setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/MTG/Full_Integration_Analysis_OutlierSamplesRemoved/scRNA-seq/Markdowns")

#Cell_Type <- gsub("_TREATED_vs_CTRL_GSEA_Sig_Genesets.tsv","",list.files("../Files/Subclusters_GSEA/"))
files <- list.files("../Files/Subclusters_DE_Genes", full.names = TRUE, pattern = "All_SigGenes_.*_Markers_PD_vs_HC_BF.csv")
DE_Aggr_Table <- vroom(files, id = "Cell_Type")
DE_Aggr_Table$Cell_Type <- gsub("../Files/Subclusters_DE_Genes/|_Markers_PD_vs_HC_BF.csv|All_SigGenes_","",DE_Aggr_Table$Cell_Type)

```


```{R}

DE_Aggr_Table_LEGenes <- DE_Aggr_Table[DE_Aggr_Table$gene %in% LEvectorFull,]

DE_Aggr_Table_LEGenes_meandifference <- DE_Aggr_Table_LEGenes$PD_mean - DE_Aggr_Table_LEGenes$HC_mean

DE_Aggr_Table_LEGenes_absmeandifference <- abs(DE_Aggr_Table_LEGenes_meandifference)

DE_Aggr_Table_LEGenes$AMD <- DE_Aggr_Table_LEGenes_absmeandifference

DE_Aggr_Table_LEGenesAMDOrder <- DE_Aggr_Table_LEGenes[order(DE_Aggr_Table_LEGenes$AMD, decreasing = TRUE),]

Top_Diff_LEGenes <- unique(DE_Aggr_Table_LEGenesAMDOrder$gene)[1:50]

Top_Diff_LEGenes_df <- as.data.frame(Top_Diff_LEGenes)

write.table(Top_Diff_LEGenes_df,"../Files/Top_50_DE_LEGenes.tsv", quote = FALSE, row.names = FALSE, sep = "\t")


```


```{R}

sum(Top_Diff_LEGenes %in% REACTOME_HSF1_ACTIVATION_LeadingEdgeGenes)

sum(Top_Diff_LEGenes %in% REACTOME_TOLL_LIKE_RECEPTOR_CASCADES_LeadingEdgeGenes)

sum(Top_Diff_LEGenes %in% REACTOME_VESICLE_MEDIATED_TRANSPORT_LeadingEdgeGenes)



```

Code for script

```{R}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(stringr)
library(reshape2)
library(sciplot)
library(Nebulosa)
library(scCustomize)

#dir.create("Figures/NebulosaPlots")

SeuratObject <- readRDS("/n/scratch3/users/j/jap0606/FullIntegration_OutlierSamplesRemoved/FullIntegrationOSR_MTG_SeuratFinal.rds")

Idents(SeuratObject) <- "seurat_clusters"

Idents(SeuratObject) <- "CellSubtypes"

Top_Diff_LEGenes_df <- read.delim("../Files/Top_50_DE_LEGenes.tsv", sep = "\t")

customPalette <- colorRampPalette(c("#FDF6B5","#F9C783","#4B1D91"))(200)

Plot_Density_Custom(SeuratObject, "Lars2", custom_palette = customPalette) +
  facet_wrap(.~SeuratObject$case)

NebulosaFigureMaker <- function(SeurObj,Genes){
  for (i in Genes) {
    TempNebulosa <- Plot_Density_Custom(SeuratObject, i, custom_palette = customPalette) +
  facet_wrap(.~SeuratObject$case)
    ggsave(TempNebulosa, filename = paste("Figures/NebulosaPlots/",i,"_Nebulosa_SplitPlot.pdf",sep = ""), device = "pdf", width = 10, height = 4, units = "in")
  }}

NebulosaFigureMaker(SeuratObject,Top_Diff_LEGenes_df$Top_Diff_LEGenes)





```




















