---
title: "Batch5-7_Seurat_Integration_Analysis"
output: html_document
---


Reading in the three different batches and assigning metadata

```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(glmpca)
library(SeuratWrappers)
```



Batch 5,6,7 read in

```{r}
BN0009.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1_ASAP_snRNA-Seq_021721/scRNA-seq/Files/cellranger_matrices/BN0009")
BN0339.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1_ASAP_snRNA-Seq_021721/scRNA-seq/Files/cellranger_matrices/BN0339")
BN0341.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1_ASAP_snRNA-Seq_021721/scRNA-seq/Files/cellranger_matrices/BN0341")
BN0415.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1_ASAP_snRNA-Seq_021721/scRNA-seq/Files/cellranger_matrices/BN0415")
BN0329.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch2_ASAP_snRNA-seq_031621/scRNA-seq/Files/cellranger_matrices/BN0329")
BN0347.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch2_ASAP_snRNA-seq_031621/scRNA-seq/Files/cellranger_matrices/BN0347")
BN0348.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch2_ASAP_snRNA-seq_031621/scRNA-seq/Files/cellranger_matrices/BN0348")
BN0464.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch2_ASAP_snRNA-seq_031621/scRNA-seq/Files/cellranger_matrices/BN0464")
BN0602.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch3_ASAP_snRNA-seq_031821/scRNA-seq/Files/cellranger_matrices/BN0602")
BN0644.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch3_ASAP_snRNA-seq_031821/scRNA-seq/Files/cellranger_matrices/BN0644")
BN1855.data <- Read10X(data.dir = "~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch3_ASAP_snRNA-seq_031821/scRNA-seq/Files/cellranger_matrices/BN1855")
```



```{r}
Batch567_MTG <- CreateSeuratObject(counts = cbind(BN0009.data,
                                           BN0339.data,
                                           BN0341.data,
                                           BN0415.data,
                                           BN0329.data,
                                           BN0347.data,
                                           BN0348.data,
                                           BN0464.data,
                                           BN0602.data,
                                           BN0644.data,
                                           BN1855.data),
                            project = "Batch567_MTG",
                            min.cells = 3)
```


1) sample ID
```{r}
Batch567_MTG@meta.data$sample_ID <- c(rep("BN0009", ncol(BN0009.data)),
                               rep("BN0339", ncol(BN0339.data)),
                               rep("BN0341", ncol(BN0341.data)),
                               rep("BN0415", ncol(BN0415.data)),
                               rep("BN0329", ncol(BN0329.data)),
                               rep("BN0347", ncol(BN0347.data)),
                               rep("BN0348", ncol(BN0348.data)),
                               rep("BN0464", ncol(BN0464.data)),
                               rep("BN0602", ncol(BN0602.data)),
                               rep("BN0644", ncol(BN0644.data)),
                               rep("BN1855", ncol(BN1855.data)))
```

2) case : Healthy Control (HC) or Parkinson Disease (PD)
```{r}
Batch567_MTG@meta.data$case <- c(rep("PD", ncol(BN0009.data)),
                          rep("HC", ncol(BN0339.data)),
                          rep("HC", ncol(BN0341.data)),
                          rep("ILB", ncol(BN0415.data)),
                          rep("PD", ncol(BN0329.data)),
                          rep("HC", ncol(BN0347.data)),
                          rep("PD", ncol(BN0348.data)),
                          rep("ILB", ncol(BN0464.data)),
                          rep("ILB", ncol(BN0602.data)),
                          rep("PD", ncol(BN0644.data)),
                          rep("HC", ncol(BN1855.data))
                          )
```

3) Batch 
```{r}
Batch567_MTG@meta.data$batch <- c(rep("Batch5", ncol(BN0009.data)),
                                rep("Batch5", ncol(BN0339.data)),
                                rep("Batch5", ncol(BN0341.data)),
                                rep("Batch5", ncol(BN0415.data)),
                                rep("Batch6", ncol(BN0329.data)),
                                rep("Batch6", ncol(BN0347.data)),
                                rep("Batch6", ncol(BN0348.data)),
                                rep("Batch6", ncol(BN0464.data)),
                                rep("Batch7", ncol(BN0602.data)),
                                rep("Batch7", ncol(BN0644.data)),
                                rep("Batch7", ncol(BN1855.data)))
```

```{r}
Batch567_MTG[["percent.mt"]] <- PercentageFeatureSet(Batch567_MTG, pattern = "^MT-")

VlnPlot(Batch567_MTG, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, pt.size=0)
```

 An object of class Seurat 
 25381 features across 6549 samples within 1 assay 
 Active assay: RNA (25381 features)


average nfeature_RNA

```{R}
library(stats)
nfeature_RNA <- Batch567_MTG@meta.data$nFeature_RNA
mean(nfeature_RNA)
MAD <- mad(nfeature_RNA, center = median(nfeature_RNA))
threeMAD <- (MAD*3)+median(Batch567_MTG@meta.data$nFeature_RNA)

```



Filtering low quality cells:
```{r}
Batch567_MTG <- subset(Batch567_MTG, subset = nFeature_RNA > 200 & nfeature_RNA < threeMAD | percent.mt < 5)
```



Log Normalizing data: 
```{r}
Batch567_MTG <- NormalizeData(Batch567_MTG, normalization.method = "LogNormalize", scale.factor = 10000)
```



Finding and plotting 2000 most variable features

```{R}
Batch567_MTG <- FindVariableFeatures(Batch567_MTG, selection.method = "vst", nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(Batch567_MTG), 10)

# plot variable features with and without labels
plot1 <- VariableFeaturePlot(Batch567_MTG)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
```

```{R}
plot1
```

```{R}
plot2
```

Run the standard workflow for visualization and clustering :

save list of all genes
```{r}
all.genes <- rownames(Batch567_MTG)
```

center and scale data
```{r}
Batch567_MTG <- ScaleData(Batch567_MTG, features = all.genes, verbose = FALSE)
```


finding the top 30 principal components for cells
```{r}
Batch567_MTG <- RunGLMPCA(Batch567_MTG, features=Batch567_MTG@assays$RNA@var.features, L = 30)
```


```{R}

saveRDS(Batch567_MTG, "Files/Batch567_MTG_PreHarmony.rds")

```

```{R}

Batch567_MTG <- readRDS("Files/Batch567_MTG_PreHarmony.rds")

```

Batch567_MTG@meta.data$sample_ID

```{R}
Batch567_MTG@meta.data$sex <- c(rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0009")),
                          rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0339")),
                          rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0341")),
                          rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0415")),
                          rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0329")),
                          rep("F", sum(Batch567_MTG@meta.data$sample_ID == "BN0347")),
                          rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0348")),
                          rep("F", sum(Batch567_MTG@meta.data$sample_ID == "BN0464")),
                          rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0602")),
                          rep("M", sum(Batch567_MTG@meta.data$sample_ID == "BN0644")),
                          rep("F", sum(Batch567_MTG@meta.data$sample_ID == "BN1855"))
                          )
```


```{R}

Batch567_MTG@meta.data$age_bracket <- c(rep("61-70", sum(Batch567_MTG@meta.data$sample_ID == "BN0009")),
                          rep("81-90", sum(Batch567_MTG@meta.data$sample_ID == "BN0339")),
                          rep("81-90", sum(Batch567_MTG@meta.data$sample_ID == "BN0341")),
                          rep("81-90", sum(Batch567_MTG@meta.data$sample_ID == "BN0415")),
                          rep("71-80", sum(Batch567_MTG@meta.data$sample_ID == "BN0329")),
                          rep("91-100", sum(Batch567_MTG@meta.data$sample_ID == "BN0347")),
                          rep("81-90", sum(Batch567_MTG@meta.data$sample_ID == "BN0348")),
                          rep("81-90", sum(Batch567_MTG@meta.data$sample_ID == "BN0464")),
                          rep("81-90", sum(Batch567_MTG@meta.data$sample_ID == "BN0602")),
                          rep("71-80", sum(Batch567_MTG@meta.data$sample_ID == "BN0644")),
                          rep("81-90", sum(Batch567_MTG@meta.data$sample_ID == "BN1855"))
                          )
```







see contribution of genes to construct each of these principal components.
```{r}
VizDimLoadings(Batch567_MTG, dims = 1:2, reduction = "pca")
```


plot cells in the first two principal components colored by case: 
```{r}
DimPlot(object = Batch567_MTG, reduction = "pca", pt.size = .1, group.by = "case")
```

Coordinate of cells in PC 1 characterized by case: 
```{r}
VlnPlot(object = Batch567_MTG, features = "PC_1", group.by = "case",  pt.size = .1)
```




Run Harmony
```{r}
Batch567_MTG <- RunHarmony(Batch567_MTG, group.by.vars = c("batch","case","sample_ID","sex","age_bracket"), reduction = "glmpca", plot_convergence = TRUE, theta = c(0.4,0.4,0.4,0.4,0.4))
```

cells in harmony axis
```{r}
harmony_embeddings <- Embeddings(Batch567_MTG, 'harmony')
harmony_embeddings[1:5, 1:5]
```

cells in harmony axis

```{r}
DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "batch")
```


```{r}
DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "case")
```




```{r}

Harmony2Dimensions_Batch_Coloured <- DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "batch") + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

Harmony2Dimensions_Batch_Coloured

ggsave(Harmony2Dimensions_Batch_Coloured, filename = "Figures/Harmony2Dimensions_BatchCaseSampleCorrected_BatchColoured_scRNA_seq_Batch567.pdf", device = "pdf", width = 6, height = 4, units = "in")

```





```{R}
testplot <- DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "batch")
testplot[[1]]$layers[[1]]$aes_params$alpha = ifelse (Batch567_MTG@meta.data$batch %in% c("Batch6","Batch7"), .0, .1)
testplot
```


```{r}
testplot <- DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "batch")
testplot[[1]]$layers[[1]]$aes_params$alpha = ifelse (Batch567_MTG@meta.data$batch %in% c("Batch56767","Batch7"), .0, .1)
testplot
```

```{r}
testplot <- DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "batch")
testplot[[1]]$layers[[1]]$aes_params$alpha = ifelse (Batch567_MTG@meta.data$batch %in% c("Batch56767","Batch6"), .0, .1)
testplot
```


```{r}
DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "batch", split.by = "batch")
```

```{r}

Harmony2DimensionsBatchColouredSplit <- DimPlot(object = Batch567_MTG, reduction = "harmony", pt.size = .1, group.by = "batch", split.by = "batch") + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"),
              legend.position = "none")

Harmony2DimensionsBatchColouredSplit

ggsave(Harmony2DimensionsBatchColouredSplit, filename = "Figures/Harmony2Dimensions_Batch_Coloured_Split_scRNA_seq_Batch567.pdf", device = "pdf", width = 6, height = 4, units = "in")
```



cells in harmony 1 axis
```{r}
VlnPlot(object = Batch567_MTG, features = "harmony_1", group.by = "case",  pt.size = .1)
```

 An object of class Seurat 
 25381 features across 6549 samples within 1 assay 
 Active assay: RNA (25381 features)
 2 dimensional reductions calculated: pca, harmony

Determing the dimensionality of the dataset

```{R}
#Batch567_MTG <- JackStraw(Batch567_MTG, num.replicate = 100)
#Batch567_MTG <- ScoreJackStraw(Batch567_MTG, dims = 1:20)
#JackStrawPlot(Batch567_MTG, dims = 1:20)



```

```{R}

ElbowPlot(Batch567_MTG, reduction = "harmony")

```

```{R}

ElbowPlot(Batch567_MTG, reduction = "glmpca")

```


12 looks like a suitable cutoff based on the elbow plot

Finding Clusters of cells:
```{r}
Batch567_MTG <- FindNeighbors(Batch567_MTG, reduction = "harmony", dims = 1:15)
Batch567_MTG <- FindClusters(Batch567_MTG, resolution = 0.5, algorithm = 4, method = "igraph")
```

run Umap based on top 12 harmony axis: 

```{r}
Batch567_MTG <- RunUMAP(Batch567_MTG, reduction = "harmony", dims = 1:15)
```

run tsne based on top 10 harmony axis: 

```{r}
#Batch567_MTG <- RunTSNE(Batch567_MTG, reduction = "harmony", dims = 1:12)
```

plot umap: 
```{r}
DimPlot(Batch567_MTG, reduction = "umap", label = TRUE,pt.size = 0.01)
```

```{r}
DimPlot(Batch567_MTG, reduction = "umap", group.by = "case",pt.size = 0.1)
```

```{r}
DimPlot(Batch567_MTG, reduction = "umap", group.by = "batch",pt.size = 0.1)
```

```{R}
UMAPBatchColouredClusters <- DimPlot(Batch567_MTG, reduction = "umap", group.by = "batch",pt.size = 0.1) + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

UMAPBatchColouredClusters

ggsave(UMAPBatchColouredClusters, filename = "Figures/UMAP_Batch_Coloured_Clusters_scRNA_seq_Batch567.pdf", device = "pdf", width = 6, height = 4, units = "in")




```


```{r}

DimPlot(Batch567_MTG, reduction = "umap", group.by = "batch",pt.size = 0.1,split.by = "batch")

UMAPBatchColouredClustersSplit <- DimPlot(Batch567_MTG, reduction = "umap", group.by = "batch",pt.size = 0.1,split.by = "batch") + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"),
              legend.position = "none")

UMAPBatchColouredClustersSplit

ggsave(UMAPBatchColouredClustersSplit, filename = "Figures/UMAPBatch_Coloured_Clusters_Split_scRNA_seq_Batch567.pdf", device = "pdf", width = 6, height = 4, units = "in")

```

```{r}
DimPlot(Batch567_MTG, label = TRUE, repel = TRUE, pt.size = 0, label.size = 2)

UMAPclusters <- DimPlot(Batch567_MTG, label = TRUE, repel = TRUE, pt.size = 0, label.size = 2.5) + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Figures/Unassigned_BatchCaseSampleSexAgeHarmony_GLMAPCA_UMAPclusters_scRNA_seq_Batch123_MTG.pdf", device = "pdf", width = 6, height = 4, units = "in")
```


```{R}
UMAPclusters <- DimPlot(Batch567_MTG, label = TRUE, repel = TRUE, pt.size = 0, label.size = 2.5, split.by = "case",ncol = 1) + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Figures/Unassigned_BatchCaseSampleSexAgeHarmony_CaseSplit_GLMPCA_UMAPclusters_scRNA_seq_Batch123_MTG.pdf", device = "pdf", width = 6, height = 10, units = "in")
```


```{R}
UMAPclusters <- DimPlot(Batch567_MTG, label = FALSE, repel = TRUE, pt.size = 0, label.size = 2.5, group.by = "case") + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Figures/Unassigned_BatchCaseSampleSexAgeHarmony_CaseGroup_GLMPCA_UMAPclusters_scRNA_seq_Batch123_MTG.pdf", device = "pdf", width = 6, height = 4, units = "in")
```

```{R}
batchumap <- DimPlot(Batch567_MTG, reduction = "umap", group.by = "batch",pt.size = 0.1, )
```



TSNE Plot cells colored by clusters and grouped by case. 
```{r}
#DimPlot(Batch567_MTG, reduction = "tsne", split.by = "case", label = TRUE, ncol = 1)
```

Find markers for every cluster compared to all remaining cells, report only the positive ones
```{r}
#Batch567_MTG.Markers <- FindAllMarkers(Batch567_MTG, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```


```{r}
unique(Batch567_MTG.Markers$cluster)
```

Top marker genes for each clusters: 
```{r}
#Batch567_MTG.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_log2FC)
#write.table(Batch567_MTG.Markers, file = "Files/AllMarkers.txt", col.names = TRUE, sep = "\t", quote = FALSE)
#top10Markers <- Batch567_MTG.Markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)
#write.table(top10Markers, file = "Files/top10Markers.txt", col.names = TRUE, sep = "\t", quote = FALSE)
```

Heatmap for some marker genes: 
```{r}
#features <- unique(top10Markers$gene)
#DoHeatmap(Batch567_MTG, features = features, size = 2, draw.lines = FALSE, angle = 45,
#          hjust = 0.2) + theme(axis.text.y = element_text(size = 5))
```


Neuron  = ENO2, RBFOX3
 Glutamatergic neurons = SLC17A6 (VGLUT2), SLC17A7 (VGLUT1)
 GABAergic neurons = SLC32A1 (VGAT), GAD1, GAD2
 Dopaminergic neurons = TH, SLC6A3, SCL18A2
 Astrocytes  = AQP4, GFAP
 Oligodendrocytes  =  PLP1, MBP
 OPCs  =  VCAN, BCAN,
 Microglia = CX3CR1, P2RY12
 Endothelial cells = FLT1, CLDN5
 

 Glu_Neurons = 1,6,8,17
 GABA_Neurons = 4,5,9,(16),21,22
 Astrocytes = 7,14
 oligo = 0,(15),20
 OPCs= (7),16
 Microglia = 13
 Endothelial = 18,19
 overlapping = 7
 novel = 2,3,10,11,12 - all likely neurons/neuron related
 

```{R}

saveRDS(Batch567_MTG, "Files/Batch567_Unassigned.rds")

```



```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```

Find cell-types by plotting marker genes.

Neuron Markers

```{r}
VlnPlot(Batch567_MTG, features = c("ENO2"),pt.size = 0)
```
Neurons: 1,4,5,6,7,9,11,12,13,14,15,16,19,20,21


```{R}
VlnPlot(Batch567_MTG, features = c("RBFOX3"),pt.size = 0)
```

Glutamatergic neuron markers

```{R}
VlnPlot(Batch567_MTG, features = c("SLC17A6"),pt.size = 0)
```

```{R}
VlnPlot(Batch567_MTG, features = c("SLC17A7"),pt.size = 0)
```
2,3,4,8,9,10,12,15,16,20,21,22 - Glu?

GABAergic markers



```{R}
VlnPlot(Batch567_MTG, features = c("SLC32A1"),pt.size = 0)
```
```{R}
VlnPlot(Batch567_MTG, features = c("GAD1"),pt.size = 0)
```
```{R}
VlnPlot(Batch567_MTG, features = c("GAD2"),pt.size = 0)
```
5,7,14,19,23 - GABA?



Astrocyte markers
```{R}
VlnPlot(Batch567_MTG, features = c("AQP4"),pt.size = 0)
```
```{R}
VlnPlot(Batch567_MTG, features = c("GFAP"),pt.size = 0)
```

```{R}
VlnPlot(Batch567_MTG, features = c("ALDH1L1"),pt.size = 0)
```


5,12 - Astrocytes

Oligodendrocyte markers


```{R}
VlnPlot(Batch567_MTG, features = c("PLP1"),pt.size = 0)
```
```{R}
VlnPlot(Batch567_MTG, features = c("MBP"),pt.size = 0)
```


```{R}
VlnPlot(Batch567_MTG, features = c("MOG"),pt.size = 0)
```

1 - oligodendrocytes - strong



OPC markers

```{R}
VlnPlot(Batch567_MTG, features = c("VCAN"),pt.size = 0)
```
```{R}
VlnPlot(Batch567_MTG, features = c("BCAN"),pt.size = 0)
```

13 - OPCs

Microglia

```{R}
VlnPlot(Batch567_MTG, features = c("CX3CR1"),pt.size = 0)
```
```{R}
VlnPlot(Batch567_MTG, features = c("P2RY12"),pt.size = 0)
```

11 - Microglia

Endothelial Cells

```{R}
VlnPlot(Batch567_MTG, features = c("FLT1"),pt.size = 0)
```


```{R}
VlnPlot(Batch567_MTG, features = c("CLDN5"),pt.size = 0)
```
17,18 - Endothelial cells


```{R}
VlnPlot(Batch567_MTG, features = c("CD69"),pt.size = 0)
```

```{R}
VlnPlot(Batch567_MTG, features = c("CD8A"),pt.size = 0)
```


Cajal-Retzius cell markers

```{R}
VlnPlot(Batch567_MTG, features = c("RELN"),pt.size = 0)
```

```{R}
VlnPlot(Batch567_MTG, features = c("CALB2"),pt.size = 0)
```


```{R}
VlnPlot(Batch567_MTG, features = c("NR2F2"),pt.size = 0)
```

```{R}
VlnPlot(Batch567_MTG, features = c("CNR1"),pt.size = 0)
```


```{R}
VlnPlot(Batch567_MTG, features = c("PAFAH1B1"),pt.size = 0)
```
7 - Cajal-retzius cells


Neural subtype markers

GABAergic subtypes

```{R}
VlnPlot(Batch567_MTG, features = c("VIP"),pt.size = 1)
```

```{R}
VlnPlot(Batch567_MTG, features = c("IL1RAPL2"),pt.size = 0)
```
Il1rapl2


```{R}
VlnPlot(Batch567_MTG, features = c("VIPR2"),pt.size = 0)
```

```{R}
VlnPlot(Batch567_MTG, features = c("PVALB"),pt.size = 1)
```
GLUtamatergic subtype markers

```{R}
Nxph4
VlnPlot(Batch567_MTG, features = c("NXPH4"),pt.size = 1)
```

```{R}
StandardCellMarkerVlnPlot1 <- VlnPlot(Batch567_MTG, features = c("ENO2","RBFOX3","SLC17A6","SLC17A7","SLC32A1","GAD1","GAD2","AQP4","GFAP"),pt.size = 0, ncol = 1)

ggsave(StandardCellMarkerVlnPlot1,filename = "Figures/StandardCellMarkerVlnPlot1.pdf", width = 20, height = 20)  

StandardCellMarkerVlnPlot2 <- VlnPlot(Batch567_MTG, features = c("PLP1","MBP","MOG","VCAN","BCAN","CX3CR1","P2RY12","FLT1","CLDN5","GFAP"),pt.size = 0, ncol = 1) 

ggsave(StandardCellMarkerVlnPlot2,filename = "Figures/StandardCellMarkerVlnPlot2.pdf", width = 20, height = 20)            
```


```{R}
saveRDS(Batch567_MTG,"Files/Batch567_Unassigned.rds")
```

#######################################################################################


```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
```



```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```

```{r}
DimPlot(Batch567_MTG, reduction = "umap", split.by = "case",pt.size = 0.1)
```

2,3,4,8,9,10,12,15,16,20,21,22 - Glu?
6,7,14,19,23 - GABA?
5,12 - Astrocytes
1 - oligodendrocytes - strong
13 - OPCs
11 - Microglia
17,18 - Endothelial cells
7 - Cajal-retzius cells
 novel = 6,- all likely neurons/neuron related
 
 
```{R}
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")

cells_per_cluster_table <- as.data.frame(table(Batch567_MTG$seurat_clusters))

colnames(cells_per_cluster_table) <- c("Cluster","Number of Cells")

write.table(cells_per_cluster_table, file = "Files/Cells_Per_Unassigned_Cluster_Table.tsv", quote = FALSE, row.names = FALSE, sep = "\t")
```
 
 
```{R}
samples_per_cluster_table <- group_by(Batch567_MTG@meta.data, seurat_clusters) %>% summarise(Sample_Count = length(unique(sample_ID)))

colnames(samples_per_cluster_table) <- c("Cluster","Number of Samples")

write.table(samples_per_cluster_table, file = "Files/Samples_Per_Unassigned_Cluster_Table.tsv", quote = FALSE, row.names = FALSE, sep = "\t")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons", `3` = "GLU_Neurons",
                      `4` = "GLU_Neurons",`5` = "Astrocytes",
                      `6` = "GABA_Neurons", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons",`9` = "GLU_Neurons",
                      `10` = "GLU_Neurons", `11` = "Microglia",`12` = "Astrocytes",
                      `13` = "OPCs",`14` = "GABA_Neurons",
                      `15` = "GLU_Neurons", `16`="GLU_Neurons", `17`="Endothelial", `18`="Endothelial",`19`="GABA_Neurons",`20`="GLU_Neurons",`21`="GLU_Neurons",`22`="GLU_Neurons",`23`="GABA_Neurons")
```




Now let's plot cells with the assigned celltypes: 
```{r}
DimPlot(Batch567_MTG, label = TRUE, repel = TRUE, pt.size = 0, label.size = 2)

UMAPclusters <- DimPlot(Batch567_MTG, label = TRUE, repel = TRUE, pt.size = 0, label.size = 2.5) + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Files/Assigned_GLMPCA_Leiden_UMAPclusters_scRNA_seq_Batch567.pdf", device = "pdf", width = 6, height = 4, units = "in")
```

Neuron  = ENO2, RBFOX3
 Glutamatergic neurons = SLC17A6, SLC17A7
 GABAergic neurons = SLC32A1, GAD1, GAD2
 Astrocytes  = AQP4, GFAP
 Oligodendrocytes  =  PLP1, MBP
 OPCs  =  VCAN, BCAN,
 Microglia = CX3CR1, P2RY12
 Endothelial cells = FLT1, CLDN5

```{R}
StandardCellMarkerVlnPlot1Assigned <- VlnPlot(Batch567_MTG, features = c("ENO2","RBFOX3","SLC17A6","SLC17A7","SLC32A1","GAD1","GAD2","AQP4","GFAP"),pt.size = 0, ncol = 1)

ggsave(StandardCellMarkerVlnPlot1Assigned,filename = "Figures/StandardCellMarkerVlnPlot1Assigned.pdf", width = 20, height = 20)  

StandardCellMarkerVlnPlot2Assigned <- VlnPlot(Batch567_MTG, features = c("PLP1","MBP","MOG","VCAN","BCAN","CX3CR1","P2RY12","FLT1","CLDN5","GFAP"),pt.size = 0, ncol = 1) 

ggsave(StandardCellMarkerVlnPlot2Assigned,filename = "Figures/StandardCellMarkerVlnPlot2Assigned.pdf", width = 20, height = 20)            
```


```{R}
Batch567_MTG_Counts <- as.data.frame(RelativeCounts(Batch567_MTG@assays$RNA@counts, scale.factor = 1e6, verbose = TRUE))

testing <- RelativeCounts(Batch567_MTG@assays$RNA@counts, scale.factor = 1e6, verbose = TRUE)

colnames(testing) <- Batch567_MTG@active.ident

colnames(Batch567_MTG_Counts) <- Batch567_MTG@active.ident

saveRDS(Batch567_MTG_Counts, "Files/Batch_MTG_Counts.rds")

saveRDS(testing, "Files/Batch_MTG_Counts_matrix.rds")
```



Extracting cell barcode to cell type assignment to a text file

```{R}

colnames(Batch567_MTG@assays$RNA@counts)[Batch567_MTG@active.ident]

CellTypeAssignment <- as.data.frame(cbind(names(Batch567_MTG@active.ident),as.vector(Batch567_MTG@active.ident)))

colnames(CellTypeAssignment) <- c("Barcode","Cell_Type")

write.table(CellTypeAssignment,file = "Files/CellTypeAssignment.tsv", quote = FALSE, row.names = FALSE, col.names = TRUE, sep = "\t")
```



################################################################################


```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(reshape2)
```


```{R}

Batch567_MTG_Counts_matrix <- readRDS("Files/Batch_MTG_Counts_matrix.rds")

```


```{R}
library(sciplot)

LRRK2_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "LRRK2"),]

clusters <- unique(names(LRRK2_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

LRRK2_mean <- unlist(lapply(clusters, meancluster, matrix=LRRK2_Counts))

LRRK2_sd <- unlist(lapply(clusters, sdcluster, matrix=LRRK2_Counts))

LRRK2_seom <- unlist(lapply(clusters, seomcluster, matrix=LRRK2_Counts))

LRRK2_se <- unlist(lapply(clusters, secluster, matrix=LRRK2_Counts))

LRRK2_table <- as.data.frame(cbind(LRRK2_mean,LRRK2_se,clusters))

colnames(LRRK2_table) <- c("mean","SE","clusters")

LRRK2_table$SE <- as.numeric(LRRK2_table$SE)

LRRK2_table$mean <- as.numeric(LRRK2_table$mean)

LRRK2_table_subset <- LRRK2_table[LRRK2_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

LRRK2_table_subset$clusters <- factor(x=LRRK2_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```

```{R}
cbbpalette <- c("#e7d129","#29e772","#293fe7","#e7299e","#5aa59e","#785aa5","#a55a61","#87a55a","#c9c727","#27c976","#2729c9","#c9277a")


```


 +
        scale_fill_manual(values = c(cbbpalette[5],cbbpalette[6],cbbpalette[7],cbbpalette[9],cbbpalette[3],cbbpalette[12]))


```{R}
ggplot(LRRK2_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("LRRK2") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        




```
```{R}
LRRK2_bar_chart <- ggplot(LRRK2_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("LRRK2") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(LRRK2_bar_chart, filename = "Figures/LRRK2_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```

```{R}
library(sciplot)

RIT2_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "RIT2"),]

clusters <- unique(names(RIT2_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

RIT2_mean <- unlist(lapply(clusters, meancluster, matrix=RIT2_Counts))

RIT2_sd <- unlist(lapply(clusters, sdcluster, matrix=RIT2_Counts))

RIT2_seom <- unlist(lapply(clusters, seomcluster, matrix=RIT2_Counts))

RIT2_se <- unlist(lapply(clusters, secluster, matrix=RIT2_Counts))

RIT2_table <- as.data.frame(cbind(RIT2_mean,RIT2_se,clusters))

colnames(RIT2_table) <- c("mean","SE","clusters")

RIT2_table$SE <- as.numeric(RIT2_table$SE)

RIT2_table$mean <- as.numeric(RIT2_table$mean)

RIT2_table_subset <- RIT2_table[RIT2_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

RIT2_table_subset$clusters <- factor(x=RIT2_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```



```{R}
ggplot(RIT2_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("RIT2") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))

```
```{R}
RIT2_bar_chart <- ggplot(RIT2_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("RIT2") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(RIT2_bar_chart, filename = "Figures/RIT2_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```


```{R}
library(sciplot)

GAK_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "GAK"),]

clusters <- unique(names(GAK_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

GAK_mean <- unlist(lapply(clusters, meancluster, matrix=GAK_Counts))

GAK_sd <- unlist(lapply(clusters, sdcluster, matrix=GAK_Counts))

GAK_seom <- unlist(lapply(clusters, seomcluster, matrix=GAK_Counts))

GAK_se <- unlist(lapply(clusters, secluster, matrix=GAK_Counts))

GAK_table <- as.data.frame(cbind(GAK_mean,GAK_se,clusters))

colnames(GAK_table) <- c("mean","SE","clusters")

GAK_table$SE <- as.numeric(GAK_table$SE)

GAK_table$mean <- as.numeric(GAK_table$mean)

GAK_table_subset <- GAK_table[GAK_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

GAK_table_subset$clusters <- factor(x=GAK_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```



```{R}

ggplot(GAK_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("GAK") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))

```
```{R}

GAK_bar_chart <- ggplot(GAK_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("GAK") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(GAK_bar_chart, filename = "Figures/GAK_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```

```{R}
library(sciplot)

VPS13C_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "VPS13C"),]

clusters <- unique(names(VPS13C_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

VPS13C_mean <- unlist(lapply(clusters, meancluster, matrix=VPS13C_Counts))

VPS13C_sd <- unlist(lapply(clusters, sdcluster, matrix=VPS13C_Counts))

VPS13C_seom <- unlist(lapply(clusters, seomcluster, matrix=VPS13C_Counts))

VPS13C_se <- unlist(lapply(clusters, secluster, matrix=VPS13C_Counts))

VPS13C_table <- as.data.frame(cbind(VPS13C_mean,VPS13C_se,clusters))

colnames(VPS13C_table) <- c("mean","SE","clusters")

VPS13C_table$SE <- as.numeric(VPS13C_table$SE)

VPS13C_table$mean <- as.numeric(VPS13C_table$mean)

VPS13C_table_subset <- VPS13C_table[VPS13C_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

VPS13C_table_subset$clusters <- factor(x=VPS13C_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```



```{R}

ggplot(VPS13C_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("VPS13C") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))

```

```{R}

VPS13C_bar_chart <- ggplot(VPS13C_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("VPS13C") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(VPS13C_bar_chart, filename = "Figures/VPS13C_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```





```{R}
library(sciplot)

SLC1A2_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "SLC1A2"),]

clusters <- unique(names(SLC1A2_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

SLC1A2_mean <- unlist(lapply(clusters, meancluster, matrix=SLC1A2_Counts))

SLC1A2_sd <- unlist(lapply(clusters, sdcluster, matrix=SLC1A2_Counts))

SLC1A2_seom <- unlist(lapply(clusters, seomcluster, matrix=SLC1A2_Counts))

SLC1A2_se <- unlist(lapply(clusters, secluster, matrix=SLC1A2_Counts))

SLC1A2_table <- as.data.frame(cbind(SLC1A2_mean,SLC1A2_se,clusters))

colnames(SLC1A2_table) <- c("mean","SE","clusters")

SLC1A2_table$SE <- as.numeric(SLC1A2_table$SE)

SLC1A2_table$mean <- as.numeric(SLC1A2_table$mean)

SLC1A2_table_subset <- SLC1A2_table[SLC1A2_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

SLC1A2_table_subset$clusters <- factor(x=SLC1A2_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```



```{R}

ggplot(SLC1A2_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("SLC1A2") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
```

```{R}

SLC1A2_bar_chart <- ggplot(SLC1A2_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("SLC1A2") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(SLC1A2_bar_chart, filename = "Figures/SLC1A2_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```

SLC1A3

```{R}
library(sciplot)

SLC1A3_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "SLC1A3"),]

clusters <- unique(names(SLC1A3_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

SLC1A3_mean <- unlist(lapply(clusters, meancluster, matrix=SLC1A3_Counts))

SLC1A3_sd <- unlist(lapply(clusters, sdcluster, matrix=SLC1A3_Counts))

SLC1A3_seom <- unlist(lapply(clusters, seomcluster, matrix=SLC1A3_Counts))

SLC1A3_se <- unlist(lapply(clusters, secluster, matrix=SLC1A3_Counts))

SLC1A3_table <- as.data.frame(cbind(SLC1A3_mean,SLC1A3_se,clusters))

colnames(SLC1A3_table) <- c("mean","SE","clusters")

SLC1A3_table$SE <- as.numeric(SLC1A3_table$SE)

SLC1A3_table$mean <- as.numeric(SLC1A3_table$mean)

SLC1A3_table_subset <- SLC1A3_table[SLC1A3_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

SLC1A3_table_subset$clusters <- factor(x=SLC1A3_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```



```{R}

ggplot(SLC1A3_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("SLC1A3") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))

```

```{R}

SLC1A3_bar_chart <- ggplot(SLC1A3_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("SLC1A3") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(SLC1A3_bar_chart, filename = "Figures/SLC1A3_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```



```{R}
library(sciplot)

AQP4_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "AQP4"),]

clusters <- unique(names(AQP4_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

AQP4_mean <- unlist(lapply(clusters, meancluster, matrix=AQP4_Counts))

AQP4_sd <- unlist(lapply(clusters, sdcluster, matrix=AQP4_Counts))

AQP4_seom <- unlist(lapply(clusters, seomcluster, matrix=AQP4_Counts))

AQP4_se <- unlist(lapply(clusters, secluster, matrix=AQP4_Counts))

AQP4_table <- as.data.frame(cbind(AQP4_mean,AQP4_se,clusters))

colnames(AQP4_table) <- c("mean","SE","clusters")

AQP4_table$SE <- as.numeric(AQP4_table$SE)

AQP4_table$mean <- as.numeric(AQP4_table$mean)

AQP4_table_subset <- AQP4_table[AQP4_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

AQP4_table_subset$clusters <- factor(x=AQP4_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```

Neuron  = ENO2, RBFOX3
 Glutamatergic neurons = SLC17A6 (VGLUT2), SLC17A7 (VGLUT1)
 GABAergic neurons = SLC32A1 (VGAT), GAD1, GAD2
 Dopaminergic neurons = TH, SLC6A3, SCL18A2
 Astrocytes  = AQP4, GFAP
 Oligodendrocytes  =  PLP1, MBP
 OPCs  =  VCAN, BCAN,
 Microglia = CX3CR1, P2RY12
 Endothelial cells = FLT1, CLDN5

```{R}

ggplot(AQP4_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("AQP4") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))

```
```{R}

AQP4_bar_chart <- ggplot(AQP4_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("AQP4") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(AQP4_bar_chart, filename = "Figures/AQP4_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```


```{R}
library(sciplot)

P2RY12_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "P2RY12"),]

clusters <- unique(names(P2RY12_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

P2RY12_mean <- unlist(lapply(clusters, meancluster, matrix=P2RY12_Counts))

P2RY12_sd <- unlist(lapply(clusters, sdcluster, matrix=P2RY12_Counts))

P2RY12_seom <- unlist(lapply(clusters, seomcluster, matrix=P2RY12_Counts))

P2RY12_se <- unlist(lapply(clusters, secluster, matrix=P2RY12_Counts))

P2RY12_table <- as.data.frame(cbind(P2RY12_mean,P2RY12_se,clusters))

colnames(P2RY12_table) <- c("mean","SE","clusters")

P2RY12_table$SE <- as.numeric(P2RY12_table$SE)

P2RY12_table$mean <- as.numeric(P2RY12_table$mean)

P2RY12_table_subset <- P2RY12_table[P2RY12_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

P2RY12_table_subset$clusters <- factor(x=P2RY12_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```



```{R}

ggplot(P2RY12_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("P2RY12") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))

```
```{R}

P2RY12_bar_chart <- ggplot(P2RY12_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("P2RY12") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(P2RY12_bar_chart, filename = "Figures/P2RY12_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```

```{R}
library(sciplot)

PLP1_Counts <- Batch567_MTG_Counts_matrix[which(rownames(Batch567_MTG_Counts_matrix) %in% "PLP1"),]

clusters <- unique(names(PLP1_Counts))

meancluster <- function(x,matrix) {
        meanval <- mean(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

sdcluster <- function(x,matrix) {
        meanval <- sd(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

seom <- function(x) sd(x)/sqrt(length(x))

seomcluster <- function(x,matrix) {
        meanval <- seom(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

secluster <- function(x,matrix) {
        meanval <- se(matrix[which(names(matrix) %in% x)])
        return(meanval)
}

PLP1_mean <- unlist(lapply(clusters, meancluster, matrix=PLP1_Counts))

PLP1_sd <- unlist(lapply(clusters, sdcluster, matrix=PLP1_Counts))

PLP1_seom <- unlist(lapply(clusters, seomcluster, matrix=PLP1_Counts))

PLP1_se <- unlist(lapply(clusters, secluster, matrix=PLP1_Counts))

PLP1_table <- as.data.frame(cbind(PLP1_mean,PLP1_se,clusters))

colnames(PLP1_table) <- c("mean","SE","clusters")

PLP1_table$SE <- as.numeric(PLP1_table$SE)

PLP1_table$mean <- as.numeric(PLP1_table$mean)

PLP1_table_subset <- PLP1_table[PLP1_table$clusters %in% c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"),]

PLP1_table_subset$clusters <- factor(x=PLP1_table_subset$clusters, levels = c("Astrocytes 1", "Astrocytes 2","Microglia","Oligodendrocytes","GLU Neurons","GABA Neurons"))
```



```{R}

ggplot(PLP1_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("PLP1") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))

```
```{R}

PLP1_bar_chart <- ggplot(PLP1_table_subset, aes(x=clusters, y=mean, fill=clusters)) +
        ggtitle("PLP1") +
        geom_bar(stat="identity", width=0.5) +
        ylab("Mean Counts per Million") +
        theme(axis.title.x = element_blank(),
              axis.text.x = element_text(angle = 65, hjust=1),
              plot.title = element_text(hjust = 0.5),
              legend.position = "none") +
        geom_errorbar(aes(ymin=mean-SE, ymax=mean+SE), width=.2,
                 position=position_dodge(.9))
        
        
ggsave(PLP1_bar_chart, filename = "Figures/PLP1_bar_chart_scRNA_seq_Batch567.pdf", device = "pdf", width = 4, height = 4, units = "in")



```
