---
title: "Finding_Progression_Markers_for_Temporal_Cortex_Glial_Cells"
output: html_document
---


```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(wesanderson)
```

```{R}
GWAS_PD_Genes <- c("KRTCAP2","NUCKS1","GBA","SIPA1L2","ITPKB","FCGR2A","PMVK","VAMP4","KCNS3","MAP4K4","TMEM163","RAB29","STK39","KCNIP3","IP6K2","SATB1","SPTSSB","TMEM175","MCCC1","BST1","GAK","FAM47E","KPNA1","MED12L","LCORL","SCARB2","SNCA","HLA-DRB5","ELOVL7","CAMK2D","CLCN3","RIMS1","C5orf24","GPNMB","CTSB","RPS12","FYN","TRIM40","BIN3","FGF20","SH3GL2","ITGA8","GBF1","UBAP2","RNF141","FAM49B","SCAF11","DLG2","BAG3","IGSF9B","INPP5F","FBRSL1","CAB39L","GALC","CHD9","GCH1","HIP1R","FGF20","CASC16","SYT17","CD19","MIPOL1","CHRNB1","WNT3","NOD2","RETREG3","UBTF","SETD1A","CRHR1","GRN","MED13","SPPL2B","DYRK1A","DNAH17","MEX3C","ASXL3","CRLS1","LRRK2","RIT1","VPS13C")

```

Genes removed due to not being in annotation: LINC00693, FAM47E-STBD1, LOC100131289, GS1-124K5.11




```{R}
#lapply(X=GWAS_PD_Genes, FUN=VlnPlot, object = Batch567_MTG)



```



```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```


```{R}
Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `0` = "Oligodendrocytes", `1` = "GLU Neurons", `2` = "GLU Neurons",
                      `3` = "GLU Neurons", `4` = "GABA Neurons", `5` = "GABA Neurons",
                      `6` = "GLU Neurons", `7` = "Astrocytes", `8` = "GLU Neurons",`9` = "GABA Neurons",
                      `10` = "GLU Neurons", `11` = "GLU Neurons",`12` = "GLU Neurons",
                      `13` = "Microglia",`14` = "Astrocytes",
                      `15` = "Oligodendrocytes", `16`="GABA Neurons", `17`="GLU Neurons", `18`="Endothelial",`19`="Endothelial",`20`="Oligodendrocytes",`21`="GABA Neurons",`22`="GABA Neurons")
```


```{R}

DimPlot(Batch567_MTG, label = TRUE, repel = TRUE, pt.size = 0, label.size = 2.5) + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

```


NOTE: MAST uses the conservative bonferroni correction to generate the adjusted p-values (p-value divided by the number of tests)

GLU_Neurons

```{R}
GLU_Neurons <- subset(Batch567_MTG, idents = "GLU Neurons")
Idents(GLU_Neurons) <- "case"
GLU_Neurons.PD_Prog.Markers <- FindMarkers(GLU_Neurons, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST")
GLU_Neurons.PD_Prog.Markers$gene <- rownames(GLU_Neurons.PD_Prog.Markers)
GLU_Neurons.PD_Prog.Markers_Filtered <- GLU_Neurons.PD_Prog.Markers[GLU_Neurons.PD_Prog.Markers$p_val_adj <= 0.05,]
top_GLU_Neurons.PD_Prog.Markers <- GLU_Neurons.PD_Prog.Markers_Filtered[1:50,]
#top_pos_GLU_Neurons.PD_Prog.Markers <- GLU_Neurons.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = avg_log2FC)
#top_neg_GLU_Neurons.PD_Prog.Markers <- GLU_Neurons.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = -avg_log2FC)
```

```{R}
avg.GLU_Neurons <- as.data.frame(AverageExpression(GLU_Neurons, verbose = FALSE)$RNA)
avg.GLU_Neurons$gene <- rownames(avg.GLU_Neurons)
avg.GLU_Neurons_topmarkers <- avg.GLU_Neurons[avg.GLU_Neurons$gene %in% unique(top_GLU_Neurons.PD_Prog.Markers$gene),]
#negfeatures_GLU_Neurons <- unique(top10_neg_GLU_Neurons.PD_Prog.Markers$gene)
#features <- c(posfeatures, negfeatures)
avg.GLU_Neurons_topmarkers <- avg.GLU_Neurons_topmarkers[match(top_GLU_Neurons.PD_Prog.Markers$gene,avg.GLU_Neurons_topmarkers$gene),]
top_GLU_Neurons.PD_Prog.Markers$PD_mean <- avg.GLU_Neurons_topmarkers$PD
top_GLU_Neurons.PD_Prog.Markers$HC_mean <- avg.GLU_Neurons_topmarkers$HC
top_GLU_Neurons.PD_Prog.Markers$Status <- "Upregulated"
top_GLU_Neurons.PD_Prog.Markers$Status[top_GLU_Neurons.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_GLU_Neurons.PD_Prog.Markers$Status <- factor(top_GLU_Neurons.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosGLU_NeuronsGenes <- top_GLU_Neurons.PD_Prog.Markers$gene[top_GLU_Neurons.PD_Prog.Markers$avg_log2FC > 1]
NegGLU_NeuronsGenes <- top_GLU_Neurons.PD_Prog.Markers$gene[top_GLU_Neurons.PD_Prog.Markers$avg_log2FC < -1]
```


```{R}

GLU_NeuronsScatter <- ggplot(top_GLU_Neurons.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("GLU_Neurons") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
                
GLU_NeuronsScatter <- LabelPoints(plot = GLU_NeuronsScatter, points = PosGLU_NeuronsGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
GLU_NeuronsScatter <- LabelPoints(plot = GLU_NeuronsScatter, points = NegGLU_NeuronsGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
GLU_NeuronsScatter

ggsave(GLU_NeuronsScatter, filename = "Figures/GLU_NeuronsScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")

#DoHeatmap(GLU_Neurons, features = top_pos_GLU_Neurons.PD_Prog.Markers$gene, size = 4, angle = 0,
#                 hjust = 0.5) + ggtitle("GLU_Neurons") + theme(axis.text.y = element_text(size = 8))

#VlnPlot(GLU_Neurons, features = top_pos_GLU_Neurons.PD_Prog.Markers$gene, adjust = TRUE, pt.size = 0)
```


```{R}

GLU_Neurons_marker_melt <- melt(GLU_Neurons_GNA14)

GLU_Neurons_barchart_table <- group_by(GLU_Neurons_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

Darjeeling <- wes_palette("Darjeeling1")

ggplot(GLU_Neurons_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.5, colors="black") +
        theme(axis.title = element_blank(), axis.text.x = element_blank(),
              axis.ticks = element_blank(),
              strip.text.y = element_text(size = 12, angle = 180, face = "bold")) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

GLU_Neurons_HCvsPDGOI_barchart_MTG_large <- ggplot(GLU_Neurons_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.5, colors="black") +
        theme(axis.title = element_blank(), axis.text.x = element_blank(),
              axis.ticks = element_blank(),
              strip.text.y = element_text(size = 12, angle = 180, face = "bold")) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(GLU_Neurons_HCvsPDGOI_barchart_MTG_large, filename = "Figures/GLU_Neurons_HCvsPDGOI_barchart_MTG_large.pdf", device = "pdf", width = 4, height = 4, units = "in")

GLU_Neurons_HCvsPDGOI_barchart_MTG <- ggplot(GLU_Neurons_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("GLU_Neurons") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(GLU_Neurons_HCvsPDGOI_barchart_MTG, filename = "Figures/GLU_Neurons_HCvsPDGOI_barchart_MTG.pdf", device = "pdf", width = 2.1, height = 1, units = "in")

```


Testing for differential expression of GWAS PD genes

```{R}

GLU_Neurons.PD_Prog.Markers_Filtered_GWAS <- GLU_Neurons.PD_Prog.Markers_Filtered[GLU_Neurons.PD_Prog.Markers_Filtered$gene %in% GWAS_PD_Genes,]

```


data_counts <- FetchData(Batch567_MTG, vars = c("ident",rownames(Batch567_MTG@assays$RNA@counts)), slot = "counts")

scaledrowSums <- 1e6/rowSums(data_counts[2:length(colnames(data_counts))])

AstrocyteCellIds <- rownames(Astro_GNA14) %in% scaledrowSums

match(AstrocyteCellIds,names(scaledrowSums))

AstrocyteScaledrowSums <- scaledrowSums[match(AstrocyteCellIds,names(scaledrowSums))]

Astro_GNA14$GNA14 <- Astro_GNA14$GNA14*AstrocyteScaledrowSums

Astro_GNA14$SLC14A1 <- Astro_GNA14$SLC14A1*AstrocyteScaledrowSums

Astro_marker_melt <- melt(Astro_GNA14)

Astro_barchart_table <- group_by(Astro_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))






ggplot(Astro_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.5, colors="black") +
        theme(axis.title = element_blank(), axis.text.x = element_text(size = 12, angle = 45, face = "bold", vjust = 0.5),
              strip.text.y = element_text(size = 12, angle = 180, face = "bold"))

Looking at expression of MT-encoded genes

```{R}
GLU_Neurons.PD_Prog.Markers_Filtered_MT <- GLU_Neurons.PD_Prog.Markers_Filtered[grep("MT-",GLU_Neurons.PD_Prog.Markers_Filtered$gene),]

GLU_Neurons_GOI <- FetchData(GLU_Neurons, vars = c("ident", GLU_Neurons.PD_Prog.Markers_Filtered_MT$gene), slot = "counts")

GLU_Neurons_marker_melt <- melt(GLU_Neurons_GOI)

GLU_Neurons_barchart_table <- group_by(GLU_Neurons_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

Darjeeling <- wes_palette("Darjeeling1")

GLU_Neurons_HCvsPDGOI_barchart_MTGenes_MTG <- ggplot(GLU_Neurons_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("GLU_Neurons") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(GLU_Neurons_HCvsPDGOI_barchart_MTGenes_MTG, filename = "Figures/GLU_Neurons_HCvsPDGOI_barchart_MTGenes_MTG.pdf", device = "pdf", width = 3.04, height = 2.015, units = "in")




```



```{R}

write.csv(top_GLU_Neurons.PD_Prog.Markers,file = "Files/Top_50_SigGenes_GLU_Neurons_Markers_HC_vs_PD_MTG.csv", quote = FALSE)

```

GABA_Neurons

```{R}
GABA_Neurons <- subset(Batch567_MTG, idents = "GABA Neurons")
Idents(GABA_Neurons) <- "case"
GABA_Neurons.PD_Prog.Markers <- FindMarkers(GABA_Neurons, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST")
GABA_Neurons.PD_Prog.Markers$gene <- rownames(GABA_Neurons.PD_Prog.Markers)
GABA_Neurons.PD_Prog.Markers_Filtered <- GABA_Neurons.PD_Prog.Markers[GABA_Neurons.PD_Prog.Markers$p_val_adj <= 0.05,]
top_GABA_Neurons.PD_Prog.Markers <- GABA_Neurons.PD_Prog.Markers_Filtered[1:50,]
#top_pos_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = avg_log2FC)
#top_neg_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = -avg_log2FC)
```

```{R}
avg.GABA_Neurons <- as.data.frame(AverageExpression(GABA_Neurons, verbose = FALSE)$RNA)
avg.GABA_Neurons$gene <- rownames(avg.GABA_Neurons)
avg.GABA_Neurons_topmarkers <- avg.GABA_Neurons[avg.GABA_Neurons$gene %in% unique(top_GABA_Neurons.PD_Prog.Markers$gene),]
avg.GABA_Neurons_topmarkers <- avg.GABA_Neurons_topmarkers[match(top_GABA_Neurons.PD_Prog.Markers$gene,avg.GABA_Neurons_topmarkers$gene),]
top_GABA_Neurons.PD_Prog.Markers$PD_mean <- avg.GABA_Neurons_topmarkers$PD
top_GABA_Neurons.PD_Prog.Markers$HC_mean <- avg.GABA_Neurons_topmarkers$HC
top_GABA_Neurons.PD_Prog.Markers$Status <- "Upregulated"
top_GABA_Neurons.PD_Prog.Markers$Status[top_GABA_Neurons.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_GABA_Neurons.PD_Prog.Markers$Status <- factor(top_GABA_Neurons.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosGABA_NeuronsGenes <- top_GABA_Neurons.PD_Prog.Markers$gene[top_GABA_Neurons.PD_Prog.Markers$avg_log2FC > 1]
NegGABA_NeuronsGenes <- top_GABA_Neurons.PD_Prog.Markers$gene[top_GABA_Neurons.PD_Prog.Markers$avg_log2FC < -1]
```


```{R}

GABA_NeuronsScatter <- ggplot(top_GABA_Neurons.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("GABA_Neurons") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
                
#GABA_NeuronsScatter <- LabelPoints(plot = GABA_NeuronsScatter, points = PosGABA_NeuronsGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
#GABA_NeuronsScatter <- LabelPoints(plot = GABA_NeuronsScatter, points = NegGABA_NeuronsGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "blue")

#GABA_NeuronsScatter

#ggsave(GABA_NeuronsScatter, filename = "Figures/GABA_NeuronsScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")

#DoHeatmap(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, size = 4, angle = 0,
#                 hjust = 0.5) + ggtitle("Astrocytes") + theme(axis.text.y = element_text(size = 8))

#VlnPlot(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, adjust = TRUE, pt.size = 0)
```
Looking at expression of MT-encoded genes

```{R}
GABA_Neurons.PD_Prog.Markers_Filtered_MT <- GABA_Neurons.PD_Prog.Markers_Filtered[grep("MT",GABA_Neurons.PD_Prog.Markers_Filtered$gene),]

GABA_Neurons_GOI <- FetchData(GABA_Neurons, vars = c("ident", GABA_Neurons.PD_Prog.Markers_Filtered_MT$gene), slot = "counts")

GABA_Neurons_marker_melt <- melt(GABA_Neurons_GOI)

GABA_Neurons_barchart_table <- group_by(GABA_Neurons_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

Darjeeling <- wes_palette("Darjeeling1")

GABA_Neurons_HCvsPDGOI_barchart_MTGenes_MTG <- ggplot(GABA_Neurons_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("GABA_Neurons") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(GABA_Neurons_HCvsPDGOI_barchart_MTGenes_MTG, filename = "Figures/GABA_Neurons_HCvsPDGOI_barchart_MTGenes_MTG.pdf", device = "pdf", width = 3.95, height = 3.03, units = "in")




```


```{R}

GABA_Neurons.PD_Prog.Markers_Filtered_GWAS <- GABA_Neurons.PD_Prog.Markers_Filtered[GABA_Neurons.PD_Prog.Markers_Filtered$gene %in% GWAS_PD_Genes,]

```


```{R}

write.csv(top_GABA_Neurons.PD_Prog.Markers,file = "Files/Top_50_SigGenes_GABA_Neurons_Markers_HC_vs_PD_MTG.csv", quote = FALSE)

```


Endothelial

```{R}
Endothelial <- subset(Batch567_MTG, idents = "Endothelial")
Idents(Endothelial) <- "case"
Endothelial.PD_Prog.Markers <- FindMarkers(Endothelial, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST")
Endothelial.PD_Prog.Markers$gene <- rownames(Endothelial.PD_Prog.Markers)
Endothelial.PD_Prog.Markers_Filtered <- Endothelial.PD_Prog.Markers[Endothelial.PD_Prog.Markers$p_val_adj <= 0.05,]
top_Endothelial.PD_Prog.Markers <- Endothelial.PD_Prog.Markers_Filtered[1:50,]
top_Endothelial.PD_Prog.Markers <- top_Endothelial.PD_Prog.Markers[!is.na(top_Endothelial.PD_Prog.Markers$p_val),]
#top_pos_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = avg_log2FC)
#top_neg_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = -avg_log2FC)
```

```{R}
avg.Endothelial <- as.data.frame(AverageExpression(Endothelial, verbose = FALSE)$RNA)
avg.Endothelial$gene <- rownames(avg.Endothelial)
avg.Endothelial_topmarkers <- avg.Endothelial[avg.Endothelial$gene %in% unique(top_Endothelial.PD_Prog.Markers$gene),]
#negfeatures_Astrocytes <- unique(top10_neg_Astrocytes.PD_Prog.Markers$gene)
#features <- c(posfeatures, negfeatures)
avg.Endothelial_topmarkers <- avg.Endothelial_topmarkers[match(top_Endothelial.PD_Prog.Markers$gene,avg.Endothelial_topmarkers$gene),]
top_Endothelial.PD_Prog.Markers$PD_mean <- avg.Endothelial_topmarkers$PD
top_Endothelial.PD_Prog.Markers$HC_mean <- avg.Endothelial_topmarkers$HC
top_Endothelial.PD_Prog.Markers$Status <- "Upregulated"
top_Endothelial.PD_Prog.Markers$Status[top_Endothelial.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_Endothelial.PD_Prog.Markers$Status <- factor(top_Endothelial.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosEndothelialGenes <- top_Endothelial.PD_Prog.Markers$gene[top_Endothelial.PD_Prog.Markers$avg_log2FC > 1]
NegEndothelialGenes <- top_Endothelial.PD_Prog.Markers$gene[top_Endothelial.PD_Prog.Markers$avg_log2FC < -1]
```


```{R}

EndothelialScatter <- ggplot(top_Endothelial.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("Endothelial") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
                
EndothelialScatter <- LabelPoints(plot = EndothelialScatter, points = PosEndothelialGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
EndothelialScatter <- LabelPoints(plot = EndothelialScatter, points = NegEndothelialGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
EndothelialScatter

ggsave(EndothelialScatter, filename = "Figures/EndothelialScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")

#DoHeatmap(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, size = 4, angle = 0,
#                 hjust = 0.5) + ggtitle("Astrocytes") + theme(axis.text.y = element_text(size = 8))

#VlnPlot(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, adjust = TRUE, pt.size = 0)
```


```{R}
Endothelial_GOI <- FetchData(Endothelial, vars = c("ident", c("PPP2R2B","CADM2","GRID2")), slot = "counts")

Endothelial_marker_melt <- melt(Endothelial_GOI)

Endothelial_barchart_table <- group_by(Endothelial_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

Darjeeling <- wes_palette("Darjeeling1")

Endothelial_HCvsPDGOI_barchart_MTG <- ggplot(Endothelial_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("Endothelial") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(Endothelial_HCvsPDGOI_barchart_MTG, filename = "Figures/Endothelial_HCvsPDGOI_barchart_MTG.pdf", device = "pdf", width = 3.03, height = 1, units = "in")




```


```{R}

Endothelial.PD_Prog.Markers_Filtered_GWAS <- Endothelial.PD_Prog.Markers_Filtered[Endothelial.PD_Prog.Markers_Filtered$gene %in% GWAS_PD_Genes,]

```


```{R}

write.csv(top_Endothelial.PD_Prog.Markers,file = "Files/Top_50_SigGenes_Endothelial_Markers_HC_vs_PD_MTG.csv", quote = FALSE)

```

