---
title: "Using MAST accounting for confounding variables"
output: html_document
---


```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(DESeq2)
library(Matrix.utils)
```


```{R}

GWAS_PD_Genes <- c("KRTCAP2","NUCKS1","GBA","SIPA1L2","ITPKB","FCGR2A","PMVK","VAMP4","KCNS3","MAP4K4","TMEM163","RAB29","STK39","KCNIP3","IP6K2","SATB1","SPTSSB","TMEM175","MCCC1","BST1","GAK","FAM47E","KPNA1","MED12L","LCORL","SCARB2","SNCA","HLA-DRB5","ELOVL7","CAMK2D","CLCN3","RIMS1","C5orf24","GPNMB","CTSB","RPS12","FYN","TRIM40","BIN3","FGF20","SH3GL2","ITGA8","GBF1","UBAP2","RNF141","FAM49B","SCAF11","DLG2","BAG3","IGSF9B","INPP5F","FBRSL1","CAB39L","GALC","CHD9","GCH1","HIP1R","FGF20","CASC16","SYT17","CD19","MIPOL1","CHRNB1","WNT3","NOD2","RETREG3","UBTF","SETD1A","CRHR1","GRN","MED13","SPPL2B","DYRK1A","DNAH17","MEX3C","ASXL3","CRLS1","LRRK2","RIT1","VPS13C")

```


```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```


```{R}
#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons", `3` = "GLU_Neurons",
                      `4` = "GLU_Neurons",`5` = "Astrocytes",
                      `6` = "GABA_Neurons", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons",`9` = "GLU_Neurons",
                      `10` = "GLU_Neurons", `11` = "Microglia",`12` = "Astrocytes",
                      `13` = "OPCs",`14` = "GABA_Neurons",
                      `15` = "GLU_Neurons", `16`="GLU_Neurons", `17`="Endothelial", `18`="Endothelial",`19`="GABA_Neurons",`20`="GLU_Neurons",`21`="GLU_Neurons",`22`="GLU_Neurons",`23`="GABA_Neurons")
```


```{R}
write.table(data.frame(as.vector(unique(Batch567_MTG@active.ident))), "Files/ClusterIdents.txt", col.names =FALSE, quote = FALSE, row.names = FALSE)

Batch567_MTG@project.name

unique(Batch567_MTG@meta.data$sample_ID)


```


Add in additional metadata 

```{R}
ID_Order <- unique(Batch567_MTG@meta.data$sample_ID)

ID_Order

sex <- c("M","M","M","M","M","F","M","F","M","M","F")

#Batch567_MTG@meta.data <- Batch567_MTG@meta.data %>% group_by(sample_ID) %>% mutate(sex = rep(sex[match(unique(sample_ID), ID_Order)], length(sample_ID)))

#Batch567_MTG@meta.data$DetRate <- as.vector(scale(colSums(Batch567_MTG@assays$RNA@counts)))

Batch567_MTG@meta.data$sex <- Batch567_MTG@meta.data %>% group_by(sample_ID) %>% mutate(Sex = rep(sex[match(unique(sample_ID), ID_Order)], length(sample_ID))) %>% .$Sex
```



NOTE: MAST uses the conservative bonferroni correction to generate the adjusted p-values (p-value divided by the number of tests)

Astrocytes

```{R}

Astrocytes <- subset(Batch567_MTG, idents = "Astrocytes")

#mkfilter <- function(cmatrix) {
#        sum(cmatrix[1,] > 0)/ncol(cmatrix)*100
#} 

mkfilter <- function(cmatrixline) {
        sum(cmatrixline > 0)/length(cmatrixline)*100
} 

Astrocytefilter <- Astrocytes[apply(Astrocytes@assays$RNA@counts,1,mkfilter) >= 20,]

#Astrocytes_Filter <- Astrocytes[rowSums(Astrocytes@assays$RNA@counts) >= 10, ]


Astrocytefilter@meta.data$DetRate <- as.vector(scale(colSums(Astrocytefilter@assays$RNA@counts > 0)))


Astrocytefilter@meta.data$
```


```{R}
Idents(Astrocytefilter) <- "case"
Astrocytes.PD_Prog.Markers <- FindMarkers(Astrocytefilter, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate","batch"))
Astrocytes.PD_Prog.Markers$gene <- rownames(Astrocytes.PD_Prog.Markers)
Astrocytes.PD_Prog.Markers_Filtered <- Astrocytes.PD_Prog.Markers[Astrocytes.PD_Prog.Markers$p_val_adj <= 0.05,]
top_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered[1:50,]
#top_pos_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = avg_log2FC)
#top_neg_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = -avg_log2FC)
```

```{R}
avg.Astrocytes <- as.data.frame(AverageExpression(Astrocytefilter, verbose = FALSE)$RNA)
avg.Astrocytes$gene <- rownames(avg.Astrocytes)
avg.Astrocytes_topmarkers <- avg.Astrocytes[avg.Astrocytes$gene %in% unique(top_Astrocytes.PD_Prog.Markers$gene),]
#negfeatures_Astrocytes <- unique(top10_neg_Astrocytes.PD_Prog.Markers$gene)
#features <- c(posfeatures, negfeatures)
avg.Astrocytes_topmarkers <- avg.Astrocytes_topmarkers[match(top_Astrocytes.PD_Prog.Markers$gene,avg.Astrocytes_topmarkers$gene),]
top_Astrocytes.PD_Prog.Markers$PD_mean <- avg.Astrocytes_topmarkers$PD
top_Astrocytes.PD_Prog.Markers$HC_mean <- avg.Astrocytes_topmarkers$HC
top_Astrocytes.PD_Prog.Markers$Status <- "Upregulated"
top_Astrocytes.PD_Prog.Markers$Status[top_Astrocytes.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_Astrocytes.PD_Prog.Markers$Status <- factor(top_Astrocytes.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosAstroGenes <- top_Astrocytes.PD_Prog.Markers$gene[top_Astrocytes.PD_Prog.Markers$avg_log2FC > 1]
NegAstroGenes <- top_Astrocytes.PD_Prog.Markers$gene[top_Astrocytes.PD_Prog.Markers$avg_log2FC < -1]
```


```{R}

AstroScatter <- ggplot(top_Astrocytes.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("Astrocytes") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
                
AstroScatter <- LabelPoints(plot = AstroScatter, points = PosAstroGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
AstroScatter <- LabelPoints(plot = AstroScatter, points = NegAstroGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
AstroScatter

#ggsave(AstroScatter, filename = "Figures/AstroScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")

#DoHeatmap(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, size = 4, angle = 0,
#                 hjust = 0.5) + ggtitle("Astrocytes") + theme(axis.text.y = element_text(size = 8))

#VlnPlot(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, adjust = TRUE, pt.size = 0)
```

Violin Plot

```{r}
VlnPlot(Astrocytes, features = c("GNA14"),pt.size = 0, split.by = "case") +
theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 8),
              legend.text = element_text(size = 8),
              title = element_text(size = 8),
              legend.key.size = unit(0.4,"cm"))
```


```{R}
Astro_GNA14 <- FetchData(Astrocytes, vars = c("ident", c("GNA14","SLC14A1")), slot = "counts")
#Astro_GNA14 <- FetchData(Astrocytes, vars = c("ident", c("GNA14","SLC14A1"), slot = "counts"))

ggplot(Astro_GNA14, aes(x=ident, y=GNA14)) +
                      geom_violin() +
                      theme(axis.text = element_text(size=8),
                      axis.title = element_text(size = 8),
                      legend.text = element_text(size = 8),
                      legend.position = "none",
                      title = element_text(size = 8))



#ggsave(AstroVlnPlot_GNA14_MTG, filename = "Figures/AstroVlnPlot_GNA14_MTG.pdf", device = "pdf", width = 1, height = 1, units = "in")
```

```{R}
library(wesanderson)

Astro_marker_melt <- melt(Astro_GNA14)

Astro_barchart_table <- group_by(Astro_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

Darjeeling <- wes_palette("Darjeeling1")

ggplot(Astro_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.5, colors="black") +
        theme(axis.title = element_blank(), axis.text.x = element_blank(),
              axis.ticks = element_blank(),
              strip.text.y = element_text(size = 12, angle = 180, face = "bold")) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

Astrocyte_HCvsPDGOI_barchart_MTG_large <- ggplot(Astro_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.5, colors="black") +
        theme(axis.title = element_blank(), axis.text.x = element_blank(),
              axis.ticks = element_blank(),
              strip.text.y = element_text(size = 12, angle = 180, face = "bold")) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(Astrocyte_HCvsPDGOI_barchart_MTG_large, filename = "Figures/Astrocyte_HCvsPDGOI_barchart_MTG_large.pdf", device = "pdf", width = 4, height = 4, units = "in")

Astrocyte_HCvsPDGOI_barchart_MTG <- ggplot(Astro_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("Astrocytes") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(Astrocyte_HCvsPDGOI_barchart_MTG, filename = "Figures/Astrocyte_HCvsPDGOI_barchart_MTG.pdf", device = "pdf", width = 2.1, height = 1, units = "in")

```


Testing for differential expression of GWAS PD genes

```{R}

Astrocytes.PD_Prog.Markers_Filtered_GWAS <- Astrocytes.PD_Prog.Markers_Filtered[Astrocytes.PD_Prog.Markers_Filtered$gene %in% GWAS_PD_Genes,]

```


data_counts <- FetchData(Batch567_MTG, vars = c("ident",rownames(Batch567_MTG@assays$RNA@counts)), slot = "counts")

scaledrowSums <- 1e6/rowSums(data_counts[2:length(colnames(data_counts))])

AstrocyteCellIds <- rownames(Astro_GNA14) %in% scaledrowSums

match(AstrocyteCellIds,names(scaledrowSums))

AstrocyteScaledrowSums <- scaledrowSums[match(AstrocyteCellIds,names(scaledrowSums))]

Astro_GNA14$GNA14 <- Astro_GNA14$GNA14*AstrocyteScaledrowSums

Astro_GNA14$SLC14A1 <- Astro_GNA14$SLC14A1*AstrocyteScaledrowSums

Astro_marker_melt <- melt(Astro_GNA14)

Astro_barchart_table <- group_by(Astro_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))






ggplot(Astro_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.5, colors="black") +
        theme(axis.title = element_blank(), axis.text.x = element_text(size = 12, angle = 45, face = "bold", vjust = 0.5),
              strip.text.y = element_text(size = 12, angle = 180, face = "bold"))

```{R}

Astro_DEProgGenes <- FetchData(Astrocytes, vars = c("ident", c("RIMS2","SLC14A1")), slot = "counts")

Astro_DEProgGenes_melt <- melt(Astro_DEProgGenes)

Astro_DEProgGenes_barchart_table <- group_by(Astro_DEProgGenes_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

ggplot(Astro_DEProgGenes_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("Astrocytes") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(Astrocyte_HCvsPDGOI_barchart_MTG, filename = "Figures/Astrocyte_HCvsPDGOI_barchart_MTG.pdf", device = "pdf", width = 2.1, height = 1, units = "in")



```



```{R}

write.csv(top_Astrocytes.PD_Prog.Markers,file = "Files/Top_50_SigGenes_Astrocyte_Markers_HC_vs_PD_MTG.csv", quote = FALSE)

```

Microglia

```{R}
Microglia <- subset(Batch567_MTG, idents = "Microglia")
Idents(Microglia) <- "case"
Microglia.PD_Prog.Markers <- FindMarkers(Microglia, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST")
Microglia.PD_Prog.Markers$gene <- rownames(Microglia.PD_Prog.Markers)
Microglia.PD_Prog.Markers_Filtered <- Microglia.PD_Prog.Markers[Microglia.PD_Prog.Markers$p_val_adj <= 0.05,]
top_Microglia.PD_Prog.Markers <- Microglia.PD_Prog.Markers_Filtered[1:50,]
#top_pos_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = avg_log2FC)
#top_neg_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = -avg_log2FC)
```

```{R}
avg.Microglia <- as.data.frame(AverageExpression(Microglia, verbose = FALSE)$RNA)
avg.Microglia$gene <- rownames(avg.Microglia)
avg.Microglia_topmarkers <- avg.Microglia[avg.Microglia$gene %in% unique(top_Microglia.PD_Prog.Markers$gene),]
#negfeatures_Astrocytes <- unique(top10_neg_Astrocytes.PD_Prog.Markers$gene)
#features <- c(posfeatures, negfeatures)
avg.Microglia_topmarkers <- avg.Microglia_topmarkers[match(top_Microglia.PD_Prog.Markers$gene,avg.Microglia_topmarkers$gene),]
top_Microglia.PD_Prog.Markers$PD_mean <- avg.Microglia_topmarkers$PD
top_Microglia.PD_Prog.Markers$HC_mean <- avg.Microglia_topmarkers$HC
top_Microglia.PD_Prog.Markers$Status <- "Upregulated"
top_Microglia.PD_Prog.Markers$Status[top_Microglia.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_Microglia.PD_Prog.Markers$Status <- factor(top_Microglia.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosMicrogliaGenes <- top_Microglia.PD_Prog.Markers$gene[top_Microglia.PD_Prog.Markers$avg_log2FC > 1]
NegMicrogliaGenes <- top_Microglia.PD_Prog.Markers$gene[top_Microglia.PD_Prog.Markers$avg_log2FC < -1]
```


```{R}

MicrogliaScatter <- ggplot(top_Microglia.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("Microglia") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
                
MicrogliaScatter <- LabelPoints(plot = MicrogliaScatter, points = PosMicrogliaGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
MicrogliaScatter <- LabelPoints(plot = MicrogliaScatter, points = NegMicrogliaGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
MicrogliaScatter

ggsave(MicrogliaScatter, filename = "Figures/MicrogliaScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")

#DoHeatmap(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, size = 4, angle = 0,
#                 hjust = 0.5) + ggtitle("Astrocytes") + theme(axis.text.y = element_text(size = 8))

#VlnPlot(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, adjust = TRUE, pt.size = 0)
```

```{R}
Microglia_GOI <- FetchData(Microglia, vars = c("ident", c("P2RY12","GRID2","SPP1")), slot = "counts")

Microglia_marker_melt <- melt(Microglia_GOI)

Microglia_barchart_table <- group_by(Microglia_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

Darjeeling <- wes_palette("Darjeeling1")

Microglia_HCvsPDGOI_barchart_MTG <- ggplot(Microglia_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("Microglia") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(Microglia_HCvsPDGOI_barchart_MTG, filename = "Figures/Microglia_HCvsPDGOI_barchart_MTG.pdf", device = "pdf", width = 3, height = 1, units = "in")




```


```{R}

Microglia.PD_Prog.Markers_Filtered_GWAS <- Microglia.PD_Prog.Markers_Filtered[Microglia.PD_Prog.Markers_Filtered$gene %in% GWAS_PD_Genes,]

```


```{R}

write.csv(top_Microglia.PD_Prog.Markers,file = "Files/Top_50_SigGenes_Microglia_Markers_HC_vs_PD_MTG.csv", quote = FALSE)

```


Oligodendrocytes

```{R}
Oligodendrocytes <- subset(Batch567_MTG, idents = "Oligodendrocytes")
Idents(Oligodendrocytes) <- "case"
Oligodendrocytes.PD_Prog.Markers <- FindMarkers(Oligodendrocytes, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST")
Oligodendrocytes.PD_Prog.Markers$gene <- rownames(Oligodendrocytes.PD_Prog.Markers)
Oligodendrocytes.PD_Prog.Markers_Filtered <- Oligodendrocytes.PD_Prog.Markers[Oligodendrocytes.PD_Prog.Markers$p_val_adj <= 0.05,]
top_Oligodendrocytes.PD_Prog.Markers <- Oligodendrocytes.PD_Prog.Markers_Filtered[1:50,]
#top_pos_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = avg_log2FC)
#top_neg_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = -avg_log2FC)
```

```{R}
avg.Oligodendrocytes <- as.data.frame(AverageExpression(Oligodendrocytes, verbose = FALSE)$RNA)
avg.Oligodendrocytes$gene <- rownames(avg.Oligodendrocytes)
avg.Oligodendrocytes_topmarkers <- avg.Oligodendrocytes[avg.Oligodendrocytes$gene %in% unique(top_Oligodendrocytes.PD_Prog.Markers$gene),]
#negfeatures_Astrocytes <- unique(top10_neg_Astrocytes.PD_Prog.Markers$gene)
#features <- c(posfeatures, negfeatures)
avg.Oligodendrocytes_topmarkers <- avg.Oligodendrocytes_topmarkers[match(top_Oligodendrocytes.PD_Prog.Markers$gene,avg.Oligodendrocytes_topmarkers$gene),]
top_Oligodendrocytes.PD_Prog.Markers$PD_mean <- avg.Oligodendrocytes_topmarkers$PD
top_Oligodendrocytes.PD_Prog.Markers$HC_mean <- avg.Oligodendrocytes_topmarkers$HC
top_Oligodendrocytes.PD_Prog.Markers$Status <- "Upregulated"
top_Oligodendrocytes.PD_Prog.Markers$Status[top_Oligodendrocytes.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_Oligodendrocytes.PD_Prog.Markers$Status <- factor(top_Oligodendrocytes.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosOligodendrocytesGenes <- top_Oligodendrocytes.PD_Prog.Markers$gene[top_Oligodendrocytes.PD_Prog.Markers$avg_log2FC > 1]
NegOligodendrocytesGenes <- top_Oligodendrocytes.PD_Prog.Markers$gene[top_Oligodendrocytes.PD_Prog.Markers$avg_log2FC < -1]
```


```{R}

OligodendrocytesScatter <- ggplot(top_Oligodendrocytes.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("Oligodendrocytes") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
                
OligodendrocytesScatter <- LabelPoints(plot = OligodendrocytesScatter, points = PosOligodendrocytesGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
OligodendrocytesScatter <- LabelPoints(plot = OligodendrocytesScatter, points = NegOligodendrocytesGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
OligodendrocytesScatter

ggsave(OligodendrocytesScatter, filename = "Figures/OligodendrocytesScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")

#DoHeatmap(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, size = 4, angle = 0,
#                 hjust = 0.5) + ggtitle("Astrocytes") + theme(axis.text.y = element_text(size = 8))

#VlnPlot(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, adjust = TRUE, pt.size = 0)
```


```{R}
Oligodendrocytes_GOI <- FetchData(Oligodendrocytes, vars = c("ident", c("PPP2R2B","CADM2","GRID2")), slot = "counts")

Oligodendrocytes_marker_melt <- melt(Oligodendrocytes_GOI)

Oligodendrocytes_barchart_table <- group_by(Oligodendrocytes_marker_melt,ident,variable) %>% summarise(mean = mean(value), SE = se(value))

Darjeeling <- wes_palette("Darjeeling1")

Oligodendrocytes_HCvsPDGOI_barchart_MTG <- ggplot(Oligodendrocytes_barchart_table, aes(x = ident, y = mean, fill = ident)) + 
        geom_bar(stat = "identity", alpha = 1) + 
        facet_wrap(~variable) +
        geom_errorbar(aes(x = ident, ymin = mean-SE, ymax = mean+SE), width = 0.4, alpha = 0.9, size = 0.1, colors="black") +
        ylab("Oligodendrocytes") +
        theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
              axis.title.y = element_text(size = 5),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_line(size = 0.2),
              strip.text = element_text(size = 3.5,face = "bold"),
              axis.text = element_text(size=4),
              legend.position = "none",) +
        scale_fill_manual(name="Case",values = Darjeeling[c(2,3)])

ggsave(Oligodendrocytes_HCvsPDGOI_barchart_MTG, filename = "Figures/Oligodendrocytes_HCvsPDGOI_barchart_MTG.pdf", device = "pdf", width = 3.03, height = 1, units = "in")




```


```{R}

Oligodendrocytes.PD_Prog.Markers_Filtered_GWAS <- Oligodendrocytes.PD_Prog.Markers_Filtered[Oligodendrocytes.PD_Prog.Markers_Filtered$gene %in% GWAS_PD_Genes,]

```


```{R}

write.csv(top_Oligodendrocytes.PD_Prog.Markers,file = "Files/Top_50_SigGenes_Oligodendrocytes_Markers_HC_vs_PD_MTG.csv", quote = FALSE)

```


Creating function to run through DE analysis for each cluster identity

```{R}
mkfilter <- function(cmatrixline) {
        sum(cmatrixline > 0)/length(cmatrixline)*100
} 

MASTDEcasePerCluster <- function(SeuObj,Cluster_Ident){
        Ident_Object <- subset(SeuObj, idents = Cluster_Ident)
        Ident_Object <- Ident_Object[apply(Ident_Object@assays$RNA@counts,1,mkfilter) >= 20,]
        Ident_Object@meta.data$DetRate <- as.vector(scale(colSums(Ident_Object@assays$RNA@counts)))
        Idents(Ident_Object) <- "case"
        
        #PD vs HC
        Ident.PDvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.PDvsHC_Prog.Markers$gene <- rownames(Ident.PDvsHC_Prog.Markers)
        Ident.PDvsHC_Prog.Markers <-
                Ident.PDvsHC_Prog.Markers[Ident.PDvsHC_Prog.Markers$p_val_adj <= 0.05,]
        
        Top_Ident.PDvsHC_Prog.Markers <- Ident.PDvsHC_Prog.Markers[1:50,]
        
        avg.Ident_Object_PDvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsHC$gene <- rownames(avg.Ident_Object_PDvsHC)
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Top_Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_topmarkers_PDvsHC[match(Top_Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_topmarkers_PDvsHC$gene),]
        Top_Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_topmarkers_PDvsHC$PD
        Top_Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_topmarkers_PDvsHC$HC
        Top_Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status <- factor(Top_Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Top_Ident.PDvsHC_Prog.Markers,file = paste("Files/DE_Genes/Top_50_SigGenes_",Cluster_Ident,"_Markers_HC_vs_PD.csv",sep = ""), quote = FALSE)
        
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_progmarkers_PDvsHC[match(Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsHC$gene),]
        Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsHC$PD
        Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_PDvsHC$HC
        Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsHC_Prog.Markers$Status[Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsHC_Prog.Markers$Status <- factor(Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_PD.csv",sep = ""), quote = FALSE)
        
        
        
        
        PosIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < -1]
     IdentScatter_PDvsHC <- ggplot(Top_Ident.PDvsHC_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
         
        if (length(PosIdentGenes_PDvsHC) > 0) {       
                IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = PosIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        
        if (length(NegIdentGenes_PDvsHC) > 0) {
                IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = NegIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
     
        ggsave(IdentScatter_PDvsHC, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_HC_and_PD.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")
        
        
        # ILB vs HC
        
        
        Ident.ILBvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "ILB", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.ILBvsHC_Prog.Markers$gene <- rownames(Ident.ILBvsHC_Prog.Markers)
        Ident.ILBvsHC_Prog.Markers <- Ident.ILBvsHC_Prog.Markers[Ident.ILBvsHC_Prog.Markers$p_val_adj <= 0.05,]
        
        Top_Ident.ILBvsHC_Prog.Markers <- Ident.ILBvsHC_Prog.Markers[1:50,]
        
        avg.Ident_Object_ILBvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_ILBvsHC$gene <- rownames(avg.Ident_Object_ILBvsHC)
        
        avg.Ident_Object_topmarkers_ILBvsHC <- avg.Ident_Object_ILBvsHC[avg.Ident_Object_ILBvsHC$gene %in% unique(Top_Ident.ILBvsHC_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_ILBvsHC <- avg.Ident_Object_topmarkers_ILBvsHC[match(Top_Ident.ILBvsHC_Prog.Markers$gene,avg.Ident_Object_topmarkers_ILBvsHC$gene),]
        Top_Ident.ILBvsHC_Prog.Markers$ILB_mean <- avg.Ident_Object_topmarkers_ILBvsHC$ILB
        Top_Ident.ILBvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_topmarkers_ILBvsHC$HC
        Top_Ident.ILBvsHC_Prog.Markers$Status <- "Upregulated"
        Top_Ident.ILBvsHC_Prog.Markers$Status[Top_Ident.ILBvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.ILBvsHC_Prog.Markers$Status <- factor(Top_Ident.ILBvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Top_Ident.ILBvsHC_Prog.Markers,file = paste("Files/DE_Genes/Top_50_SigGenes_",Cluster_Ident,"_Markers_HC_vs_ILB.csv",sep = ""), quote = FALSE)
        
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_ILBvsHC[avg.Ident_Object_ILBvsHC$gene %in% unique(Ident.ILBvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_progmarkers_ILBvsHC[match(Ident.ILBvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_ILBvsHC$gene),]
        Ident.ILBvsHC_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_ILBvsHC$ILB
        Ident.ILBvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_ILBvsHC$HC
        Ident.ILBvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.ILBvsHC_Prog.Markers$Status[Ident.ILBvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.ILBvsHC_Prog.Markers$Status <- factor(Ident.ILBvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.ILBvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_ILB.csv",sep = ""), quote = FALSE)
        
        PosIdentGenes_ILBvsHC <- Top_Ident.ILBvsHC_Prog.Markers$gene[Top_Ident.ILBvsHC_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_ILBvsHC <- Top_Ident.ILBvsHC_Prog.Markers$gene[Top_Ident.ILBvsHC_Prog.Markers$avg_log2FC < -1]
     IdentScatter_ILBvsHC <- ggplot(Top_Ident.ILBvsHC_Prog.Markers, aes(HC_mean, ILB_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("ILB Mean") +
                scale_colour_manual(values = (c("red","blue")))
     
        
        if (length(PosIdentGenes_ILBvsHC) > 0) {       
                IdentScatter_ILBvsHC <- LabelPoints(plot = IdentScatter_ILBvsHC, points = PosIdentGenes_ILBvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        if (length(NegIdentGenes_ILBvsHC)) {
                IdentScatter_ILBvsHC <- LabelPoints(plot = IdentScatter_ILBvsHC, points = NegIdentGenes_ILBvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
        ggsave(IdentScatter_ILBvsHC, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_HC_and_ILB.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")
        
        
        # PD vs ILB
        
        
                Ident.PDvsILB_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "ILB", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.PDvsILB_Prog.Markers$gene <- rownames(Ident.PDvsILB_Prog.Markers)
        Ident.PDvsILB_Prog.Markers <-
                Ident.PDvsILB_Prog.Markers[Ident.PDvsILB_Prog.Markers$p_val_adj <= 0.05,]
        
        Top_Ident.PDvsILB_Prog.Markers <- Ident.PDvsILB_Prog.Markers[1:50,]
        
        avg.Ident_Object_PDvsILB <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsILB$gene <- rownames(avg.Ident_Object_PDvsILB)
        avg.Ident_Object_topmarkers_PDvsILB <- avg.Ident_Object_PDvsILB[avg.Ident_Object_PDvsILB$gene %in% unique(Top_Ident.PDvsILB_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_PDvsILB <- avg.Ident_Object_topmarkers_PDvsILB[match(Top_Ident.PDvsILB_Prog.Markers$gene,avg.Ident_Object_topmarkers_PDvsILB$gene),]
        Top_Ident.PDvsILB_Prog.Markers$PD_mean <- avg.Ident_Object_topmarkers_PDvsILB$PD
        Top_Ident.PDvsILB_Prog.Markers$ILB_mean <- avg.Ident_Object_topmarkers_PDvsILB$ILB
        Top_Ident.PDvsILB_Prog.Markers$Status <- "Upregulated"
        Top_Ident.PDvsILB_Prog.Markers$Status[Top_Ident.PDvsILB_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.PDvsILB_Prog.Markers$Status <- factor(Top_Ident.PDvsILB_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Top_Ident.PDvsILB_Prog.Markers,file = paste("Files/DE_Genes/Top_50_SigGenes_",Cluster_Ident,"_Markers_ILB_vs_PD.csv",sep = ""), quote = FALSE)
        
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_PDvsILB[avg.Ident_Object_PDvsILB$gene %in% unique(Ident.PDvsILB_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_progmarkers_PDvsILB[match(Ident.PDvsILB_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsILB$gene),]
        Ident.PDvsILB_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsILB$PD
        Ident.PDvsILB_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_PDvsILB$ILB
        Ident.PDvsILB_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsILB_Prog.Markers$Status[Ident.PDvsILB_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsILB_Prog.Markers$Status <- factor(Ident.PDvsILB_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsILB_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_ILB_vs_PD.csv",sep = ""), quote = FALSE)
        
        
        PosIdentGenes_PDvsILB <- Top_Ident.PDvsILB_Prog.Markers$gene[Top_Ident.PDvsILB_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_PDvsILB <- Top_Ident.PDvsILB_Prog.Markers$gene[Top_Ident.PDvsILB_Prog.Markers$avg_log2FC < -1]
     IdentScatter_PDvsILB <- ggplot(Top_Ident.PDvsILB_Prog.Markers, aes(ILB_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("ILB Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
        
        
        if (length(PosIdentGenes_PDvsILB) > 0) {       
                IdentScatter_PDvsILB <- LabelPoints(plot = IdentScatter_PDvsILB, points = PosIdentGenes_PDvsILB, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        if (length(NegIdentGenes_PDvsILB) > 0) {
                IdentScatter_PDvsILB <- LabelPoints(plot = IdentScatter_PDvsILB, points = NegIdentGenes_PDvsILB, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
        ggsave(IdentScatter_PDvsILB, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_ILB_and_PD.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")
        
        
}
```

```{R}
Clusters <- unique(as.vector(Batch567_MTG@active.ident))

MASTDEcasePerCluster(Batch567_MTG,"Astrocytes")

lapply(X = Clusters, FUN = MASTDEcasePerCluster, SeuObj = Batch567_MTG)

```

SeuObj,Cluster_Ident

```{R}
avg.Astrocytes <- as.data.frame(AverageExpression(Astrocytefilter, verbose = FALSE)$RNA)
avg.Astrocytes$gene <- rownames(avg.Astrocytes)
avg.Astrocytes_topmarkers <- avg.Astrocytes[avg.Astrocytes$gene %in% unique(top_Astrocytes.PD_Prog.Markers$gene),]
#negfeatures_Astrocytes <- unique(top10_neg_Astrocytes.PD_Prog.Markers$gene)
#features <- c(posfeatures, negfeatures)
avg.Astrocytes_topmarkers <- avg.Astrocytes_topmarkers[match(top_Astrocytes.PD_Prog.Markers$gene,avg.Astrocytes_topmarkers$gene),]
top_Astrocytes.PD_Prog.Markers$PD_mean <- avg.Astrocytes_topmarkers$PD
top_Astrocytes.PD_Prog.Markers$HC_mean <- avg.Astrocytes_topmarkers$HC
top_Astrocytes.PD_Prog.Markers$Status <- "Upregulated"
top_Astrocytes.PD_Prog.Markers$Status[top_Astrocytes.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_Astrocytes.PD_Prog.Markers$Status <- factor(top_Astrocytes.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosAstroGenes <- top_Astrocytes.PD_Prog.Markers$gene[top_Astrocytes.PD_Prog.Markers$avg_log2FC > 1]
NegAstroGenes <- top_Astrocytes.PD_Prog.Markers$gene[top_Astrocytes.PD_Prog.Markers$avg_log2FC < -1]
```


```{R}

AstroScatter <- ggplot(top_Astrocytes.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("Astrocytes") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
                
AstroScatter <- LabelPoints(plot = AstroScatter, points = PosAstroGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
AstroScatter <- LabelPoints(plot = AstroScatter, points = NegAstroGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
AstroScatter

#ggsave(AstroScatter, filename = "Figures/AstroScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")
```

GABA Neurons


```{R}
GABA_Neurons <- subset(Batch567_MTG, idents = "GABA_Neurons")
Idents(GABA_Neurons) <- "case"
GABA_Neurons.PD_Prog.Markers <- FindMarkers(GABA_Neurons, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST")
GABA_Neurons.PD_Prog.Markers$gene <- rownames(GABA_Neurons.PD_Prog.Markers)
GABA_Neurons.PD_Prog.Markers_Filtered <- GABA_Neurons.PD_Prog.Markers[GABA_Neurons.PD_Prog.Markers$p_val_adj <= 0.05,]
top_GABA_Neurons.PD_Prog.Markers <- GABA_Neurons.PD_Prog.Markers_Filtered[1:50,]
#top_pos_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = avg_log2FC)
#top_neg_Astrocytes.PD_Prog.Markers <- Astrocytes.PD_Prog.Markers_Filtered %>% top_n(n = 50, wt = -avg_log2FC)
```

```{R}
avg.GABA_Neurons <- as.data.frame(AverageExpression(GABA_Neurons, verbose = FALSE)$RNA)
avg.GABA_Neurons$gene <- rownames(avg.GABA_Neurons)
avg.GABA_Neurons_topmarkers <- avg.GABA_Neurons[avg.GABA_Neurons$gene %in% unique(top_GABA_Neurons.PD_Prog.Markers$gene),]
#negfeatures_Astrocytes <- unique(top10_neg_Astrocytes.PD_Prog.Markers$gene)
#features <- c(posfeatures, negfeatures)
avg.GABA_Neurons_topmarkers <- avg.GABA_Neurons_topmarkers[match(top_GABA_Neurons.PD_Prog.Markers$gene,avg.GABA_Neurons_topmarkers$gene),]
top_GABA_Neurons.PD_Prog.Markers$PD_mean <- avg.GABA_Neurons_topmarkers$PD
top_GABA_Neurons.PD_Prog.Markers$HC_mean <- avg.GABA_Neurons_topmarkers$HC
top_GABA_Neurons.PD_Prog.Markers$Status <- "Upregulated"
top_GABA_Neurons.PD_Prog.Markers$Status[top_GABA_Neurons.PD_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
top_GABA_Neurons.PD_Prog.Markers$Status <- factor(top_GABA_Neurons.PD_Prog.Markers$Status, c("Upregulated","Downregulated"))
PosGABA_NeuronsGenes <- top_GABA_Neurons.PD_Prog.Markers$gene[top_GABA_Neurons.PD_Prog.Markers$avg_log2FC > 1]
NegGABA_NeuronsGenes <- top_GABA_Neurons.PD_Prog.Markers$gene[top_GABA_Neurons.PD_Prog.Markers$avg_log2FC < -1]
```

if (length(PosGABA_NeuronsGenes) > 0) {                
        GABA_NeuronsScatter <- LabelPoints(plot = GABA_NeuronsScatter, points = PosGABA_NeuronsGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
}

```{R}

GABA_NeuronsScatter <- ggplot(top_GABA_Neurons.PD_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle("GABA_Neurons") +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
                
               
if (length(PosGABA_NeuronsGenes) > 0) {                
        GABA_NeuronsScatter <- LabelPoints(plot = GABA_NeuronsScatter, points = PosGABA_NeuronsGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
}


if (length(NegGABA_NeuronsGenes) > 0) {
        GABA_NeuronsScatter <- LabelPoints(plot = GABA_NeuronsScatter, points = NegGABA_NeuronsGenes, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
}
GABA_NeuronsScatter

ggsave(GABA_NeuronsScatter, filename = "Figures/GABA_NeuronsScatter_Top_50_SigGenes_HC_and_PD_MTG.pdf", device = "pdf", width = 4, height = 4, units = "in")

#DoHeatmap(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, size = 4, angle = 0,
#                 hjust = 0.5) + ggtitle("Astrocytes") + theme(axis.text.y = element_text(size = 8))

#VlnPlot(Astrocytes, features = top_pos_Astrocytes.PD_Prog.Markers$gene, adjust = TRUE, pt.size = 0)
```

```{R}
Ident_Object <- subset(Batch567_MTG, idents = "GABA_Neurons")
Ident_Object <- Ident_Object[apply(Ident_Object@assays$RNA@counts,1,mkfilter) >= 20,]
Ident_Object@meta.data$DetRate <- as.vector(scale(colSums(Ident_Object@assays$RNA@counts)))
Idents(Ident_Object) <- "case"
        
        #PD vs HC
Ident.PDvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "ILB", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
Ident.PDvsHC_Prog.Markers$gene <- rownames(Ident.PDvsHC_Prog.Markers)
Ident.PDvsHC_Prog.Markers <-
Ident.PDvsHC_Prog.Markers[Ident.PDvsHC_Prog.Markers$p_val_adj <= 0.05,]
avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Ident.PDvsHC_Prog.Markers$gene),]
avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_progmarkers_PDvsHC[match(Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsHC$gene),]
Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsHC$PD
Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_PDvsHC$HC
Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
Ident.PDvsHC_Prog.Markers$Status[Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
Ident.PDvsHC_Prog.Markers$Status <- factor(Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
write.csv(Ident.PDvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_PD.csv",sep = ""), quote = FALSE)

if (length(Ident.PDvsHC_Prog.Markers$gene) > 50) {    
        Top_Ident.PDvsHC_Prog.Markers <- Ident.PDvsHC_Prog.Markers[1:50,]
        avg.Ident_Object_PDvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsHC$gene <- rownames(avg.Ident_Object_PDvsHC)
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Top_Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_topmarkers_PDvsHC[match(Top_Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_topmarkers_PDvsHC$gene),]
        Top_Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_topmarkers_PDvsHC$PD
        Top_Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_topmarkers_PDvsHC$HC
        Top_Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status <- factor(Top_Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        PosIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < -1]
        IdentScatter_PDvsHC <- ggplot(Top_Ident.PDvsHC_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
         
        if (length(PosIdentGenes_PDvsHC) > 0) {       
        IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = PosIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        
        if (length(NegIdentGenes_PDvsHC) > 0) {
        IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = NegIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
     
        ggsave(IdentScatter_PDvsHC, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_HC_and_PD.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")
        
} else if (length(Ident.PDvsHC_Prog.Markers$gene) < 50 & length(Ident.PDvsHC_Prog.Markers$gene) > 0) {
        Top_Ident.PDvsHC_Prog.Markers <- Ident.PDvsHC_Prog.Markers[1:length(Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_PDvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsHC$gene <- rownames(avg.Ident_Object_PDvsHC)
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Top_Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_topmarkers_PDvsHC[match(Top_Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_topmarkers_PDvsHC$gene),]
        Top_Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_topmarkers_PDvsHC$PD
        Top_Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_topmarkers_PDvsHC$HC
        Top_Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status <- factor(Top_Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        PosIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < -1]
        IdentScatter_PDvsHC <- ggplot(Top_Ident.PDvsHC_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
         
        if (length(PosIdentGenes_PDvsHC) > 0) {       
        IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = PosIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        
        if (length(NegIdentGenes_PDvsHC) > 0) {
        IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = NegIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
     
        ggsave(IdentScatter_PDvsHC, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_HC_and_PD.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")
        
} else if (length(Ident.PDvsHC_Prog.Markers$gene == 0)) {
        break
}

        



        

        
        
        
        












```


```{R}
mkfilter <- function(cmatrixline) {
        sum(cmatrixline > 0)/length(cmatrixline)*100
} 

MASTDEcasePerCluster <- function(SeuObj,Cluster_Ident){
        Ident_Object <- subset(SeuObj, idents = Cluster_Ident)
        Ident_Object <- Ident_Object[apply(Ident_Object@assays$RNA@counts,1,mkfilter) >= 20,]
        Ident_Object@meta.data$DetRate <- as.vector(scale(colSums(Ident_Object@assays$RNA@counts)))
        Idents(Ident_Object) <- "case"
        
        #PD vs HC
        Ident.PDvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.PDvsHC_Prog.Markers$gene <- rownames(Ident.PDvsHC_Prog.Markers)
        Ident.PDvsHC_Prog.Markers <-
                Ident.PDvsHC_Prog.Markers[Ident.PDvsHC_Prog.Markers$p_val_adj <= 0.05,]
        
        Top_Ident.PDvsHC_Prog.Markers <- Ident.PDvsHC_Prog.Markers[1:50,]
        
        avg.Ident_Object_PDvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsHC$gene <- rownames(avg.Ident_Object_PDvsHC)
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Top_Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_PDvsHC <- avg.Ident_Object_topmarkers_PDvsHC[match(Top_Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_topmarkers_PDvsHC$gene),]
        Top_Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_topmarkers_PDvsHC$PD
        Top_Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_topmarkers_PDvsHC$HC
        Top_Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.PDvsHC_Prog.Markers$Status <- factor(Top_Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Top_Ident.PDvsHC_Prog.Markers,file = paste("Files/DE_Genes/Top_50_SigGenes_",Cluster_Ident,"_Markers_HC_vs_PD.csv",sep = ""), quote = FALSE)
        
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_progmarkers_PDvsHC[match(Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsHC$gene),]
        Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsHC$PD
        Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_PDvsHC$HC
        Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsHC_Prog.Markers$Status[Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsHC_Prog.Markers$Status <- factor(Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_PD.csv",sep = ""), quote = FALSE)
        
        
        
        
        PosIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_PDvsHC <- Top_Ident.PDvsHC_Prog.Markers$gene[Top_Ident.PDvsHC_Prog.Markers$avg_log2FC < -1]
     IdentScatter_PDvsHC <- ggplot(Top_Ident.PDvsHC_Prog.Markers, aes(HC_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
         
        if (length(PosIdentGenes_PDvsHC) > 0) {       
                IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = PosIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        
        if (length(NegIdentGenes_PDvsHC) > 0) {
                IdentScatter_PDvsHC <- LabelPoints(plot = IdentScatter_PDvsHC, points = NegIdentGenes_PDvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
     
        ggsave(IdentScatter_PDvsHC, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_HC_and_PD.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")
        
        
        # ILB vs HC
        
        
        Ident.ILBvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "ILB", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.ILBvsHC_Prog.Markers$gene <- rownames(Ident.ILBvsHC_Prog.Markers)
        Ident.ILBvsHC_Prog.Markers <- Ident.ILBvsHC_Prog.Markers[Ident.ILBvsHC_Prog.Markers$p_val_adj <= 0.05,]
        
        Top_Ident.ILBvsHC_Prog.Markers <- Ident.ILBvsHC_Prog.Markers[1:50,]
        
        avg.Ident_Object_ILBvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_ILBvsHC$gene <- rownames(avg.Ident_Object_ILBvsHC)
        
        avg.Ident_Object_topmarkers_ILBvsHC <- avg.Ident_Object_ILBvsHC[avg.Ident_Object_ILBvsHC$gene %in% unique(Top_Ident.ILBvsHC_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_ILBvsHC <- avg.Ident_Object_topmarkers_ILBvsHC[match(Top_Ident.ILBvsHC_Prog.Markers$gene,avg.Ident_Object_topmarkers_ILBvsHC$gene),]
        Top_Ident.ILBvsHC_Prog.Markers$ILB_mean <- avg.Ident_Object_topmarkers_ILBvsHC$ILB
        Top_Ident.ILBvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_topmarkers_ILBvsHC$HC
        Top_Ident.ILBvsHC_Prog.Markers$Status <- "Upregulated"
        Top_Ident.ILBvsHC_Prog.Markers$Status[Top_Ident.ILBvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.ILBvsHC_Prog.Markers$Status <- factor(Top_Ident.ILBvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Top_Ident.ILBvsHC_Prog.Markers,file = paste("Files/DE_Genes/Top_50_SigGenes_",Cluster_Ident,"_Markers_HC_vs_ILB.csv",sep = ""), quote = FALSE)
        
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_ILBvsHC[avg.Ident_Object_ILBvsHC$gene %in% unique(Ident.ILBvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_progmarkers_ILBvsHC[match(Ident.ILBvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_ILBvsHC$gene),]
        Ident.ILBvsHC_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_ILBvsHC$ILB
        Ident.ILBvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_ILBvsHC$HC
        Ident.ILBvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.ILBvsHC_Prog.Markers$Status[Ident.ILBvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.ILBvsHC_Prog.Markers$Status <- factor(Ident.ILBvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.ILBvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_ILB.csv",sep = ""), quote = FALSE)
        
        PosIdentGenes_ILBvsHC <- Top_Ident.ILBvsHC_Prog.Markers$gene[Top_Ident.ILBvsHC_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_ILBvsHC <- Top_Ident.ILBvsHC_Prog.Markers$gene[Top_Ident.ILBvsHC_Prog.Markers$avg_log2FC < -1]
     IdentScatter_ILBvsHC <- ggplot(Top_Ident.ILBvsHC_Prog.Markers, aes(HC_mean, ILB_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("HC Mean") +
                ylab("ILB Mean") +
                scale_colour_manual(values = (c("red","blue")))
     
        
        if (length(PosIdentGenes_ILBvsHC) > 0) {       
                IdentScatter_ILBvsHC <- LabelPoints(plot = IdentScatter_ILBvsHC, points = PosIdentGenes_ILBvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        if (length(NegIdentGenes_ILBvsHC)) {
                IdentScatter_ILBvsHC <- LabelPoints(plot = IdentScatter_ILBvsHC, points = NegIdentGenes_ILBvsHC, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
        ggsave(IdentScatter_ILBvsHC, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_HC_and_ILB.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")
        
        
        # PD vs ILB
        
        
                Ident.PDvsILB_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "ILB", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.PDvsILB_Prog.Markers$gene <- rownames(Ident.PDvsILB_Prog.Markers)
        Ident.PDvsILB_Prog.Markers <-
                Ident.PDvsILB_Prog.Markers[Ident.PDvsILB_Prog.Markers$p_val_adj <= 0.05,]
        
        Top_Ident.PDvsILB_Prog.Markers <- Ident.PDvsILB_Prog.Markers[1:50,]
        
        avg.Ident_Object_PDvsILB <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsILB$gene <- rownames(avg.Ident_Object_PDvsILB)
        avg.Ident_Object_topmarkers_PDvsILB <- avg.Ident_Object_PDvsILB[avg.Ident_Object_PDvsILB$gene %in% unique(Top_Ident.PDvsILB_Prog.Markers$gene),]
        avg.Ident_Object_topmarkers_PDvsILB <- avg.Ident_Object_topmarkers_PDvsILB[match(Top_Ident.PDvsILB_Prog.Markers$gene,avg.Ident_Object_topmarkers_PDvsILB$gene),]
        Top_Ident.PDvsILB_Prog.Markers$PD_mean <- avg.Ident_Object_topmarkers_PDvsILB$PD
        Top_Ident.PDvsILB_Prog.Markers$ILB_mean <- avg.Ident_Object_topmarkers_PDvsILB$ILB
        Top_Ident.PDvsILB_Prog.Markers$Status <- "Upregulated"
        Top_Ident.PDvsILB_Prog.Markers$Status[Top_Ident.PDvsILB_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Top_Ident.PDvsILB_Prog.Markers$Status <- factor(Top_Ident.PDvsILB_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Top_Ident.PDvsILB_Prog.Markers,file = paste("Files/DE_Genes/Top_50_SigGenes_",Cluster_Ident,"_Markers_ILB_vs_PD.csv",sep = ""), quote = FALSE)
        
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_PDvsILB[avg.Ident_Object_PDvsILB$gene %in% unique(Ident.PDvsILB_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_progmarkers_PDvsILB[match(Ident.PDvsILB_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsILB$gene),]
        Ident.PDvsILB_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsILB$PD
        Ident.PDvsILB_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_PDvsILB$ILB
        Ident.PDvsILB_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsILB_Prog.Markers$Status[Ident.PDvsILB_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsILB_Prog.Markers$Status <- factor(Ident.PDvsILB_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsILB_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_ILB_vs_PD.csv",sep = ""), quote = FALSE)
        
        
        PosIdentGenes_PDvsILB <- Top_Ident.PDvsILB_Prog.Markers$gene[Top_Ident.PDvsILB_Prog.Markers$avg_log2FC > 1]
        NegIdentGenes_PDvsILB <- Top_Ident.PDvsILB_Prog.Markers$gene[Top_Ident.PDvsILB_Prog.Markers$avg_log2FC < -1]
     IdentScatter_PDvsILB <- ggplot(Top_Ident.PDvsILB_Prog.Markers, aes(ILB_mean, PD_mean, color = Status)) + geom_point(size = 0.5) + ggtitle(deparse(substitute(Cluster_Ident))) +
                theme_cowplot(12) +
                theme(legend.position = "none",
                      plot.title = element_text(hjust = 0.5)) +
                xlab("ILB Mean") +
                ylab("PD Mean") +
                scale_colour_manual(values = (c("red","blue")))
        
        
        if (length(PosIdentGenes_PDvsILB) > 0) {       
                IdentScatter_PDvsILB <- LabelPoints(plot = IdentScatter_PDvsILB, points = PosIdentGenes_PDvsILB, repel = TRUE, xnudge = 0, ynudge = 0, size = 3, color = "red")
        }
        if (length(NegIdentGenes_PDvsILB) > 0) {
                IdentScatter_PDvsILB <- LabelPoints(plot = IdentScatter_PDvsILB, points = NegIdentGenes_PDvsILB, repel = TRUE, xnudge = 0, ynudge = 0, size = 3,
                  color = "blue")
        }
        ggsave(IdentScatter_PDvsILB, filename = paste("Figures/DE_Genes/",Cluster_Ident,"Scatter_Top_50_SigGenes_ILB_and_PD.pdf",sep=""), device = "pdf", width = 6, height = 6, units = "in")

}

```

stripped down version

```{R}
MASTDEcasePerCluster <- function(SeuObj,Cluster_Ident){
        Ident_Object <- subset(SeuObj, idents = Cluster_Ident)
        Ident_Object <- Ident_Object[apply(Ident_Object@assays$RNA@counts,1,mkfilter) >= 20,]
        Ident_Object@meta.data$DetRate <- as.vector(scale(colSums(Ident_Object@assays$RNA@counts)))
        Idents(Ident_Object) <- "case"
        
        #PD vs HC
        Ident.PDvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.PDvsHC_Prog.Markers$gene <- rownames(Ident.PDvsHC_Prog.Markers)
        Ident.PDvsHC_Prog.Markers <-
                Ident.PDvsHC_Prog.Markers[Ident.PDvsHC_Prog.Markers$p_val_adj <= 0.05,]
        
        avg.Ident_Object_PDvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsHC$gene <- rownames(avg.Ident_Object_PDvsHC)
        
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_progmarkers_PDvsHC[match(Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsHC$gene),]
        Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsHC$PD
        Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_PDvsHC$HC
        Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsHC_Prog.Markers$Status[Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsHC_Prog.Markers$Status <- factor(Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_PD.csv",sep = ""), quote = FALSE)

        
        
        # ILB vs HC
        
        
        Ident.ILBvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "ILB", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.ILBvsHC_Prog.Markers$gene <- rownames(Ident.ILBvsHC_Prog.Markers)
        Ident.ILBvsHC_Prog.Markers <- Ident.ILBvsHC_Prog.Markers[Ident.ILBvsHC_Prog.Markers$p_val_adj <= 0.05,]
        
        avg.Ident_Object_ILBvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_ILBvsHC$gene <- rownames(avg.Ident_Object_ILBvsHC)
        
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_ILBvsHC[avg.Ident_Object_ILBvsHC$gene %in% unique(Ident.ILBvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_progmarkers_ILBvsHC[match(Ident.ILBvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_ILBvsHC$gene),]
        Ident.ILBvsHC_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_ILBvsHC$ILB
        Ident.ILBvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_ILBvsHC$HC
        Ident.ILBvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.ILBvsHC_Prog.Markers$Status[Ident.ILBvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.ILBvsHC_Prog.Markers$Status <- factor(Ident.ILBvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.ILBvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_ILB.csv",sep = ""), quote = FALSE)
        
        # PD vs ILB
        
        Ident.PDvsILB_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "ILB", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
        Ident.PDvsILB_Prog.Markers$gene <- rownames(Ident.PDvsILB_Prog.Markers)
        Ident.PDvsILB_Prog.Markers <-
                Ident.PDvsILB_Prog.Markers[Ident.PDvsILB_Prog.Markers$p_val_adj <= 0.05,]
        
        avg.Ident_Object_PDvsILB <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsILB$gene <- rownames(avg.Ident_Object_PDvsILB)
        
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_PDvsILB[avg.Ident_Object_PDvsILB$gene %in% unique(Ident.PDvsILB_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_progmarkers_PDvsILB[match(Ident.PDvsILB_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsILB$gene),]
        Ident.PDvsILB_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsILB$PD
        Ident.PDvsILB_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_PDvsILB$ILB
        Ident.PDvsILB_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsILB_Prog.Markers$Status[Ident.PDvsILB_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsILB_Prog.Markers$Status <- factor(Ident.PDvsILB_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsILB_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_ILB_vs_PD.csv",sep = ""), quote = FALSE)
    
}

```


```{R}



```



```{R}



```





non function version

```{R}
Ident_Object <- subset(SeuObj, idents = Cluster_Ident)
Cluster_Ident
Ident_Object <- Ident_Object[apply(Ident_Object@assays$RNA@counts,1,mkfilter) >= 20,]
Ident_Object@meta.data$DetRate <- as.vector(scale(colSums(Ident_Object@assays$RNA@counts)))
Idents(Ident_Object) <- "case"

#PD vs HC
Ident.PDvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
Ident.PDvsHC_Prog.Markers$gene <- rownames(Ident.PDvsHC_Prog.Markers)
Ident.PDvsHC_Prog.Markers <-
        Ident.PDvsHC_Prog.Markers[Ident.PDvsHC_Prog.Markers$p_val_adj <= 0.05,]

if (length(Ident.PDvsHC_Prog.Markers$gene) != 0) {
        avg.Ident_Object_PDvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsHC$gene <- rownames(avg.Ident_Object_PDvsHC)
        
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_PDvsHC[avg.Ident_Object_PDvsHC$gene %in% unique(Ident.PDvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsHC <- avg.Ident_Object_progmarkers_PDvsHC[match(Ident.PDvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsHC$gene),]
        Ident.PDvsHC_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsHC$PD
        Ident.PDvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_PDvsHC$HC
        Ident.PDvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsHC_Prog.Markers$Status[Ident.PDvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsHC_Prog.Markers$Status <- factor(Ident.PDvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_PD.csv",sep = ""), quote = FALSE)
}



# ILB vs HC


Ident.ILBvsHC_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "ILB", ident.2 = "HC", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
Ident.ILBvsHC_Prog.Markers$gene <- rownames(Ident.ILBvsHC_Prog.Markers)
Ident.ILBvsHC_Prog.Markers <- Ident.ILBvsHC_Prog.Markers[Ident.ILBvsHC_Prog.Markers$p_val_adj <= 0.05,]

if (length(Ident.ILBvsHC_Prog.Markers$gene) != 0){
        avg.Ident_Object_ILBvsHC <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_ILBvsHC$gene <- rownames(avg.Ident_Object_ILBvsHC)
        
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_ILBvsHC[avg.Ident_Object_ILBvsHC$gene %in% unique(Ident.ILBvsHC_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_ILBvsHC <- avg.Ident_Object_progmarkers_ILBvsHC[match(Ident.ILBvsHC_Prog.Markers$gene,avg.Ident_Object_progmarkers_ILBvsHC$gene),]
        Ident.ILBvsHC_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_ILBvsHC$ILB
        Ident.ILBvsHC_Prog.Markers$HC_mean <- avg.Ident_Object_progmarkers_ILBvsHC$HC
        Ident.ILBvsHC_Prog.Markers$Status <- "Upregulated"
        Ident.ILBvsHC_Prog.Markers$Status[Ident.ILBvsHC_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.ILBvsHC_Prog.Markers$Status <- factor(Ident.ILBvsHC_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.ILBvsHC_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_HC_vs_ILB.csv",sep = ""), quote = FALSE)
}

# PD vs ILB

Ident.PDvsILB_Prog.Markers <- FindMarkers(Ident_Object, ident.1 = "PD", ident.2 = "ILB", verbose = FALSE, test.use = "MAST", latent.vars = c("sex","DetRate"))
Ident.PDvsILB_Prog.Markers$gene <- rownames(Ident.PDvsILB_Prog.Markers)
Ident.PDvsILB_Prog.Markers <-
        Ident.PDvsILB_Prog.Markers[Ident.PDvsILB_Prog.Markers$p_val_adj <= 0.05,]

if (length(Ident.PDvsILB_Prog.Markers$gene) != 0) {
        avg.Ident_Object_PDvsILB <- as.data.frame(AverageExpression(Ident_Object, verbose = FALSE)$RNA)
        avg.Ident_Object_PDvsILB$gene <- rownames(avg.Ident_Object_PDvsILB)
        
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_PDvsILB[avg.Ident_Object_PDvsILB$gene %in% unique(Ident.PDvsILB_Prog.Markers$gene),]
        avg.Ident_Object_progmarkers_PDvsILB <- avg.Ident_Object_progmarkers_PDvsILB[match(Ident.PDvsILB_Prog.Markers$gene,avg.Ident_Object_progmarkers_PDvsILB$gene),]
        Ident.PDvsILB_Prog.Markers$PD_mean <- avg.Ident_Object_progmarkers_PDvsILB$PD
        Ident.PDvsILB_Prog.Markers$ILB_mean <- avg.Ident_Object_progmarkers_PDvsILB$ILB
        Ident.PDvsILB_Prog.Markers$Status <- "Upregulated"
        Ident.PDvsILB_Prog.Markers$Status[Ident.PDvsILB_Prog.Markers$avg_log2FC < 0] <- "Downregulated"
        Ident.PDvsILB_Prog.Markers$Status <- factor(Ident.PDvsILB_Prog.Markers$Status, c("Upregulated","Downregulated"))
        
        write.csv(Ident.PDvsILB_Prog.Markers,file = paste("Files/DE_Genes/All_SigGenes_",Cluster_Ident,"_Markers_ILB_vs_PD.csv",sep = ""), quote = FALSE)
}
```



