---
title: "Using DE Gene Tables to Construct a Heatmap"
output: html_document
---

Constructing heatmap version 1 

```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```


```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```


```{R}
#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons", `3` = "GLU_Neurons",
                      `4` = "GLU_Neurons",`5` = "Astrocytes",
                      `6` = "GABA_Neurons", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons",`9` = "GLU_Neurons",
                      `10` = "GLU_Neurons", `11` = "Microglia",`12` = "Astrocytes",
                      `13` = "OPCs",`14` = "GABA_Neurons",
                      `15` = "GLU_Neurons", `16`="GLU_Neurons", `17`="Endothelial", `18`="Endothelial",`19`="GABA_Neurons",`20`="GLU_Neurons",`21`="GLU_Neurons",`22`="GLU_Neurons",`23`="GABA_Neurons")
```


```{R}
#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes/",list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes/All_SigGenes_|.csv","",i),read.csv(i))
}
```


```{R}
DEdfList <- ls()[grep("HC_vs_PD",ls())]

DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]


TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  if (nrow(tempdf) >= 10){
  tempdf <- tempdf[1:10,]
  tempdf$CellType <- gsub("_Markers_HC_vs_PD","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
  }
}

```

###############################################################################################################################################

Checking that the scale.data slot is the scaled version of the data in the "data" slot - Yes!

```{R}

Batch567_MTG@assays$RNA@scale.data[,1]

0.5755392

1.185521594

(0.5755392 - mean(Batch567_MTG@assays$RNA@data[9,]))/sd(Batch567_MTG@assays$RNA@data[9,])
```

################################################################################################################################################

```{R}

TopDEGenes <- TopDEGenes %>% group_by(CellType) %>% arrange(desc(avg_log2FC),.by_group = TRUE)

```

```{R}
#PerSampleAvgExp <- AverageExpression(Batch567_MTG, slot = "scale.data", group.by = "sample_ID", features = TopDEGenes$gene)$RNA

#PerSampleAvgExp <- as.data.frame(AverageExpression(Batch567_MTG, slot = "data", group.by = "sample_ID")$RNA)

PerSampleAvgExp=NULL
for (Cluster in unique(TopDEGenes$CellType)){
  geneset <- TopDEGenes$gene[TopDEGenes$CellType %in% Cluster]
  IdentObj <- subset(Batch567_MTG, idents = Cluster)
  AvgIdentExp <- AverageExpression(IdentObj, slot = "scale.data", group.by = "sample_ID", features = geneset)$RNA
  rownames(AvgIdentExp) <- paste(Cluster,rownames(AvgIdentExp), sep = "_")
  PerSampleAvgExp <- rbind(PerSampleAvgExp,AvgIdentExp)
}

samplecase <- NULL
for (sname in colnames(PerSampleAvgExp)) {
  samplecase <- c(samplecase, unique(Batch567_MTG@meta.data[Batch567_MTG@meta.data$sample_ID %in% sname,]$case))
}


```


```{R}

PerSampleAvgExpFilter <- PerSampleAvgExp[,-which(samplecase %in% "ILB")]

samplecasefilter <- samplecase[-which(samplecase %in% "ILB")]

names(samplecasefilter) <- colnames(PerSampleAvgExpFilter)

samplecasefilter[order(samplecasefilter)]



```

```{R}
PerSampleAvgExpFilterorder <- PerSampleAvgExpFilter[,order(samplecasefilter)]


colnames(PerSampleAvgExpFilterorder)
```





heatframe=NULL
for (celltype in unique(TopDEGenes$CellType)) {
  tempdf <- PerSampleAvgExpFilterorder[rownames(PerSampleAvgExpFilterorder) %in% TopDEGenes$gene[TopDEGenes$CellType %in% celltype],]
  rownames(tempdf) <- paste(celltype,rownames(tempdf),sep = "_")
  heatframe <- rbind(heatframe,tempdf)
}

```{R}

heatframe <- as.matrix(PerSampleAvgExpFilterorder)

heatmap.2(as.matrix(heatframe),dendrogram='none', Rowv=FALSE, Colv=FALSE,trace='none')

heatmap(as.matrix(heatframe),dendrogram='none', Rowv=FALSE, Colv=FALSE,trace='none')
```

```{R}



heatframety <- heatframe[-grep("MALAT|KCNIP4", rownames(heatframe)),]

pheatmap(as.matrix(heatframe),cluster_rows=FALSE, cluster_cols=FALSE)




```

Hmmm not very informative - here we scale (z-score norm) across the rows instead?

```{R}
#PerSampleAvgExp <- AverageExpression(Batch567_MTG, slot = "scale.data", group.by = "sample_ID", features = TopDEGenes$gene)$RNA

#PerSampleAvgExp <- as.data.frame(AverageExpression(Batch567_MTG, slot = "data", group.by = "sample_ID")$RNA)


PerSampleAvgExp=NULL
for (Cluster in unique(TopDEGenes$CellType)){
  geneset <- TopDEGenes$gene[TopDEGenes$CellType %in% Cluster]
  IdentObj <- subset(Batch567_MTG, idents = Cluster)
  AvgIdentExp <- AverageExpression(IdentObj, slot = "data", group.by = "sample_ID", features = geneset)$RNA
  rownames(AvgIdentExp) <- paste(Cluster,rownames(AvgIdentExp), sep = "_")
  PerSampleAvgExp <- rbind(PerSampleAvgExp,AvgIdentExp)
}

samplecase <- NULL
for (sname in colnames(PerSampleAvgExp)) {
  samplecase <- c(samplecase, unique(Batch567_MTG@meta.data[Batch567_MTG@meta.data$sample_ID %in% sname,]$case))
}


```



```{R}

PerSampleAvgExpFilter <- PerSampleAvgExp[,-which(samplecase %in% "ILB")]

samplecasefilter <- samplecase[-which(samplecase %in% "ILB")]

names(samplecasefilter) <- colnames(PerSampleAvgExpFilter)

samplecasefilter[order(samplecasefilter)]



```

```{R}
PerSampleAvgExpFilterorder <- PerSampleAvgExpFilter[,order(samplecasefilter)]

colnames(PerSampleAvgExpFilterorder)
```


```{R}

test <- t(apply(PerSampleAvgExpFilterorder,1,scale))

```


```{R}

heatframety <- heatframe[-grep("MALAT|KCNIP4", rownames(heatframe)),]

pheatmap(as.matrix(test),cluster_rows=FALSE, cluster_cols=FALSE)

```

Still not great, too variable across samples

with means?


```{R}

test <- as.matrix(TopDEGenes[,c(8,9)])

test2 <- t(apply(test,1,scale))

pheatmap(as.matrix(test2),cluster_rows=FALSE, cluster_cols=FALSE)

```

Doesn't really work.....


#########################################################################################################################################################


Constructing heatmap version 2 (Clemens' version)


```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```


```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```


```{R}
#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons", `3` = "GLU_Neurons",
                      `4` = "GLU_Neurons",`5` = "Astrocytes",
                      `6` = "GABA_Neurons", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons",`9` = "GLU_Neurons",
                      `10` = "GLU_Neurons", `11` = "Microglia",`12` = "Astrocytes",
                      `13` = "OPCs",`14` = "GABA_Neurons",
                      `15` = "GLU_Neurons", `16`="GLU_Neurons", `17`="Endothelial", `18`="Endothelial",`19`="GABA_Neurons",`20`="GLU_Neurons",`21`="GLU_Neurons",`22`="GLU_Neurons",`23`="GABA_Neurons")
```


```{R}
#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes/",list.files("Files/DE_Genes/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes/AllGenes_|.csv","",i),read.csv(i))
}

```



```{R}
DEdfList <- ls()[grep("HC_vs_PD",ls())]

DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]

#  tempdf <- tempdf[1:10,]

TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  tempdf <- tempdf[tempdf$p_val <= tempdf$BF_p_val,]
  tempdf$CellType <- gsub("_Markers_HC_vs_PD","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
}

```



```{R}

TriplicateGenes=NULL
for (genename in unique(TopDEGenes$gene)){
  genesub <- TopDEGenes$gene[TopDEGenes$gene %in% genename]
  if (length(genesub) >= 3){
    TriplicateGenes <- c(TriplicateGenes,genename)
  }
} 

```


```{R}

Pvalframe=NULL
Thresholdframe=NULL
for (geneident in TriplicateGenes){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfList){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_p_val <- tempdf[tempdf$gene %in% geneident,]$p_val
      if (temp_p_val <= unique(tempdf$BF_p_val)){
        temp_threshold <- 1
      } else {
        temp_threshold <- 0.5
      }
      if (temp_p_val == 0){
        temp_p_val <- .Machine$double.xmin
        temp_threshold <- 1
      }
      generow <- cbind(generow, temp_p_val)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, NA)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 Pvalframe <- rbind(Pvalframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(Pvalframe) <- gsub("_Markers_HC_vs_PD","",DEdfList)
rownames(Pvalframe) <- TriplicateGenes

colnames(Thresholdframe) <- gsub("_Markers_HC_vs_PD","",DEdfList)
rownames(Thresholdframe) <- TriplicateGenes

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```


```{R}




```


```{R}

pheatmap(as.matrix(test),cluster_rows=FALSE, cluster_cols=FALSE)



```



```{R}

pheatmap(as.matrix(Pvalframe))

```


```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

ggsave(testmap,filename = "Files/HeatmapTest.pdf", device = "pdf", height = 30, width = 6, units = "in")

```


Making version 5 (top 5 genes for aggregate groups across separate clusters, colour by p-value)


```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```


```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```


```{R}
#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons_1", `3` = "GLU_Neurons_2",
                      `4` = "GLU_Neurons_3",`5` = "Astrocytes_1",
                      `6` = "GABA_Neurons_1", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons_4",`9` = "GLU_Neurons_5",
                      `10` = "GLU_Neurons_6", `11` = "Microglia",`12` = "Astrocytes_2",
                      `13` = "OPCs",`14` = "GABA_Neurons_2",
                      `15` = "GLU_Neurons_7", `16`="GLU_Neurons_8", `17`="Endothelial_1", `18`="Endothelial_2",`19`="GABA_Neurons_3",`20`="GLU_Neurons_9",`21`="GLU_Neurons_10",`22`="GLU_Neurons_11",`23`="GABA_Neurons_4")
```



```{R}
#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes/",list.files("Files/DE_Genes/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes/AllGenes_|.csv","",i),read.csv(i))
}

```



```{R}
DEdfList <- ls()[grep("HC_vs_PD",ls())]

DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]

#  tempdf <- tempdf[1:10,]

TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  tempdf <- tempdf[tempdf$p_val <= tempdf$BF_p_val,]
  if (nrow(tempdf) >= 5){
  tempdf <- tempdf[1:10,]
  tempdf$CellType <- gsub("_Markers_HC_vs_PD","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
  }
}

```

```{R}

library(gdata)

keep(TopDEGenes, sure = TRUE)

```


```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/|.csv","",i),read.csv(i))
}

```

```{R}

DEdfListAllGenes<- ls()[grep("AllGenes",ls())]

```

```{R}

lfcframe=NULL
Thresholdframe=NULL
for (geneident in unique(TopDEGenes$gene)){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfListAllGenes){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_lfc <- tempdf[tempdf$gene %in% geneident,]$avg_log2FC
      if (temp_lfc >= 0.5){
        temp_threshold <- 1
      } 
      
      if (temp_lfc <= -0.5) {
        temp_threshold <- -1
      }
      if (temp_lfc > -0.5 & temp_lfc < 0.5) {
        temp_threshold <- 0
      }
      generow <- cbind(generow, temp_lfc)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, NA)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 lfcframe <- rbind(lfcframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(lfcframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(lfcframe) <- unique(TopDEGenes$gene)

colnames(Thresholdframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(Thresholdframe) <- unique(TopDEGenes$gene)

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```


```{R}

pheatmap(Thresholdframe)

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

ggsave(testmap,filename = "Files/HeatmapTest2.pdf", device = "pdf", height = 30, width = 6, units = "in")

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("magenta","black","#F1C40F"), cellwidth = 10, cellheight = 10)

ggsave(testmap,filename = "Files/HeatmapTest3.pdf", device = "pdf", height = 30, width = 8, units = "in")

```
Repeating above heatmap but implementing a p-value cut-off


```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```


```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```


```{R}
#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons_1", `3` = "GLU_Neurons_2",
                      `4` = "GLU_Neurons_3",`5` = "Astrocytes_1",
                      `6` = "GABA_Neurons_1", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons_4",`9` = "GLU_Neurons_5",
                      `10` = "GLU_Neurons_6", `11` = "Microglia",`12` = "Astrocytes_2",
                      `13` = "OPCs",`14` = "GABA_Neurons_2",
                      `15` = "GLU_Neurons_7", `16`="GLU_Neurons_8", `17`="Endothelial_1", `18`="Endothelial_2",`19`="GABA_Neurons_3",`20`="GLU_Neurons_9",`21`="GLU_Neurons_10",`22`="GLU_Neurons_11",`23`="GABA_Neurons_4")
```



```{R}
#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes/",list.files("Files/DE_Genes/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes/AllGenes_|.csv","",i),read.csv(i))
}

```



```{R}
DEdfList <- ls()[grep("HC_vs_PD",ls())]

DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]

#  tempdf <- tempdf[1:10,]

TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  tempdf <- tempdf[tempdf$p_val <= tempdf$BF_p_val,]
  if (nrow(tempdf) >= 5){
  tempdf <- tempdf[1:10,]
  tempdf$CellType <- gsub("_Markers_HC_vs_PD","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
  }
}

```

```{R}

library(gdata)

keep(TopDEGenes, sure = TRUE)

```


```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/|.csv","",i),read.csv(i))
}

```

```{R}

DEdfListAllGenes<- ls()[grep("AllGenes",ls())]

```

```{R}

lfcframe=NULL
Thresholdframe=NULL
for (geneident in unique(TopDEGenes$gene)){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfListAllGenes){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_lfc <- tempdf[tempdf$gene %in% geneident,]$avg_log2FC
      temp_pval <- tempdf[tempdf$gene %in% geneident,]$p_val
      if (temp_lfc > 0 & temp_pval <= unique(tempdf$BF_p_val)){
        temp_threshold <- 1
      } 
      
      if (temp_lfc < 0 & temp_pval <= unique(tempdf$BF_p_val)) {
        temp_threshold <- -1
      }
      if (temp_pval >= unique(tempdf$BF_p_val)) {
        temp_threshold <- 0
      }
      generow <- cbind(generow, temp_lfc)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, NA)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 lfcframe <- rbind(lfcframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(lfcframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(lfcframe) <- unique(TopDEGenes$gene)

colnames(Thresholdframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(Thresholdframe) <- unique(TopDEGenes$gene)

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```


```{R}

pheatmap(Thresholdframe)

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

ggsave(testmap,filename = "Files/HeatmapTest2.pdf", device = "pdf", height = 30, width = 6, units = "in")

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("magenta","black","#F1C40F"), cellwidth = 10, cellheight = 10)

ggsave(testmap,filename = "Files/HeatmapTest4.pdf", device = "pdf", height = 30, width = 8, units = "in")

```



Further alternative with finer logFC gradations

```{R}

lfcframe=NULL
Thresholdframe=NULL
for (geneident in unique(TopDEGenes$gene)){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfListAllGenes){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_lfc <- tempdf[tempdf$gene %in% geneident,]$avg_log2FC
      temp_pval <- tempdf[tempdf$gene %in% geneident,]$p_val
      if (temp_lfc >= 0.5 & temp_pval <= unique(tempdf$BF_p_val)){
        temp_threshold <- 1
      } 
      if (temp_lfc > 0 & temp_lfc < 0.5 & temp_pval <= unique(tempdf$BF_p_val)){
        temp_threshold <- 0.5
      } 
      if (temp_lfc <= -0.5 & temp_pval <= unique(tempdf$BF_p_val)) {
        temp_threshold <- -1
      }
      if (temp_lfc > -0.5 & temp_lfc < 0 & temp_pval <= unique(tempdf$BF_p_val)) {
        temp_threshold <- -0.5
      }
      if (temp_pval >= unique(tempdf$BF_p_val)) {
        temp_threshold <- 0
      }
      generow <- cbind(generow, temp_lfc)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, NA)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 lfcframe <- rbind(lfcframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(lfcframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(lfcframe) <- unique(TopDEGenes$gene)

colnames(Thresholdframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(Thresholdframe) <- unique(TopDEGenes$gene)

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```

```{R}

pheatmap(Thresholdframe)

```


```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

ggsave(testmap,filename = "Files/HeatmapTest2.pdf", device = "pdf", height = 30, width = 6, units = "in")

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("magenta","#CCCCFF","black","#F1C40F","#E67E22"), cellwidth = 10, cellheight = 10)

ggsave(testmap,filename = "Files/HeatmapTest5.pdf", device = "pdf", height = 30, width = 8, units = "in")

```


Using top genes DE for each subcluster

```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```


```{R}
Batch567_MTG=readRDS("Files/Batch567_Unassigned.rds")
Batch567_MTG@meta.data$batch <- gsub("Batch56767","Batch5",Batch567_MTG@meta.data$batch)
```


```{R}
#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")
```


```{R}
Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons_1", `3` = "GLU_Neurons_2",
                      `4` = "GLU_Neurons_3",`5` = "Astrocytes_1",
                      `6` = "GABA_Neurons_1", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons_4",`9` = "GLU_Neurons_5",
                      `10` = "GLU_Neurons_6", `11` = "Microglia",`12` = "Astrocytes_2",
                      `13` = "OPCs",`14` = "GABA_Neurons_2",
                      `15` = "GLU_Neurons_7", `16`="GLU_Neurons_8", `17`="Endothelial_1", `18`="Endothelial_2",`19`="GABA_Neurons_3",`20`="GLU_Neurons_9",`21`="GLU_Neurons_10",`22`="GLU_Neurons_11",`23`="GABA_Neurons_4")
```



```{R}
#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/AllGenes_|.csv","",i),read.csv(i))
}

```



```{R}
DEdfList <- ls()[grep("HC_vs_PD",ls())]

#DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]

#  tempdf <- tempdf[1:10,]

TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  tempdf <- tempdf[tempdf$p_val <= tempdf$BF_p_val,]
  if (nrow(tempdf) >= 10){
  tempdf <- tempdf[1:5,]
  tempdf$CellType <- gsub("_Markers_HC_vs_PD","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
  }
}

```

```{R}

library(gdata)

keep(TopDEGenes, sure = TRUE)

```


```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/|.csv","",i),read.csv(i))
}

```

```{R}

DEdfListAllGenes<- ls()[grep("AllGenes",ls())]

```

```{R}

lfcframe=NULL
Thresholdframe=NULL
for (geneident in unique(TopDEGenes$gene)){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfListAllGenes){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_lfc <- tempdf[tempdf$gene %in% geneident,]$avg_log2FC
      temp_pval <- tempdf[tempdf$gene %in% geneident,]$p_val
      if (temp_lfc > 0 & temp_pval <= unique(tempdf$BF_p_val)){
        temp_threshold <- 1
      } 
      
      if (temp_lfc < 0 & temp_pval <= unique(tempdf$BF_p_val)) {
        temp_threshold <- -1
      }
      if (temp_pval >= unique(tempdf$BF_p_val)) {
        temp_threshold <- 0
      }
      generow <- cbind(generow, temp_lfc)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, 0)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 lfcframe <- rbind(lfcframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(lfcframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(lfcframe) <- unique(TopDEGenes$gene)

colnames(Thresholdframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(Thresholdframe) <- unique(TopDEGenes$gene)

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```


```{R}

pheatmap(lfcframe)

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

#ggsave(testmap,filename = "Files/HeatmapTest2.pdf", device = "pdf", height = 30, width = 6, units = "in")

```
scale_colour_gradient(low = "magenta", high = "yellow")
```{R}
werd <- colorRampPalette(c("magenta","#C228C8","#813384","#000000","#817E3E","#BEB72F","#FFF300"))(200)

testmap <- pheatmap(lfcframe, border_color = "gray", color = werd, cellwidth = 10, cellheight = 10, breaks = seq(to=-0.5,from=0.5, length.out=200))

ggsave(testmap,filename = "Files/HeatmapTest4.pdf", device = "pdf", height = 30, width = 8, units = "in")

```









