---
title: "Differentiating_between_DA_Neuron_subtypes"
output: html_document
---

In this markdown we attempt to differentiate between the two different DA neuron subtypes we identified as clusters 12 and 18. 

First we show that these clusters are highly likely to be DA neurons by showing they express DA specific markers


```{R}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
library(DOSE)
library(GOSemSim)
library(enrichplot)

```

```{R}
AllMB <- readRDS("Pilot1_and2_Integration_Files/Case_Corrected_Pilot1and2_MB_Unassigned.rds")
```


```{r}
AllMB <- RenameIdents(AllMB, `1` = "Oligodendrocytes", `2` = "Astrocytes",
                      `3` = "Oligodendrocytes", `4` = "GABA_Neurons", `5` = "Microglia",
                      `6` = "OPCs", `7` = "Endothelial", `8` = "DA Neurons 1",`9` = "Endothelial",
                      `10` = "GLU Neurons", `11` = "Oligodendrocytes",`12` = "GLU Neurons",
                      `13` = "DA Neurons 2",`14` = "Novel Cluster 14",
                      `15` = "CD8+ T Cells", `16`="Novel Cluster 16", `17`="GABA Neurons", `18`="GLU Neurons")
```


```{R}
VlnPlot(AllMB, features = c("TH"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

THVln <- VlnPlot(AllMB, features = c("TH"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(THVln, filename = "Files/TH_Vln_scRNA_seq_Batch4.pdf", device = "pdf", width = 6, height = 4, units = "in")
```
SLC6A3, 

```{R}
VlnPlot(AllMB, features = c("SLC6A3"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

SLC6A3Vln <- VlnPlot(AllMB, features = c("SLC6A3"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(SLC6A3Vln, filename = "Files/SLC6A3_Vln_scRNA_seq_Batch4.pdf", device = "pdf", width = 6, height = 4, units = "in")
```


```{R}
VlnPlot(AllMB, features = c("SLC18A2"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

SLC18A2Vln <- VlnPlot(AllMB, features = c("SLC18A2"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(SLC18A2Vln, filename = "Files/SLC18A2_Vln_scRNA_seq_Batch4.pdf", device = "pdf", width = 6, height = 4, units = "in")
```


These two neuronal subtypes could represent DA neurons from the substantia nigra and the ventral tegmental area. Therefore we will explore the expression of a set of literature derived markers across the two clusters.


CALB1 was historically associated with VTA DA neurons whereas GIRK2/KCNJ6,Sox6 was associated with SN DA neurons. According to a study in mice (see Defining Midbrain Dopaminergic Neuron Diversity by Single-Cell Gene Expression Profiling), SOX6 is indicative of DA1 type neurons (more SN-like). Neither ALDH1A1 or SLC17A6/Vglut2 are indicative in and of themselves of a neuron subtype, but co-expression of Vglut2 and SOX6 is indicative of a more SN-localised/SN-like DA neuron, and neurons expressed ALDH1A1 are reportedly more senstive in a mouse model of PD (see Classification of Midbrain Dopamine Neurons Using Single-Cell Gene Expression Profiling Approaches). 



```{r}
DANeuronSubtypesCellMarkerVlnPlot <- VlnPlot(AllMB, features = c("ALDH1A1","SOX6","CALB1","KCNJ6","SLC17A6"),pt.size = 0, ncol = 1) 

ggsave(DANeuronSubtypesCellMarkerVlnPlot,filename = "Files/DANeuronSubtypesCellMarkerVlnPlot.pdf", width = 20, height = 20)       
```



If one of these clusters corresponds to DA neurons/DA-like neurons then we might expect that the % of cells relative to the whole represented by this cluster would go down in PD patients relative to healthy controls. 

```{R}
table(Idents(AllMB), AllMB$case)
celltable <- table(Idents(AllMB), AllMB$case)
cell_type_number_df <- as.data.frame(celltable)
cell_type_number_df
```



```{R}
totalHC <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC"])
totalPD <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD"])
HCDAneuronsC1 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Dopaminergic Neurons group 1"])
PDDAneuronsC1 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Dopaminergic Neurons group 1"])
DAC1_Fishers_allcells <- cbind(totalHC - HCDAneuronsC1, totalPD - PDDAneuronsC1)
DAC1_Fishers_DAC1_Neurons <- cbind(HCDAneuronsC1, PDDAneuronsC1)

DAC1_Fishers_Table <- rbind(DAC1_Fishers_allcells, DAC1_Fishers_DAC1_Neurons)

colnames(DAC1_Fishers_Table) <- c("HC","PD")
rownames(DAC1_Fishers_Table) <- c("Total Non-DAC1","DAC1")

fisher.test(DAC1_Fishers_Table)

DAC1_Fishers_Table
```

```{R}
totalHC <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC"])
totalPD <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD"])
HCDAneuronsC2 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Dopaminergic Neurons group 2"])
PDDAneuronsC2 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Dopaminergic Neurons group 2"])
DAC2_Fishers_allcells <- cbind(totalHC - HCDAneuronsC2, totalPD - PDDAneuronsC2)
DAC2_Fishers_DAC2_Neurons <- cbind(HCDAneuronsC2, PDDAneuronsC2)

DAC2_Fishers_Table <- rbind(DAC2_Fishers_allcells, DAC2_Fishers_DAC2_Neurons)

colnames(DAC2_Fishers_Table) <- c("HC","PD")
rownames(DAC2_Fishers_Table) <- c("Total Non-DAC2","DAC2")

fisher.test(DAC2_Fishers_Table)

DAC2_Fishers_Table
```


```{R}
prop.table(table(Idents(AllMB), AllMB$case), margin = 2)
```


```{R}
HCOligo <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Oligodendrocytes"])
PDOligo <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Oligodendrocytes"])
Oligo_Fishers_allcells <- cbind(totalHC - HCOligo, totalPD - PDOligo)
Oligo_Fishers_Oligo_Neurons <- cbind(HCOligo, PDOligo)

Oligo_Fishers_Table <- rbind(Oligo_Fishers_allcells, Oligo_Fishers_Oligo_Neurons)

colnames(Oligo_Fishers_Table) <- c("HC","PD")
rownames(Oligo_Fishers_Table) <- c("Total Non-Oligodendrocytes","Oligodenrocytes")

fisher.test(Oligo_Fishers_Table)

Oligo_Fishers_Table
```

```{R}
HCAstro <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Astrocytes"])
PDAstro <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Astrocytes"])
Astro_Fishers_allcells <- cbind(totalHC - HCAstro, totalPD - PDAstro)
Astro_Fishers_Astro_Neurons <- cbind(HCAstro, PDAstro)

Astro_Fishers_Table <- rbind(Astro_Fishers_allcells, Astro_Fishers_Astro_Neurons)

colnames(Astro_Fishers_Table) <- c("HC","PD")
rownames(Astro_Fishers_Table) <- c("Total Non-Astrocytes","Astrocytes")

fisher.test(Astro_Fishers_Table)

Astro_Fishers_Table
```

```{R}
HCAstro <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Astrocytes"])
PDAstro <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Astrocytes"])
Astro_Fishers_allcells <- cbind(totalHC - HCAstro, totalPD - PDAstro)
Astro_Fishers_Astro_Neurons <- cbind(HCAstro, PDAstro)

Astro_Fishers_Table <- rbind(Astro_Fishers_allcells, Astro_Fishers_Astro_Neurons)

colnames(Astro_Fishers_Table) <- c("HC","PD")
rownames(Astro_Fishers_Table) <- c("Total Non-Astrocytes","Astrocytes")

fisher.test(Astro_Fishers_Table)

Astro_Fishers_Table
```



```{R}
HCOPCs <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "OPCs"])
PDOPCs <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "OPCs"])
OPCs_Fishers_allcells <- cbind(totalHC - HCOPCs, totalPD - PDOPCs)
OPCs_Fishers_OPCs_Neurons <- cbind(HCOPCs, PDOPCs)

OPCs_Fishers_Table <- rbind(OPCs_Fishers_allcells, OPCs_Fishers_OPCs_Neurons)

colnames(OPCs_Fishers_Table) <- c("HC","PD")
rownames(OPCs_Fishers_Table) <- c("Total Non-OPCs","OPCs")

fisher.test(OPCs_Fishers_Table)

OPCs_Fishers_Table

```


```{R}
HCGlu_GABA_Neurons <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Glu_GABA neurons"])
PDGlu_GABA_Neurons <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Glu_GABA neurons"])
Glu_GABA_Neurons_Fishers_allcells <- cbind(totalHC - HCGlu_GABA_Neurons, totalPD - PDGlu_GABA_Neurons)
Glu_GABA_Neurons_Fishers_Glu_GABA_Neurons_Neurons <- cbind(HCGlu_GABA_Neurons, PDGlu_GABA_Neurons)

Glu_GABA_Neurons_Fishers_Table <- rbind(Glu_GABA_Neurons_Fishers_allcells, Glu_GABA_Neurons_Fishers_Glu_GABA_Neurons_Neurons)

colnames(Glu_GABA_Neurons_Fishers_Table) <- c("HC","PD")
rownames(Glu_GABA_Neurons_Fishers_Table) <- c("Total Non-Glu_GABA neurons","Glu_GABA neurons")

fisher.test(Glu_GABA_Neurons_Fishers_Table)

Glu_GABA_Neurons_Fishers_Table
```


```{R}
HCMicroglia <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Microglia"])
PDMicroglia <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Microglia"])
Microglia_Fishers_allcells <- cbind(totalHC - HCMicroglia, totalPD - PDMicroglia)
Microglia_Fishers_Microglia_Neurons <- cbind(HCMicroglia, PDMicroglia)

Microglia_Fishers_Table <- rbind(Microglia_Fishers_allcells, Microglia_Fishers_Microglia_Neurons)

colnames(Microglia_Fishers_Table) <- c("HC","PD")
rownames(Microglia_Fishers_Table) <- c("Total Non-Microglia","Microglia")

fisher.test(Microglia_Fishers_Table)

Microglia_Fishers_Table
```


```{R}
HCEndothelialcells <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Endothelial cells"])
PDEndothelialcells <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Endothelial cells"])
Endothelialcells_Fishers_allcells <- cbind(totalHC - HCEndothelialcells, totalPD - PDEndothelialcells)
Endothelialcells_Fishers_Endothelialcells_Neurons <- cbind(HCEndothelialcells, PDEndothelialcells)

Endothelialcells_Fishers_Table <- rbind(Endothelialcells_Fishers_allcells, Endothelialcells_Fishers_Endothelialcells_Neurons)

colnames(Endothelialcells_Fishers_Table) <- c("HC","PD")
rownames(Endothelialcells_Fishers_Table) <- c("Total Non-Endothelial cells","Endothelial cells")

fisher.test(Endothelialcells_Fishers_Table)

Endothelialcells_Fishers_Table
```

```{R}
HCGlu_Neurons <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Glu_Neurons"])
PDGlu_Neurons <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Glu_Neurons"])
Glu_Neurons_Fishers_allcells <- cbind(totalHC - HCGlu_Neurons, totalPD - PDGlu_Neurons)
Glu_Neurons_Fishers_Glu_Neurons_Neurons <- cbind(HCGlu_Neurons, PDGlu_Neurons)

Glu_Neurons_Fishers_Table <- rbind(Glu_Neurons_Fishers_allcells, Glu_Neurons_Fishers_Glu_Neurons_Neurons)

colnames(Glu_Neurons_Fishers_Table) <- c("HC","PD")
rownames(Glu_Neurons_Fishers_Table) <- c("Total Non-Glu_Neurons","Glu_Neurons")

fisher.test(Glu_Neurons_Fishers_Table)

Glu_Neurons_Fishers_Table
```


```{R}
HCNaturalKillerCells <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "Natural Killer Cells?? "])
PDNaturalKillerCells <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "Natural Killer Cells?? "])
NaturalKillerCells_Fishers_allcells <- cbind(totalHC - HCNaturalKillerCells, totalPD - PDNaturalKillerCells)
NaturalKillerCells_Fishers_NaturalKillerCells_Neurons <- cbind(HCNaturalKillerCells, PDNaturalKillerCells)

NaturalKillerCells_Fishers_Table <- rbind(NaturalKillerCells_Fishers_allcells, NaturalKillerCells_Fishers_NaturalKillerCells_Neurons)

colnames(NaturalKillerCells_Fishers_Table) <- c("HC","PD")
rownames(NaturalKillerCells_Fishers_Table) <- c("Total Non-NaturalKillerCells","NaturalKillerCells")

fisher.test(NaturalKillerCells_Fishers_Table)

NaturalKillerCells_Fishers_Table
```


```{R}
HCnovel_Cluster14 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "novel_Cluster14"])
PDnovel_Cluster14 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "novel_Cluster14"])
novel_Cluster14_Fishers_allcells <- cbind(totalHC - HCnovel_Cluster14, totalPD - PDnovel_Cluster14)
novel_Cluster14_Fishers_novel_Cluster14_Neurons <- cbind(HCnovel_Cluster14, PDnovel_Cluster14)

novel_Cluster14_Fishers_Table <- rbind(novel_Cluster14_Fishers_allcells, novel_Cluster14_Fishers_novel_Cluster14_Neurons)

colnames(novel_Cluster14_Fishers_Table) <- c("HC","PD")
rownames(novel_Cluster14_Fishers_Table) <- c("Total Non-novel_Cluster14","novel_Cluster14")

fisher.test(novel_Cluster14_Fishers_Table)

novel_Cluster14_Fishers_Table
```

```{R}
HCnovel_Cluster16 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "novel_Cluster16"])
PDnovel_Cluster16 <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "novel_Cluster16"])
novel_Cluster16_Fishers_allcells <- cbind(totalHC - HCnovel_Cluster16, totalPD - PDnovel_Cluster16)
novel_Cluster16_Fishers_novel_Cluster16_Neurons <- cbind(HCnovel_Cluster16, PDnovel_Cluster16)

novel_Cluster16_Fishers_Table <- rbind(novel_Cluster16_Fishers_allcells, novel_Cluster16_Fishers_novel_Cluster16_Neurons)

colnames(novel_Cluster16_Fishers_Table) <- c("HC","PD")
rownames(novel_Cluster16_Fishers_Table) <- c("Total Non-novel_Cluster16","novel_Cluster16")

fisher.test(novel_Cluster16_Fishers_Table)

novel_Cluster16_Fishers_Table
```

```{R}
HCGABA_Neurons <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "HC" & cell_type_number_df$Var1 == "GABA Neurons"])
PDGABA_Neurons <- sum(cell_type_number_df$Freq[cell_type_number_df$Var2 == "PD" & cell_type_number_df$Var1 == "GABA Neurons"])
GABA_Neurons_Fishers_allcells <- cbind(totalHC - HCGABA_Neurons, totalPD - PDGABA_Neurons)
GABA_Neurons_Fishers_GABA_Neurons_Neurons <- cbind(HCGABA_Neurons, PDGABA_Neurons)

GABA_Neurons_Fishers_Table <- rbind(GABA_Neurons_Fishers_allcells, GABA_Neurons_Fishers_GABA_Neurons_Neurons)

colnames(GABA_Neurons_Fishers_Table) <- c("HC","PD")
rownames(GABA_Neurons_Fishers_Table) <- c("Total Non-GABA Neurons","GABA Neurons")

fisher.test(GABA_Neurons_Fishers_Table)

GABA_Neurons_Fishers_Table
```


Instead of Fishers test, opting for a Wilcoxon test approach where each data point is the number of cells in a sample.


```{R}
sampleframe <- table(AllMB$sample_ID,AllMB$case)
sampleframe <- as.data.frame(sampleframe)
HealthyControls <- as.vector(sampleframe$Var1[which(sampleframe$Var2 == "HC" & sampleframe$Freq > 0)])
ParkinsonsPatients <- as.vector(sampleframe$Var1[which(sampleframe$Var2 == "PD" & sampleframe$Freq > 0)])
```

```{R}
table(Idents(AllMB), AllMB$sample_ID)
celltable2 <- table(Idents(AllMB), AllMB$sample_ID)
cell_type_number_df2 <- as.data.frame(celltable2, stringsAsFactors=FALSE)

cell_type_number_df2
```


```{R}
library(dplyr)
library(tidyr)

cell_type_number_df2 <- group_by(cell_type_number_df2, Var2) %>% mutate(sampletotal = sum(Freq)) 

cell_type_number_df2$cellfraction <- cell_type_number_df2$Freq/cell_type_number_df2$sampletotal
```

```{R}
cell_type_number_df2$status <- "HC"
cell_type_number_df2$status[cell_type_number_df2$Var2 %in% ParkinsonsPatients] <- "PD"
cell_type_number_df2_ordered <- cell_type_number_df2[order(cell_type_number_df2$status),]
```

creating an excel table from cell_type_number_df2


```{R}
samplesheet = NULL
samplesheet <- data.frame(row.names = unique(cell_type_number_df2_ordered$Var1))
for (i in unique(cell_type_number_df2_ordered$Var2)) {
        tempdf1 <- cell_type_number_df2_ordered[cell_type_number_df2_ordered$Var2 == i,]
        tempdf2 <- as.data.frame(tempdf1$cellfraction)
        rownames(tempdf2) <- tempdf1$Var1
        colnames(tempdf2) <- paste(unique(tempdf1$status),unique(tempdf1$Var2),"Fraction",sep="_")
        samplesheet <- as.data.frame(cbind(samplesheet,tempdf2))
}

#write.table(samplesheet, file = "Files/Batch4_Cell_Cluster_Fractions.csv",quote = FALSE, sep = ",",row.names = TRUE,col.names = NA)
```



```{R}
samplesheetrawnumbers = NULL
samplesheetrawnumbers <- data.frame(row.names = unique(cell_type_number_df2_ordered$Var1))
for (i in unique(cell_type_number_df2_ordered$Var2)) {
        tempdf1 <- cell_type_number_df2_ordered[cell_type_number_df2_ordered$Var2 == i,]
        tempdf2 <- as.data.frame(tempdf1$Freq)
        rownames(tempdf2) <- tempdf1$Var1
        colnames(tempdf2) <- paste(unique(tempdf1$status),unique(tempdf1$Var2),"Count",sep="_")
        samplesheetrawnumbers <- as.data.frame(cbind(samplesheetrawnumbers,tempdf2))
}

#write.table(samplesheetrawnumbers, file = "Files/Batch4_Cell_Cluster_Counts.csv",quote = FALSE, sep = ",",row.names = TRUE,col.names = NA)
```





```{R}
library(reshape2)
melted_cell_type_number_df2 <- melt(cell_type_number_df2)
```

```{R}
melted_cell_type_number_df2 <- data.frame(lapply(melted_cell_type_number_df2, as.vector), stringsAsFactors=FALSE)

melted_cell_type_number_df2_fractions <- melted_cell_type_number_df2[melted_cell_type_number_df2$variable=="cellfraction",]

melted_cell_type_number_df2_rawcounts <- melted_cell_type_number_df2[melted_cell_type_number_df2$variable=="Freq",]
```




```{R}
melted_cell_type_number_df2_fractions$variable[which(melted_cell_type_number_df2_fractions$Var2 %in% HealthyControls)] <- "Healthy_Control"

melted_cell_type_number_df2_rawcounts$variable[which(melted_cell_type_number_df2_rawcounts$Var2 %in% HealthyControls)] <- "Healthy_Control"
```




```{R}
melted_cell_type_number_df2_fractions$variable[which(melted_cell_type_number_df2_fractions$Var2 %in% ParkinsonsPatients)] <- "Parkinsons_Patient"

melted_cell_type_number_df2_rawcounts$variable[which(melted_cell_type_number_df2_rawcounts$Var2 %in% ParkinsonsPatients)] <- "Parkinsons_Patient"
```


```{R}
#lapply(unique(melted_cell_type_number_df2$Var1), wilcox.test, y=)

wilcoxfun <- function (x,y) {
        tempdf <- y[y$Var1 == x,]
        wilcox.test(tempdf$value~tempdf$variable)
}

ttestfun <- function (x,y) {
        tempdf <- y[y$Var1 == x,]
        t.test(tempdf$value~tempdf$variable)
}

wilcoxfun("Endothelial cells")

  #lapply(kidney[,c("time", "age", "frail")], function(x) wilcox.test(x ~ kidney$sex, var.equal = TRUE))

ttestfun("Endothelial cells")
```

lapply for custom wilcox test function

```{R}
unique(melted_cell_type_number_df2_fractions$Var1)
```


```{R}
lapply(unique(melted_cell_type_number_df2_fractions$Var1), y=melted_cell_type_number_df2_fractions, wilcoxfun)
```

No, significant results, likely due to there not being enough data points, something to return to.

Using MAST (recommended by "best practices" paper) to look for marker genes/genes DE between the two DA neuron subtypes


repeating for t-test

```{R}
lapply(unique(melted_cell_type_number_df2_fractions$Var1),y=melted_cell_type_number_df2_fractions, ttestfun)
```


Repeating t-test for raw cluster counts instead of fractions


```{R}
lapply(unique(melted_cell_type_number_df2_rawcounts$Var1),y=melted_cell_type_number_df2_rawcounts, ttestfun)
```



```{R}
library(MAST)

DA1vsDA2DEGenes <- FindMarkers(AllMB, ident.1 = "Dopaminergic Neurons group 1", ident.2 = "Dopaminergic Neurons group 2", test.use = "MAST")

```


We get 2/3rds of the non-filtered genes ending up as differentially expressed... the basic assumptions underlying DE analyses fall apart, probably not reliable, "Best practices" paper does warn of overinflated p-values. Again, maybe something to come back to later?

```{R}
DA1vsDA2DEGenesPvalFilter <- DA1vsDA2DEGenes[DA1vsDA2DEGenes$p_val_adj <= 0.05,]

```




```{R}

head(DA1vsDA2DEGenesPvalFilter,10)



```

```{R}
VlnPlot(AllMB, features = c("MALAT1"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

```

```{R}
VlnPlot(AllMB, features = c("ATP1B1"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

```


```{R}

VlnPlot(AllMB, features = c("HSP90AA1"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))
```
+ logFCs mean the gene is more highly expressed in DA1 neurons, and - logFCs the opposite

We are getting alot of genes that appear to be DE between DA neurons and all the rest of the genes. This could be interesting in itself as these "global differences" could explain why DA2 neurons do not appear to be significantly lost in PD patients whereas other types of neuron are. However, we are also interested in looking at genes specifically DE between DA1 and DA2 neurons. Therefore we will run DE analyses for DA1 and DA2 vs all, combine this genelist and remove them from the DA1 vs DA2 analysis. 


```{R}

DA1vsALLGenes <- FindMarkers(AllMB, ident.1 = "Dopaminergic Neurons group 1", test.use = "MAST")


DA1vsALLGenesPvalFilter <- DA1vsALLGenes[DA1vsALLGenes$p_val_adj <= 0.05,]

DA1vsALLGenesPvalFilterGenes <- rownames(DA1vsALLGenesPvalFilter)
```


```{R}

DA2vsALLGenes <- FindMarkers(AllMB, ident.1 = "Dopaminergic Neurons group 2", test.use = "MAST")


DA2vsALLGenesPvalFilter <- DA2vsALLGenes[DA2vsALLGenes$p_val_adj <= 0.05,]


DA2vsALLGenesPvalFilterGenes <- rownames(DA2vsALLGenesPvalFilter)
```


```{R}
DAneuronsvsALLDEgenes <- unique(c(DA1vsALLGenesPvalFilterGenes,DA2vsALLGenesPvalFilterGenes))



```




```{R}
DA1vsDA2DEGenesPvalFiltervsALLfilter <- DA1vsDA2DEGenesPvalFilter[-which(rownames(DA1vsDA2DEGenesPvalFilter) %in% DAneuronsvsALLDEgenes),]


```



```{R}
head(DA1vsDA2DEGenesPvalFiltervsALLfilter,50)


```

```{R}

VlnPlot(AllMB, features = c("NRG3"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))
```




```{R}

VlnPlot(AllMB, features = c("MGAT3"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))
```



```{R}

VlnPlot(AllMB, features = c("DAG1"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))
```



```{R}

VlnPlot(AllMB, features = c("IPO7"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))
```



```{R}

VlnPlot(AllMB, features = c("TDP2"),pt.size = 0) +
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))
```


Some of these results look very odd, it seems like the DE algorithm only needs the gene to be expressed in a minority of cells at a high level to call differential expression...... Maybe this is OK for defining clusters in conjunction with violin plots but for a more robust analysis maybe create raw counts matrix where columns are the cluster that cell was assigned to. Maybe also include the sample and batch at a later time?


Writing raw counts matrix to csv file in case we want to explore different avenues for differential expression analysis

#```{R}

RawCountMatrix <- GetAssayData(object = AllMB, slot = "counts")
#colnames(RawCountMatrix)

colnames(RawCountMatrix) <- make.unique(as.vector(AllMB@active.ident))
#```


```{R}

write.table(as.matrix(RawCountMatrix), 
            'Files/RawCountsMatrix.csv', 
            sep = ',', row.names = T, col.names = T, quote = F)

```



######################################################################################################################


Making table of cell type numbers


```{R}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
library(DOSE)
library(GOSemSim)
library(enrichplot)

```

```{R}
AllMB <- readRDS("Pilot1_and2_Integration_Files/Case_Corrected_Pilot1and2_MB_Unassigned.rds")
```


```{r}
AllMB <- RenameIdents(AllMB, `1` = "Oligodendrocytes", `2` = "Astrocytes",
                      `3` = "Oligodendrocytes", `4` = "GABA Neurons", `5` = "Microglia",
                      `6` = "OPCs", `7` = "Endothelial", `8` = "DA Neurons",`9` = "Endothelial",
                      `10` = "GLU Neurons", `11` = "Oligodendrocytes",`12` = "GLU Neurons",
                      `13` = "DA Neurons",`14` = "Other",
                      `15` = "CD8+ T Cells", `16`="Other", `17`="GABA Neurons", `18`="GLU Neurons")
```



```{R}

CellTypeNumbers <- as.data.frame(table(Idents(AllMB)))

colnames(CellTypeNumbers) <- c("Cell_Type","Frequency")

```


```{R}

write.csv(CellTypeNumbers, file = "Pilot1_and2_Integration_Files/Midbrain_CellType_Numbers.csv", quote = FALSE, row.names = FALSE)

```


