---
title: "Seurat Workflow and Cluster Assignment for Pilot_1_and_2"
output: html_document
---

```{R}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(SeuratWrappers)
library(glmpca)
library(stringr)
library(RColorBrewer)
```

```{R}

Pilot1and2_MB_Microglia <- readRDS("Files/Pilot1and2_MB_Microglia")

```



 Quality Checking : 

 Seurat allows you to easily explore QC metrics and filter cells based on any user-defined criteria.
 A few QC metrics commonly used by the community include : 
 The number of unique genes detected in each cell.
 Low-quality cells or empty droplets will often have very few genes
 Cell doublets or multiplets may exhibit an aberrantly high gene count
 Similarly, the total number of molecules detected within a cell (correlates strongly with unique genes)
 The percentage of reads that map to the mitochondrial genome
 Low-quality / dying cells often exhibit extensive mitochondrial contamination
 We calculate mitochondrial QC metrics with the PercentageFeatureSet() function, which calculates the percentage of counts originating from a set of features
 We use the set of all genes starting with MT- as a set of mitochondrial genes

```{r}
Pilot1and2_MB_Microglia[["percent.mt"]] <- PercentageFeatureSet(Pilot1and2_MB_Microglia, pattern = "^MT-")
VlnPlot(Pilot1and2_MB_Microglia, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```

 An object of class Seurat 
 25381 features across 6549 samples within 1 assay 
 Active assay: RNA (25381 features)
 
 
```{R}
library(stats)
nfeature_RNA <- Pilot1and2_MB_Microglia@meta.data$nFeature_RNA
mean(nfeature_RNA)
MAD <- mad(nfeature_RNA, center = median(nfeature_RNA))
threeMAD <- (MAD*3)+median(Pilot1and2_MB_Microglia@meta.data$nFeature_RNA)

```

```{R}

dim(Pilot1and2_MB_Microglia)
```


Filtering low quality cells:
```{r}
Pilot1and2_MB_Microglia <- subset(Pilot1and2_MB_Microglia, subset = nFeature_RNA > 200 & nFeature_RNA < threeMAD | percent.mt < 5)
```


```{R}
dim(Pilot1and2_MB_Microglia)


```


Log Normalizing data: 
```{r}
Pilot1and2_MB_Microglia <- NormalizeData(Pilot1and2_MB_Microglia, normalization.method = "LogNormalize", scale.factor = 10000)
```


Finding and plotting 2000 most variable features

```{R}
Pilot1and2_MB_Microglia <- FindVariableFeatures(Pilot1and2_MB_Microglia, selection.method = "vst", nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(Pilot1and2_MB_Microglia), 10)

# plot variable features with and without labels
plot1 <- VariableFeaturePlot(Pilot1and2_MB_Microglia)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot1 + plot2
plot1
```

```{R}
plot2
```

Run the standard workflow for visualization and clustering :

save list of all genes
```{r}
all.genes <- rownames(Pilot1and2_MB_Microglia)
```

center and scale data
```{r}
Pilot1and2_MB_Microglia <- ScaleData(Pilot1and2_MB_Microglia, features = all.genes, verbose = FALSE)
```

finding the top 30 principal components for cells
```{r}
Pilot1and2_MB_Microglia <- RunGLMPCA(Pilot1and2_MB_Microglia, features=Pilot1and2_MB_Microglia@assays$RNA@var.features, L = 30)
```

see contribution of genes to construct each of these principal components.
```{r}
VizDimLoadings(Pilot1and2_MB_Microglia, dims = 1:2, reduction = "glmpca")
```


plot cells in the first two principal components colored by case: 
```{r}
DimPlot(object = Pilot1and2_MB_Microglia, reduction = "glmpca", pt.size = .1, group.by = "case")
```



Run Harmony
```{r}
Pilot1and2_MB_Microglia <- RunHarmony(Pilot1and2_MB_Microglia, group.by.vars = c("batch"), plot_convergence = TRUE, reduction = "glmpca", theta = c(1))
```




cells in harmony axis
```{r}
DimPlot(object = Pilot_1_MB, reduction = "harmony", pt.size = .1, group.by = "case")
```
cells in harmony 1 axis
```{r}
VlnPlot(object = Pilot_1_MB, features = "harmony_1", group.by = "case",  pt.size = .1)
```


```{R}

ElbowPlot(Pilot1and2_MB_Microglia, reduction = "harmony",ndims = 30)

```


```{R}

ElbowPlot(Pilot1and2_MB_Microglia, reduction = "glmpca",ndims = 30)

```

 An object of class Seurat 
 25381 features across 6549 samples within 1 assay 
 Active assay: RNA (25381 features)
 2 dimensional reductions calculated: pca, harmony

Finding Clusters of cells:
```{r}
Pilot1and2_MB_Microglia <- FindNeighbors(Pilot1and2_MB_Microglia, reduction = "harmony", dims = 1:20)
Pilot1and2_MB_Microglia <- FindClusters(Pilot1and2_MB_Microglia, resolution = 0.5, algorithm = 4)
```

run Umap based on top 20 harmony axis: 

```{r}
Pilot1and2_MB_Microglia <- RunUMAP(Pilot1and2_MB_Microglia, reduction = "harmony", dims = 1:20)
```

run tsne based on top 20 harmony axis: 

```{r}
#AllMB <- RunTSNE(AllMB, reduction = "harmony", dims = 1:15)
```

plot umap: 
```{r}
DimPlot(Pilot1and2_MB_Microglia, reduction = "umap", label = TRUE,pt.size = 0.01)
```

```{r}
DimPlot(Pilot1and2_MB_Microglia, reduction = "umap", group.by = "case",pt.size = 0.1)
```


TSNE Plot cells colored by clusters and grouped by case. 
```{r}
DimPlot(Pilot1and2_MB_Microglia, reduction = "umap", split.by = "case", label = TRUE, ncol = 1)
```

```{r}
DimPlot(Pilot1and2_MB_Microglia, label = TRUE, repel = TRUE, pt.size = 1, label.size = 3)

UMAPclusters <- DimPlot(Pilot1and2_MB_Microglia, label = TRUE, repel = TRUE, pt.size = 1, label.size = 3) + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Pilot1_and2_Integration_Figures/Unassigned_UMAPclusters_scRNA_seq_Pilot_2and1_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")
```


```{R}
UMAPclusters <- DimPlot(Pilot1and2_MB_Microglia, label = TRUE, repel = TRUE, pt.size = 1, label.size = 3, split.by = "case") + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Pilot1_and2_Integration_Figures/Unassigned_Split_UMAPclusters_scRNA_seq_Pilot_2and1_Microglia.pdf", device = "pdf", width = 12, height = 4, units = "in")
```


```{r}
Pilot1and2_MB_Microglia.markers <- FindAllMarkers(Pilot1and2_MB_Microglia, only.pos = TRUE, min.pct = 0.1, logfc.threshold = 0.25)
```



```{R}
test <- Pilot1and2_MB_Microglia.markers

Activated_Microglia <- c("IL1B","APOE","CLEC7A",)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CLEC7A"),pt.size = 0)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("APOE"),pt.size = 0)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("RETNLB"),pt.size = 1)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CD68"),pt.size = 1)
```
activated  microglia markers

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CLEC7A"),pt.size = 0)
```


M1

CD86 Higher expression

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CD86"),pt.size = 0)
```

CD16

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("FCGR3A"),pt.size = 0)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CD40"),pt.size = 0)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CXCL1"),pt.size = 0)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CXCL9"),pt.size = 0)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CXCL9"),pt.size = 0)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CCL5"),pt.size = 0)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("TNF"),pt.size = 0)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("IFNG"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("IL1B"),pt.size = 1)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("IL6"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("FCGR2A"),pt.size = 0)
```




M2 Markers - anti-inflammatory

CD206

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("MRC1"),pt.size = 0)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CCL2"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CCL24"),pt.size = 1)
```




```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CD163"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CCL24"),pt.size = 1)
```




```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("TGFB1"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CD14"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("FCGR3A"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("TREM2"),pt.size = 1, split.by = "case")
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("IGF1"),pt.size = 1)
```


Markers of "Dark Microglia"

CD11b positive

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("ITGAM"),pt.size = 1)
```

P2RY12 downregulated

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("P2RY12"),pt.size = 0)
```

IBA1/AIF1 downregulated

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("AIF1"),pt.size = 1)
```

CX3CR1 - downregulated

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CX3CR1"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("MRC1"),pt.size = 1)
```
CD206

```{r}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("MRC1"), split.by = "case")
```

MHCII

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("HLA-DRA"),pt.size = 0)
```

```{R}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("HLA-DRA"))



```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("ITGAX"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CD14"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("FCGR3A"),pt.size = 1)
```


```{R}

VlnPlot(Pilot1and2_MB_Microglia, features = c("HLA-DRB1"),pt.size = 1)

```



```{R}

VlnPlot(Pilot1and2_MB_Microglia, features = c("TLR4"),pt.size = 1)

```


```{R}

VlnPlot(Pilot1and2_MB_Microglia, features = c("CD8A"),pt.size = 1)

```


NLRP3 - marker of infiltrating macrophages

```{R}

VlnPlot(Pilot1and2_MB_Microglia, features = c("SIGLEC1"),pt.size = 1, split.by = "case")

SIGLEC1_VlnPlot_Microglia <- VlnPlot(Pilot1and2_MB_Microglia, features = c("SIGLEC1"),pt.size = 0)

ggsave(SIGLEC1_VlnPlot_Microglia, filename = "Pilot1_and2_Integration_Figures/SIGLEC1_VlnPlot_Microglia.pdf", device = "pdf", width = 4, height = 4, units = "in")

```
 
 Present only on microglia, absent from infiltrating cells

```{R}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("SIGLEC1"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

SIGLEC1_FeaturePlot_Microglia <- FeaturePlot(Pilot1and2_MB_Microglia, features = c("SIGLEC1"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

ggsave(SIGLEC1_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/SIGLEC1_FeaturePlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```

low in Microglia (except M2)

```{r}

VlnPlot(Pilot1and2_MB_Microglia, features = c("CD163"),pt.size = 0)

CD63_VlnPlot_Microglia <- VlnPlot(Pilot1and2_MB_Microglia, features = c("CD163"),pt.size = 0)

ggsave(CD63_VlnPlot_Microglia, filename = "Pilot1_and2_Integration_Figures/CD163_VlnPlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```



```{R}

FeaturePlot(Pilot1and2_MB_Microglia, features = c("CD163"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

CD163_FeaturePlot_Microglia <- FeaturePlot(Pilot1and2_MB_Microglia, features = c("CD163"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

ggsave(CD163_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/CD163_FeaturePlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")


```



```{r}

VlnPlot(Pilot1and2_MB_Microglia, features = c("CD163"),pt.size = 0)

CD63_FeaturePlot_Microglia <- VlnPlot(Pilot1and2_MB_Microglia, features = c("CD163"),pt.size = 0)

ggsave(CD63_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/CD163_VlnPlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```



```{R}

FeaturePlot(Pilot1and2_MB_Microglia, features = c("CD163"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

CD163_FeaturePlot_Microglia <- FeaturePlot(Pilot1and2_MB_Microglia, features = c("CD163"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

ggsave(CD163_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/CD163_FeaturePlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```

low in Micrglia (except M2)

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("MRC1"),pt.size = 0)
```

```{r}

VlnPlot(Pilot1and2_MB_Microglia, features = c("MRC1"),pt.size = 0)

MRC1_FeaturePlot_Microglia <- VlnPlot(Pilot1and2_MB_Microglia, features = c("MRC1"),pt.size = 0)

ggsave(MRC1_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/MRC1_VlnPlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```



```{R}

FeaturePlot(Pilot1and2_MB_Microglia, features = c("MRC1"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

MRC1_FeaturePlot_Microglia <- FeaturePlot(Pilot1and2_MB_Microglia, features = c("MRC1"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

ggsave(MRC1_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/MRC1_FeaturePlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```


```{r}

VlnPlot(Pilot1and2_MB_Microglia, features = c("P2RY12"),pt.size = 0)

P2RY12_FeaturePlot_Microglia <- VlnPlot(Pilot1and2_MB_Microglia, features = c("P2RY12"),pt.size = 0)

ggsave(P2RY12_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/P2RY12_VlnPlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```



```{R}

FeaturePlot(Pilot1and2_MB_Microglia, features = c("P2RY12"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

P2RY12_FeaturePlot_Microglia <- FeaturePlot(Pilot1and2_MB_Microglia, features = c("P2RY12"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

ggsave(P2RY12_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/P2RY12_FeaturePlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```
CX3CR1

```{r}

VlnPlot(Pilot1and2_MB_Microglia, features = c("CX3CR1"),pt.size = 0)

CX3CR1_FeaturePlot_Microglia <- VlnPlot(Pilot1and2_MB_Microglia, features = c("CX3CR1"),pt.size = 0)

ggsave(CX3CR1_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/CX3CR1_VlnPlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```



```{R}

FeaturePlot(Pilot1and2_MB_Microglia, features = c("CX3CR1"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

CX3CR1_FeaturePlot_Microglia <- FeaturePlot(Pilot1and2_MB_Microglia, features = c("CX3CR1"), split.by = "case") + scale_colour_gradientn(colours = c("lightgrey",brewer.pal(n = 20, name = "YlOrRd")))

ggsave(CX3CR1_FeaturePlot_Microglia, filename = "Pilot1_and2_Integration_Figures/CX3CR1_FeaturePlot_Microglia.pdf", device = "pdf", width = 6, height = 4, units = "in")

```

```{R}

sum(Pilot1and2_MB_Microglia@meta.data$case %in% "HC")

sum(Pilot1and2_MB_Microglia@meta.data$case %in% "PD")


```


Trying to characterise cluster 3


```{R}

Pilot1and2_MB_Microglia.markers_cluster3 <- Pilot1and2_MB_Microglia.markers[Pilot1and2_MB_Microglia.markers$cluster == 3,]

```




```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("CD22"),pt.size = 0)
```

```{r}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("CD22"),pt.size = 0)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("TLR2"),pt.size = 1)
```

NK cell markers

CD56

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("NCAM1"),pt.size = 1)
```

```{r}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("NCAM1"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("NCAM2"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("DLG2"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("DLG1"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("QDPR"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("NFIX"),pt.size = 1)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("PLP1"),pt.size = 1)
```

```{r}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("PLP1"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("MBP"),pt.size = 1)
```


```{r}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("MBP"),pt.size = 1)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("IGF1R"),pt.size = 1)
```
```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("S100B"),pt.size = 1)
```


```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("BCAS1"),pt.size = 1, split.by = "case")
```

```{r}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("BCAS1"),pt.size = 1)
```




```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("GRIA4"),pt.size = 1)
```

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("ITIH3"),pt.size = 1)
```




```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("PROX1"),pt.size = 1)
```





```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("MOG"),pt.size = 1)
```


```{r}
FeaturePlot(Pilot1and2_MB_Microglia, features = c("MOG"),pt.size = 1)
```

CLUSTER 3 LIKELY "EARLY MYELINATING OLIGODENDROCYTES" OR SOME KIND OF IMMUNE-RELATED OLIGODENDROCYTE? - Or infiltrating schwann cells - these do express NCAM

```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("MAG"),pt.size = 1)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("SOX10"),pt.size = 1)
```



```{r}
VlnPlot(Pilot1and2_MB_Microglia, features = c("DHH"),pt.size = 1)
```