---
title: "Seurat Workflow and Cluster Assignment for Pilot_1_and_2"
output: html_document
---

```{R}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(SeuratWrappers)
library(glmpca)
library(stringr)
```

```{R}

Pilot1and2_MB_Astrocytes <- readRDS("Files/Pilot1and2_MB_Astrocytes")

```



 Quality Checking : 

 Seurat allows you to easily explore QC metrics and filter cells based on any user-defined criteria.
 A few QC metrics commonly used by the community include : 
 The number of unique genes detected in each cell.
 Low-quality cells or empty droplets will often have very few genes
 Cell doublets or multiplets may exhibit an aberrantly high gene count
 Similarly, the total number of molecules detected within a cell (correlates strongly with unique genes)
 The percentage of reads that map to the mitochondrial genome
 Low-quality / dying cells often exhibit extensive mitochondrial contamination
 We calculate mitochondrial QC metrics with the PercentageFeatureSet() function, which calculates the percentage of counts originating from a set of features
 We use the set of all genes starting with MT- as a set of mitochondrial genes

```{r}
Pilot1and2_MB_Astrocytes[["percent.mt"]] <- PercentageFeatureSet(Pilot1and2_MB_Astrocytes, pattern = "^MT-")
VlnPlot(Pilot1and2_MB_Astrocytes, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```

 An object of class Seurat 
 25381 features across 6549 samples within 1 assay 
 Active assay: RNA (25381 features)
 
 
```{R}
library(stats)
nfeature_RNA <- Pilot1and2_MB_Astrocytes@meta.data$nFeature_RNA
mean(nfeature_RNA)
MAD <- mad(nfeature_RNA, center = median(nfeature_RNA))
threeMAD <- (MAD*3)+median(Pilot1and2_MB_Astrocytes@meta.data$nFeature_RNA)

```

```{R}

dim(Pilot1and2_MB_Astrocytes)
```


Filtering low quality cells:
```{r}
Pilot1and2_MB_Astrocytes <- subset(Pilot1and2_MB_Astrocytes, subset = nFeature_RNA > 200 & nFeature_RNA < threeMAD | percent.mt < 5)
```


```{R}
dim(Pilot1and2_MB_Astrocytes)


```


Log Normalizing data: 
```{r}
Pilot1and2_MB_Astrocytes <- NormalizeData(Pilot1and2_MB_Astrocytes, normalization.method = "LogNormalize", scale.factor = 10000)
```


Finding and plotting 2000 most variable features

```{R}
Pilot1and2_MB_Astrocytes <- FindVariableFeatures(Pilot1and2_MB_Astrocytes, selection.method = "vst", nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(Pilot1and2_MB_Astrocytes), 10)

# plot variable features with and without labels
plot1 <- VariableFeaturePlot(Pilot1and2_MB_Astrocytes)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot1 + plot2
plot1
```

```{R}
plot2
```

Run the standard workflow for visualization and clustering :

save list of all genes
```{r}
all.genes <- rownames(Pilot1and2_MB_Astrocytes)
```

center and scale data
```{r}
Pilot1and2_MB_Astrocytes <- ScaleData(Pilot1and2_MB_Astrocytes, features = all.genes, verbose = FALSE)
```

finding the top 30 principal components for cells
```{r}
Pilot1and2_MB_Astrocytes <- RunGLMPCA(Pilot1and2_MB_Astrocytes, features=Pilot1and2_MB_Astrocytes@assays$RNA@var.features, L = 30)
```

see contribution of genes to construct each of these principal components.
```{r}
VizDimLoadings(Pilot1and2_MB_Astrocytes, dims = 1:2, reduction = "glmpca")
```


plot cells in the first two principal components colored by case: 
```{r}
DimPlot(object = Pilot1and2_MB_Astrocytes, reduction = "glmpca", pt.size = .1, group.by = "case")
```

Coordinate of cells in PC 1 characterized by case: 
```{r}
VlnPlot(object = Pilot1and2_MB_Astrocytes, features = "PC_1", group.by = "case",  pt.size = .1)
```

Run Harmony
```{r}
Pilot1and2_MB_Astrocytes <- RunHarmony(Pilot1and2_MB_Astrocytes, group.by.vars = c("batch"), plot_convergence = TRUE, reduction = "glmpca", theta = c(1))
```


cells in harmony axis
```{r}
harmony_embeddings <- Embeddings(AllMB, 'harmony')
harmony_embeddings[1:5, 1:5]
```

cells in harmony axis
```{r}
DimPlot(object = Pilot_1_MB, reduction = "harmony", pt.size = .1, group.by = "case")
```
cells in harmony 1 axis
```{r}
VlnPlot(object = Pilot_1_MB, features = "harmony_1", group.by = "case",  pt.size = .1)
```


```{R}

ElbowPlot(Pilot1and2_MB_Astrocytes, reduction = "harmony",ndims = 30)

```


```{R}

ElbowPlot(Pilot1and2_MB_Astrocytes, reduction = "glmpca",ndims = 30)

```

 An object of class Seurat 
 25381 features across 6549 samples within 1 assay 
 Active assay: RNA (25381 features)
 2 dimensional reductions calculated: pca, harmony

Finding Clusters of cells:
```{r}
Pilot1and2_MB_Astrocytes <- FindNeighbors(Pilot1and2_MB_Astrocytes, reduction = "harmony", dims = 1:30)
Pilot1and2_MB_Astrocytes <- FindClusters(Pilot1and2_MB_Astrocytes, resolution = 0.5, algorithm = 4)
```

run Umap based on top 20 harmony axis: 

```{r}
Pilot1and2_MB_Astrocytes <- RunUMAP(Pilot1and2_MB_Astrocytes, reduction = "harmony", dims = 1:30)
```

run tsne based on top 20 harmony axis: 

```{r}
#AllMB <- RunTSNE(AllMB, reduction = "harmony", dims = 1:15)
```

plot umap: 
```{r}
DimPlot(Pilot1and2_MB_Astrocytes, reduction = "umap", label = TRUE,pt.size = 0.01)
```

```{r}
DimPlot(Pilot1and2_MB_Astrocytes, reduction = "umap", group.by = "case",pt.size = 0.1)
```


TSNE Plot cells colored by clusters and grouped by case. 
```{r}
DimPlot(Pilot1and2_MB_Astrocytes, reduction = "umap", split.by = "case", label = TRUE, ncol = 1)
```

```{r}
DimPlot(Pilot1and2_MB_Astrocytes, label = TRUE, repel = TRUE, pt.size = 0, label.size = 2)

UMAPclusters <- DimPlot(Pilot1and2_MB_Astrocytes, label = TRUE, repel = TRUE, pt.size = 0, label.size = 3) + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Pilot1_and2_Integration_Figures/Unassigned_UMAPclusters_scRNA_seq_Pilot_2and1_Astrocytes.pdf", device = "pdf", width = 6, height = 4, units = "in")
```


```{R}
UMAPclusters <- DimPlot(Pilot1and2_MB_Astrocytes, label = TRUE, repel = TRUE, pt.size = 0, label.size = 3, split.by = "case") + 
        theme(axis.text = element_text(size=8),
              axis.title = element_text(size = 12),
              legend.text = element_text(size = 8),
              title = element_text(size = 12),
              legend.key.size = unit(0.4,"cm"))

ggsave(UMAPclusters, filename = "Pilot1_and2_Integration_Figures/Unassigned_Split_UMAPclusters_scRNA_seq_Pilot_2and1_Astrocytes.pdf", device = "pdf", width = 12, height = 4, units = "in")
```











