---
title: "Using DE Gene Tables to Construct a Heatmap"
output: html_document
---

New heatmap - top 10 DE genes per subcluster, DE in at least two subclusters, no threshold

Using top genes DE for each subcluster

```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```






#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")




Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons_1", `3` = "GLU_Neurons_2",
                      `4` = "GLU_Neurons_3",`5` = "Astrocytes_1",
                      `6` = "GABA_Neurons_1", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons_4",`9` = "GLU_Neurons_5",
                      `10` = "GLU_Neurons_6", `11` = "Microglia",`12` = "Astrocytes_2",
                      `13` = "OPCs",`14` = "GABA_Neurons_2",
                      `15` = "GLU_Neurons_7", `16`="GLU_Neurons_8", `17`="Endothelial_1", `18`="Endothelial_2",`19`="GABA_Neurons_3",`20`="GLU_Neurons_9",`21`="GLU_Neurons_10",`22`="GLU_Neurons_11",`23`="GABA_Neurons_4")




```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/AllGenes_|.csv","",i),read.csv(i))
}

```



```{R}
DEdfList <- ls()[grep("HC_vs_PD",ls())]

#DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]

#  tempdf <- tempdf[1:10,]

TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  tempdf <- tempdf[tempdf$p_val <= tempdf$BF_p_val,]
  if (nrow(tempdf) >= 10){
  tempdf <- tempdf[1:10,]
  tempdf$CellType <- gsub("_Markers_HC_vs_PD","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
  }
}

TopDEGenes <- TopDEGenes[duplicated(TopDEGenes$gene),]

```


```{R}

library(gdata)

keep(TopDEGenes, sure = TRUE)

```


```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/|.csv","",i),read.csv(i))
}

```

```{R}

DEdfListAllGenes<- ls()[grep("AllGenes",ls())]

DEdfListAllGenes <- c(DEdfListAllGenes[13],DEdfListAllGenes[16],DEdfListAllGenes[17],DEdfListAllGenes[18],DEdfListAllGenes[19],DEdfListAllGenes[20],DEdfListAllGenes[21],DEdfListAllGenes[22],DEdfListAllGenes[23],DEdfListAllGenes[14],DEdfListAllGenes[15],DEdfListAllGenes[12],DEdfListAllGenes[6],DEdfListAllGenes[7],DEdfListAllGenes[8],DEdfListAllGenes[9],DEdfListAllGenes[10],DEdfListAllGenes[11],DEdfListAllGenes[2],DEdfListAllGenes[27],DEdfListAllGenes[28],DEdfListAllGenes[29],DEdfListAllGenes[4],DEdfListAllGenes[5],DEdfListAllGenes[1],DEdfListAllGenes[26],DEdfListAllGenes[25],DEdfListAllGenes[24],DEdfListAllGenes[3])

```

```{R}

lfcframe=NULL
Thresholdframe=NULL
for (geneident in unique(TopDEGenes$gene)){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfListAllGenes){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_lfc <- tempdf[tempdf$gene %in% geneident,]$avg_log2FC
      temp_pval <- tempdf[tempdf$gene %in% geneident,]$p_val
      if (temp_lfc > 0 & temp_pval <= unique(tempdf$BF_p_val)){
        temp_threshold <- 1
      } 
      
      if (temp_lfc < 0 & temp_pval <= unique(tempdf$BF_p_val)) {
        temp_threshold <- -1
      }
      if (temp_pval >= unique(tempdf$BF_p_val)) {
        temp_threshold <- 0
      }
      generow <- cbind(generow, temp_lfc)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, 0)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 lfcframe <- rbind(lfcframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(lfcframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(lfcframe) <- unique(TopDEGenes$gene)

colnames(Thresholdframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(Thresholdframe) <- unique(TopDEGenes$gene)

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```


```{R}

pheatmap(lfcframe)

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

#ggsave(testmap,filename = "Files/HeatmapTest2.pdf", device = "pdf", height = 30, width = 6, units = "in")

```

scale_colour_gradient(low = "magenta", high = "yellow")

```{R}
werd <- colorRampPalette(c("magenta","#C228C8","#813384","#000000","#817E3E","#BEB72F","#FFF300"))(200)


testmap <- pheatmap(lfcframe, border_color = "gray", color = werd, cellwidth = 10, cellheight = 10, breaks = seq(from=-0.5,to=0.5, length.out=200), cluster_cols = FALSE)

ggsave(testmap,filename = "Files/HeatmapTest.pdf", device = "pdf", height = 30, width = 8, units = "in")

```


#############################################################################################################################

New heatmap - top 10 DE genes per subcluster, DE in at least two subclusters, with 0.1 pct and >= 0.25 LFC threshold

Using top genes DE for each subcluster

```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```






#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")




Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons_1", `3` = "GLU_Neurons_2",
                      `4` = "GLU_Neurons_3",`5` = "Astrocytes_1",
                      `6` = "GABA_Neurons_1", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons_4",`9` = "GLU_Neurons_5",
                      `10` = "GLU_Neurons_6", `11` = "Microglia",`12` = "Astrocytes_2",
                      `13` = "OPCs",`14` = "GABA_Neurons_2",
                      `15` = "GLU_Neurons_7", `16`="GLU_Neurons_8", `17`="Endothelial_1", `18`="Endothelial_2",`19`="GABA_Neurons_3",`20`="GLU_Neurons_9",`21`="GLU_Neurons_10",`22`="GLU_Neurons_11",`23`="GABA_Neurons_4")




```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "All_SigGenes.+PD_vs_HC.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/All_SigGenes_|.csv","",i),read.csv(i))
}

```



```{R}
DEdfList <- ls()[grep("PD_vs_HC",ls())]

#DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]

#  tempdf <- tempdf[1:10,]

TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  tempdf <- tempdf[tempdf$p_val <= tempdf$BF_p_val,]
  if (nrow(tempdf) >= 10){
  tempdf <- tempdf[1:10,]
  tempdf$CellType <- gsub("_Markers_PD_vs_HC","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
  }
}

TopDEGenes <- TopDEGenes[duplicated(TopDEGenes$gene),]

```


```{R}

library(gdata)

keep(TopDEGenes, sure = TRUE)

```


```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/|.csv","",i),read.csv(i))
}

```

```{R}

DEdfListAllGenes<- ls()[grep("AllGenes",ls())]

DEdfListAllGenes <- c(DEdfListAllGenes[12],DEdfListAllGenes[6],DEdfListAllGenes[7],DEdfListAllGenes[8],DEdfListAllGenes[9],DEdfListAllGenes[10],DEdfListAllGenes[11],DEdfListAllGenes[2],DEdfListAllGenes[27],DEdfListAllGenes[28],DEdfListAllGenes[29],DEdfListAllGenes[13],DEdfListAllGenes[16],DEdfListAllGenes[17],DEdfListAllGenes[18],DEdfListAllGenes[19],DEdfListAllGenes[20],DEdfListAllGenes[21],DEdfListAllGenes[22],DEdfListAllGenes[23],DEdfListAllGenes[14],DEdfListAllGenes[15],DEdfListAllGenes[3],DEdfListAllGenes[4],DEdfListAllGenes[5],DEdfListAllGenes[1],DEdfListAllGenes[26],DEdfListAllGenes[25],DEdfListAllGenes[24])

```

```{R}

lfcframe=NULL
Thresholdframe=NULL
for (geneident in unique(TopDEGenes$gene)){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfListAllGenes){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_lfc <- tempdf[tempdf$gene %in% geneident,]$avg_log2FC
      temp_pval <- tempdf[tempdf$gene %in% geneident,]$p_val
      if (temp_lfc > 0 & temp_pval <= unique(tempdf$BF_p_val)){
        temp_threshold <- 1
      } 
      
      if (temp_lfc < 0 & temp_pval <= unique(tempdf$BF_p_val)) {
        temp_threshold <- -1
      }
      if (temp_pval >= unique(tempdf$BF_p_val)) {
        temp_threshold <- 0
      }
      generow <- cbind(generow, temp_lfc)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, 0)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 lfcframe <- rbind(lfcframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(lfcframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
colnames(lfcframe) <- gsub("CD8\\+_T_Cells","Unknown_Cluster_IM",colnames(lfcframe))
rownames(lfcframe) <- unique(TopDEGenes$gene)

colnames(Thresholdframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
colnames(Thresholdframe) <- gsub("CD8\\+_T_Cells","Unknown_Cluster_IM",colnames(Thresholdframe))
rownames(Thresholdframe) <- unique(TopDEGenes$gene)

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```


```{R}

pheatmap(lfcframe)

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

#ggsave(testmap,filename = "Files/HeatmapTest2.pdf", device = "pdf", height = 30, width = 6, units = "in")

```
scale_colour_gradient(low = "magenta", high = "yellow")
```{R}
werd <- colorRampPalette(c("magenta","#C228C8","#813384","#000000","#817E3E","#BEB72F","#FFF300"))(200)


Top10DEGenesHeatmap_DEThreshold <- pheatmap(lfcframe, border_color = "gray", color = werd, cellwidth = 10, cellheight = 10, breaks = seq(from=-0.5,to=0.5, length.out=200), cluster_cols = FALSE)

Top10DEGenesHeatmap_0.4_Saturation_DEThreshold <- pheatmap(lfcframe, border_color = "gray", color = werd, cellwidth = 10, cellheight = 10, breaks = seq(from=-0.4,to=0.4, length.out=200), cluster_cols = FALSE)

Top10DEGenesHeatmap_0.3_Saturation_DEThreshold <- pheatmap(lfcframe, border_color = "gray", color = werd, cellwidth = 10, cellheight = 10, breaks = seq(from=-0.3,to=0.3, length.out=200), cluster_cols = FALSE)

ggsave(Top10DEGenesHeatmap_DEThreshold,filename = "Figures/Heatmaps/Top10DEGenesHeatmap_DEThreshold.pdf", device = "pdf", height = 30, width = 8, units = "in")

ggsave(Top10DEGenesHeatmap_0.4_Saturation_DEThreshold,filename = "Figures/Heatmaps/Top10DEGenesHeatmap_0.4_Saturation_DEThreshold.pdf", device = "pdf", height = 30, width = 8, units = "in")

ggsave(Top10DEGenesHeatmap_0.3_Saturation_DEThreshold,filename = "Figures/Heatmaps/Top10DEGenesHeatmap_0.3_Saturation_DEThreshold.pdf", device = "pdf", height = 30, width = 8, units = "in")
```

#############################################################################################################################################

Previous heatmap - Top 5 DE genes in each subcluster

Using top genes DE for each subcluster

```{R, message=FALSE}
library(Seurat)
library(cowplot)
library(ggplot2)
library(ggsci)
library(dplyr)
library(psych)
library(pheatmap)
library(harmony)
#library(clusterProfiler)
#library(org.Hs.eg.db)
library(DOSE)
library(GOSemSim)
library(enrichplot)
library(sciplot)
library(reshape2)
library(MAST)
library(fgsea)
library(SeuratWrappers)
library(purrr)
```






#Batch567_MTG$case <- gsub("ILB","PD",Batch567_MTG$case)
case_clusters <- paste(Batch567_MTG$case,Batch567_MTG$seurat_clusters,sep = "_")




Batch567_MTG <- RenameIdents(Batch567_MTG, `1` = "Oligodendrocytes", `2` = "GLU_Neurons_1", `3` = "GLU_Neurons_2",
                      `4` = "GLU_Neurons_3",`5` = "Astrocytes_1",
                      `6` = "GABA_Neurons_1", `7` = "Cajal_Retzius_Cells", `8` = "GLU_Neurons_4",`9` = "GLU_Neurons_5",
                      `10` = "GLU_Neurons_6", `11` = "Microglia",`12` = "Astrocytes_2",
                      `13` = "OPCs",`14` = "GABA_Neurons_2",
                      `15` = "GLU_Neurons_7", `16`="GLU_Neurons_8", `17`="Endothelial_1", `18`="Endothelial_2",`19`="GABA_Neurons_3",`20`="GLU_Neurons_9",`21`="GLU_Neurons_10",`22`="GLU_Neurons_11",`23`="GABA_Neurons_4")




```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/AllGenes_|.csv","",i),read.csv(i))
}

```



```{R}
DEdfList <- ls()[grep("HC_vs_PD",ls())]

#DEdfList <- DEdfList[grep("Microglia|Endothelial|GABA|GLU|Oligodendrocytes|OPCs|Astrocytes",DEdfList)]

#  tempdf <- tempdf[1:10,]

TopDEGenes=NULL
for (i in DEdfList){
  tempdf <- get(i)
  tempdf <- tempdf[tempdf$p_val <= tempdf$BF_p_val,]
  if (nrow(tempdf) >= 5){
  tempdf <- tempdf[1:5,]
  tempdf$CellType <- gsub("_Markers_HC_vs_PD","",i)
  TopDEGenes <- rbind(TopDEGenes,tempdf)
  }
}

```

```{R}

library(gdata)

keep(TopDEGenes, sure = TRUE)

```


```{R}

#setwd("~/Documents/Projects/PD5D_Repository/scRNA_and_scATAC-seq_Markdowns/batch1-3IntegrationAnalysis/scRNA-seq")
#test <- do.call(read.delim(list.files("Files/DE_Genes/", pattern = "*HC_vs_PD.csv")))

for (i in paste("Files/DE_Genes_Subclusters/",list.files("Files/DE_Genes_Subclusters/", pattern = "AllGenes.+HC_vs_PD.csv"), sep = "")){
  assign(gsub("Files/DE_Genes_Subclusters/|.csv","",i),read.csv(i))
}

```

```{R}

DEdfListAllGenes<- ls()[grep("AllGenes",ls())]

```

```{R}

lfcframe=NULL
Thresholdframe=NULL
for (geneident in unique(TopDEGenes$gene)){
  generow = NULL
  genethresholdrow = NULL
  for (i in DEdfListAllGenes){
    tempdf <- get(i)
    if (geneident %in% tempdf$gene){
      temp_lfc <- tempdf[tempdf$gene %in% geneident,]$avg_log2FC
      temp_pval <- tempdf[tempdf$gene %in% geneident,]$p_val
      if (temp_lfc > 0 & temp_pval <= unique(tempdf$BF_p_val)){
        temp_threshold <- 1
      } 
      
      if (temp_lfc < 0 & temp_pval <= unique(tempdf$BF_p_val)) {
        temp_threshold <- -1
      }
      if (temp_pval >= unique(tempdf$BF_p_val)) {
        temp_threshold <- 0
      }
      generow <- cbind(generow, temp_lfc)
      genethresholdrow <- cbind(genethresholdrow, temp_threshold)
    } else {
      generow <- cbind(generow, 0)
      genethresholdrow <- cbind(genethresholdrow,0)
    }
  }
 lfcframe <- rbind(lfcframe,generow)
 Thresholdframe <- rbind(Thresholdframe, genethresholdrow)
}

colnames(lfcframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(lfcframe) <- unique(TopDEGenes$gene)

colnames(Thresholdframe) <- gsub("AllGenes|_Markers_HC_vs_PD","",DEdfListAllGenes)
rownames(Thresholdframe) <- unique(TopDEGenes$gene)

#pvallog <- function(val){
#  logpval <- -log10(val)
#  return(logpval)
#}




#logPvalFrame <- t(apply(Pvalframe,1, pvallog))

```


```{R}

pheatmap(lfcframe)

```

```{R}


testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

testmap <- pheatmap(Thresholdframe, border_color = "gray", color = c("black","cornflowerblue","#E74C3C"), cellwidth = 20, cellheight = 10)

#ggsave(testmap,filename = "Files/HeatmapTest2.pdf", device = "pdf", height = 30, width = 6, units = "in")

```
scale_colour_gradient(low = "magenta", high = "yellow")
```{R}
werd <- colorRampPalette(c("magenta","#C228C8","#813384","#000000","#817E3E","#BEB72F","#FFF300"))(200)


testmap <- pheatmap(lfcframe, border_color = "gray", color = werd, cellwidth = 10, cellheight = 10, breaks = seq(from=-0.5,to=0.5, length.out=200))

ggsave(testmap,filename = "Files/HeatmapTest.pdf", device = "pdf", height = 30, width = 8, units = "in")

```








